{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "BERTbase_SegIDs.ipynb",
      "provenance": [],
      "collapsed_sections": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU",
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "0b30b13844ca4402b65101330f5a68b5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "state": {
            "_view_name": "HBoxView",
            "_dom_classes": [],
            "_model_name": "HBoxModel",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "box_style": "",
            "layout": "IPY_MODEL_9dcd0fe084364cb1be3705e0ebed2d2f",
            "_model_module": "@jupyter-widgets/controls",
            "children": [
              "IPY_MODEL_e7621ae87ee741d4b44e2b595c58c641",
              "IPY_MODEL_444e452b19be417181c538db9f4267b5"
            ]
          }
        },
        "9dcd0fe084364cb1be3705e0ebed2d2f": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "e7621ae87ee741d4b44e2b595c58c641": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "state": {
            "_view_name": "ProgressView",
            "style": "IPY_MODEL_5ecf8c7eae5a46688ba4291ab80dcea7",
            "_dom_classes": [],
            "description": "Downloading: 100%",
            "_model_name": "FloatProgressModel",
            "bar_style": "success",
            "max": 231508,
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": 231508,
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "orientation": "horizontal",
            "min": 0,
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_636ac0fd8ff14b3b8c05858226fd71d3"
          }
        },
        "444e452b19be417181c538db9f4267b5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "state": {
            "_view_name": "HTMLView",
            "style": "IPY_MODEL_b3f445ad554b49b5b7aa87ade8aabb58",
            "_dom_classes": [],
            "description": "",
            "_model_name": "HTMLModel",
            "placeholder": "​",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": " 232k/232k [00:02&lt;00:00, 94.9kB/s]",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_b7a88f5a84f0491999727d4314ee6b32"
          }
        },
        "5ecf8c7eae5a46688ba4291ab80dcea7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "ProgressStyleModel",
            "description_width": "initial",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "bar_color": null,
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "636ac0fd8ff14b3b8c05858226fd71d3": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "b3f445ad554b49b5b7aa87ade8aabb58": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "DescriptionStyleModel",
            "description_width": "",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "b7a88f5a84f0491999727d4314ee6b32": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "cb1c0507649940ed94f4bec361b176e4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "state": {
            "_view_name": "HBoxView",
            "_dom_classes": [],
            "_model_name": "HBoxModel",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "box_style": "",
            "layout": "IPY_MODEL_dcaa341008d4451da76fa6c607f28172",
            "_model_module": "@jupyter-widgets/controls",
            "children": [
              "IPY_MODEL_1f24979222c843b3be54a0dbb617ccd3",
              "IPY_MODEL_3412b2f1c5c84e4b99be28296bd43c6b"
            ]
          }
        },
        "dcaa341008d4451da76fa6c607f28172": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "1f24979222c843b3be54a0dbb617ccd3": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "state": {
            "_view_name": "ProgressView",
            "style": "IPY_MODEL_1677975bbd8342e984b8e0251ca5312a",
            "_dom_classes": [],
            "description": "Downloading: 100%",
            "_model_name": "FloatProgressModel",
            "bar_style": "success",
            "max": 28,
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": 28,
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "orientation": "horizontal",
            "min": 0,
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_ec6cc4b517294497b53381e0837260bd"
          }
        },
        "3412b2f1c5c84e4b99be28296bd43c6b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "state": {
            "_view_name": "HTMLView",
            "style": "IPY_MODEL_e2a281b7b79c4175be3d7ef59dc3acc6",
            "_dom_classes": [],
            "description": "",
            "_model_name": "HTMLModel",
            "placeholder": "​",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": " 28.0/28.0 [00:00&lt;00:00, 46.9B/s]",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_48f82f8bc04d4fc79c3f35374d16ac57"
          }
        },
        "1677975bbd8342e984b8e0251ca5312a": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "ProgressStyleModel",
            "description_width": "initial",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "bar_color": null,
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "ec6cc4b517294497b53381e0837260bd": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "e2a281b7b79c4175be3d7ef59dc3acc6": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "DescriptionStyleModel",
            "description_width": "",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "48f82f8bc04d4fc79c3f35374d16ac57": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "a65d5ddc0e9d47f4ac88f9ddb2f02ab0": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "state": {
            "_view_name": "HBoxView",
            "_dom_classes": [],
            "_model_name": "HBoxModel",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "box_style": "",
            "layout": "IPY_MODEL_696498cd19a44030851579ff8636ade2",
            "_model_module": "@jupyter-widgets/controls",
            "children": [
              "IPY_MODEL_2e5e320c279c4527af437cacee277b31",
              "IPY_MODEL_6fc62e6674244c59a45277c6852369f3"
            ]
          }
        },
        "696498cd19a44030851579ff8636ade2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "2e5e320c279c4527af437cacee277b31": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "state": {
            "_view_name": "ProgressView",
            "style": "IPY_MODEL_5bdfbedf3f2546b89934e33fac261dca",
            "_dom_classes": [],
            "description": "Downloading: 100%",
            "_model_name": "FloatProgressModel",
            "bar_style": "success",
            "max": 466062,
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": 466062,
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "orientation": "horizontal",
            "min": 0,
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_69d8345150114e9c9535c91bb03b5ea2"
          }
        },
        "6fc62e6674244c59a45277c6852369f3": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "state": {
            "_view_name": "HTMLView",
            "style": "IPY_MODEL_5c566e21cdb44e3aaec4b84508bb055e",
            "_dom_classes": [],
            "description": "",
            "_model_name": "HTMLModel",
            "placeholder": "​",
            "_view_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "value": " 466k/466k [00:00&lt;00:00, 1.06MB/s]",
            "_view_count": null,
            "_view_module_version": "1.5.0",
            "description_tooltip": null,
            "_model_module": "@jupyter-widgets/controls",
            "layout": "IPY_MODEL_13459093f7c64373a8a6828ef20e53a9"
          }
        },
        "5bdfbedf3f2546b89934e33fac261dca": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "ProgressStyleModel",
            "description_width": "initial",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "bar_color": null,
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "69d8345150114e9c9535c91bb03b5ea2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        },
        "5c566e21cdb44e3aaec4b84508bb055e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "state": {
            "_view_name": "StyleView",
            "_model_name": "DescriptionStyleModel",
            "description_width": "",
            "_view_module": "@jupyter-widgets/base",
            "_model_module_version": "1.5.0",
            "_view_count": null,
            "_view_module_version": "1.2.0",
            "_model_module": "@jupyter-widgets/controls"
          }
        },
        "13459093f7c64373a8a6828ef20e53a9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "state": {
            "_view_name": "LayoutView",
            "grid_template_rows": null,
            "right": null,
            "justify_content": null,
            "_view_module": "@jupyter-widgets/base",
            "overflow": null,
            "_model_module_version": "1.2.0",
            "_view_count": null,
            "flex_flow": null,
            "width": null,
            "min_width": null,
            "border": null,
            "align_items": null,
            "bottom": null,
            "_model_module": "@jupyter-widgets/base",
            "top": null,
            "grid_column": null,
            "overflow_y": null,
            "overflow_x": null,
            "grid_auto_flow": null,
            "grid_area": null,
            "grid_template_columns": null,
            "flex": null,
            "_model_name": "LayoutModel",
            "justify_items": null,
            "grid_row": null,
            "max_height": null,
            "align_content": null,
            "visibility": null,
            "align_self": null,
            "height": null,
            "min_height": null,
            "padding": null,
            "grid_auto_rows": null,
            "grid_gap": null,
            "max_width": null,
            "order": null,
            "_view_module_version": "1.2.0",
            "grid_template_areas": null,
            "object_position": null,
            "object_fit": null,
            "grid_auto_columns": null,
            "margin": null,
            "display": null,
            "left": null
          }
        }
      }
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/sahithikodali1/Summarization-of-Biomedical-evidence/blob/main/BERTbase_SegIDs.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Jymts-aLDSIJ",
        "outputId": "6f5bd7ad-2256-4d60-ae44-3fb5883d3ccf"
      },
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Mounted at /content/drive\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f_bsiLbS4jT4",
        "outputId": "cef85124-1492-410a-eabe-bfa9e4d4ce8f"
      },
      "source": [
        "import random\n",
        "import numpy as np\n",
        "import pandas as pd\n",
        "import os\n",
        "import json\n",
        "import csv\n",
        "import pickle\n",
        "import matplotlib.pyplot as plt\n",
        "import matplotlib.pyplot as plt\n",
        "% matplotlib inline\n",
        "\n",
        "# install\n",
        "!pip install pytorch-pretrained-bert pytorch-nlp\n",
        "\n",
        "# BERT imports\n",
        "import torch\n",
        "\n",
        "from torch.utils.data import TensorDataset, DataLoader, RandomSampler, SequentialSampler\n",
        "from keras.preprocessing.sequence import pad_sequences\n",
        "from sklearn.model_selection import train_test_split\n",
        "from tqdm import tqdm, trange\n",
        "\n",
        "from torch.nn import CrossEntropyLoss\n",
        "import copy\n",
        "from sklearn.metrics import f1_score\n",
        "\n",
        "!pip install transformers\n",
        "from transformers import get_linear_schedule_with_warmup\n",
        "\n",
        "# specify GPU device\n",
        "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
        "n_gpu = torch.cuda.device_count()\n",
        "torch.cuda.get_device_name(0)\n",
        "\n",
        "#Data file path\n",
        "DATA_DIR = \"/content/drive/MyDrive/Thesis_B\"\n",
        "file = '8b_data.csv'\n",
        "\n",
        "#Test file path\n",
        "test_file = '/content/drive/MyDrive/Thesis_B/8B5_golden.json'"
      ],
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Collecting pytorch-pretrained-bert\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/d7/e0/c08d5553b89973d9a240605b9c12404bcf8227590de62bae27acbcfe076b/pytorch_pretrained_bert-0.6.2-py3-none-any.whl (123kB)\n",
            "\u001b[K     |████████████████████████████████| 133kB 11.4MB/s \n",
            "\u001b[?25hCollecting pytorch-nlp\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/4f/51/f0ee1efb75f7cc2e3065c5da1363d6be2eec79691b2821594f3f2329528c/pytorch_nlp-0.5.0-py3-none-any.whl (90kB)\n",
            "\u001b[K     |████████████████████████████████| 92kB 7.8MB/s \n",
            "\u001b[?25hRequirement already satisfied: torch>=0.4.1 in /usr/local/lib/python3.7/dist-packages (from pytorch-pretrained-bert) (1.8.1+cu101)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from pytorch-pretrained-bert) (2.23.0)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from pytorch-pretrained-bert) (1.19.5)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.7/dist-packages (from pytorch-pretrained-bert) (4.41.1)\n",
            "Collecting boto3\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/29/fe/6156cb9ad11c01728adaaf6eebd855e66ac17d12a3826cb6fa6d3c2b3dab/boto3-1.17.78-py2.py3-none-any.whl (131kB)\n",
            "\u001b[K     |████████████████████████████████| 133kB 20.3MB/s \n",
            "\u001b[?25hRequirement already satisfied: regex in /usr/local/lib/python3.7/dist-packages (from pytorch-pretrained-bert) (2019.12.20)\n",
            "Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch>=0.4.1->pytorch-pretrained-bert) (3.7.4.3)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->pytorch-pretrained-bert) (3.0.4)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->pytorch-pretrained-bert) (1.24.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->pytorch-pretrained-bert) (2020.12.5)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->pytorch-pretrained-bert) (2.10)\n",
            "Collecting s3transfer<0.5.0,>=0.4.0\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/63/d0/693477c688348654ddc21dcdce0817653a294aa43f41771084c25e7ff9c7/s3transfer-0.4.2-py2.py3-none-any.whl (79kB)\n",
            "\u001b[K     |████████████████████████████████| 81kB 7.9MB/s \n",
            "\u001b[?25hCollecting botocore<1.21.0,>=1.20.78\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/b4/9e/b5c2ba127c653d22b5b9a48b173fb3e12f3bafc4f6c52004fa737157e63b/botocore-1.20.78-py2.py3-none-any.whl (7.5MB)\n",
            "\u001b[K     |████████████████████████████████| 7.6MB 14.8MB/s \n",
            "\u001b[?25hCollecting jmespath<1.0.0,>=0.7.1\n",
            "  Downloading https://files.pythonhosted.org/packages/07/cb/5f001272b6faeb23c1c9e0acc04d48eaaf5c862c17709d20e3469c6e0139/jmespath-0.10.0-py2.py3-none-any.whl\n",
            "Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in /usr/local/lib/python3.7/dist-packages (from botocore<1.21.0,>=1.20.78->boto3->pytorch-pretrained-bert) (2.8.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/dist-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.21.0,>=1.20.78->boto3->pytorch-pretrained-bert) (1.15.0)\n",
            "\u001b[31mERROR: botocore 1.20.78 has requirement urllib3<1.27,>=1.25.4, but you'll have urllib3 1.24.3 which is incompatible.\u001b[0m\n",
            "Installing collected packages: jmespath, botocore, s3transfer, boto3, pytorch-pretrained-bert, pytorch-nlp\n",
            "Successfully installed boto3-1.17.78 botocore-1.20.78 jmespath-0.10.0 pytorch-nlp-0.5.0 pytorch-pretrained-bert-0.6.2 s3transfer-0.4.2\n",
            "Collecting transformers\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/d5/43/cfe4ee779bbd6a678ac6a97c5a5cdeb03c35f9eaebbb9720b036680f9a2d/transformers-4.6.1-py3-none-any.whl (2.2MB)\n",
            "\u001b[K     |████████████████████████████████| 2.3MB 10.4MB/s \n",
            "\u001b[?25hCollecting tokenizers<0.11,>=0.10.1\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/ae/04/5b870f26a858552025a62f1649c20d29d2672c02ff3c3fb4c688ca46467a/tokenizers-0.10.2-cp37-cp37m-manylinux2010_x86_64.whl (3.3MB)\n",
            "\u001b[K     |████████████████████████████████| 3.3MB 31.4MB/s \n",
            "\u001b[?25hRequirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.7/dist-packages (from transformers) (2019.12.20)\n",
            "Collecting sacremoses\n",
            "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/75/ee/67241dc87f266093c533a2d4d3d69438e57d7a90abb216fa076e7d475d4a/sacremoses-0.0.45-py3-none-any.whl (895kB)\n",
            "\u001b[K     |████████████████████████████████| 901kB 37.1MB/s \n",
            "\u001b[?25hRequirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from transformers) (2.23.0)\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.7/dist-packages (from transformers) (1.19.5)\n",
            "Requirement already satisfied: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from transformers) (4.0.1)\n",
            "Collecting huggingface-hub==0.0.8\n",
            "  Downloading https://files.pythonhosted.org/packages/a1/88/7b1e45720ecf59c6c6737ff332f41c955963090a18e72acbcbeac6b25e86/huggingface_hub-0.0.8-py3-none-any.whl\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.7/dist-packages (from transformers) (3.0.12)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.7/dist-packages (from transformers) (4.41.1)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.7/dist-packages (from transformers) (20.9)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (1.0.1)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (1.15.0)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (8.0.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (2020.12.5)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (1.24.3)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (3.0.4)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (2.10)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata; python_version < \"3.8\"->transformers) (3.4.1)\n",
            "Requirement already satisfied: typing-extensions>=3.6.4; python_version < \"3.8\" in /usr/local/lib/python3.7/dist-packages (from importlib-metadata; python_version < \"3.8\"->transformers) (3.7.4.3)\n",
            "Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from packaging->transformers) (2.4.7)\n",
            "Installing collected packages: tokenizers, sacremoses, huggingface-hub, transformers\n",
            "Successfully installed huggingface-hub-0.0.8 sacremoses-0.0.45 tokenizers-0.10.2 transformers-4.6.1\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 163,
          "referenced_widgets": [
            "0b30b13844ca4402b65101330f5a68b5",
            "9dcd0fe084364cb1be3705e0ebed2d2f",
            "e7621ae87ee741d4b44e2b595c58c641",
            "444e452b19be417181c538db9f4267b5",
            "5ecf8c7eae5a46688ba4291ab80dcea7",
            "636ac0fd8ff14b3b8c05858226fd71d3",
            "b3f445ad554b49b5b7aa87ade8aabb58",
            "b7a88f5a84f0491999727d4314ee6b32",
            "cb1c0507649940ed94f4bec361b176e4",
            "dcaa341008d4451da76fa6c607f28172",
            "1f24979222c843b3be54a0dbb617ccd3",
            "3412b2f1c5c84e4b99be28296bd43c6b",
            "1677975bbd8342e984b8e0251ca5312a",
            "ec6cc4b517294497b53381e0837260bd",
            "e2a281b7b79c4175be3d7ef59dc3acc6",
            "48f82f8bc04d4fc79c3f35374d16ac57",
            "a65d5ddc0e9d47f4ac88f9ddb2f02ab0",
            "696498cd19a44030851579ff8636ade2",
            "2e5e320c279c4527af437cacee277b31",
            "6fc62e6674244c59a45277c6852369f3",
            "5bdfbedf3f2546b89934e33fac261dca",
            "69d8345150114e9c9535c91bb03b5ea2",
            "5c566e21cdb44e3aaec4b84508bb055e",
            "13459093f7c64373a8a6828ef20e53a9"
          ]
        },
        "id": "H0owgoQMvEik",
        "outputId": "0a757563-8494-41c1-c220-ff1eb3230a2a"
      },
      "source": [
        "from transformers import BertTokenizer\n",
        "\n",
        "# Tokenize with BERT tokenizer\n",
        "tokenizer = BertTokenizer.from_pretrained('bert-base-uncased', do_lower_case=True)"
      ],
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "0b30b13844ca4402b65101330f5a68b5",
              "version_minor": 0,
              "version_major": 2
            },
            "text/plain": [
              "HBox(children=(FloatProgress(value=0.0, description='Downloading', max=231508.0, style=ProgressStyle(descripti…"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "stream",
          "text": [
            "\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "cb1c0507649940ed94f4bec361b176e4",
              "version_minor": 0,
              "version_major": 2
            },
            "text/plain": [
              "HBox(children=(FloatProgress(value=0.0, description='Downloading', max=28.0, style=ProgressStyle(description_w…"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "stream",
          "text": [
            "\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.jupyter.widget-view+json": {
              "model_id": "a65d5ddc0e9d47f4ac88f9ddb2f02ab0",
              "version_minor": 0,
              "version_major": 2
            },
            "text/plain": [
              "HBox(children=(FloatProgress(value=0.0, description='Downloading', max=466062.0, style=ProgressStyle(descripti…"
            ]
          },
          "metadata": {
            "tags": []
          }
        },
        {
          "output_type": "stream",
          "text": [
            "\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "aJj7zIEd2Qth"
      },
      "source": [
        "def find_datasize(path,filename):\n",
        "    file = pd.read_csv(os.path.join(path,filename))\n",
        "    f_values = file.values\n",
        "    q_id_split = np.split(f_values, np.where(np.diff(f_values[:,0]))[0]+1)\n",
        "    col_names = file.columns\n",
        "    print(col_names)\n",
        "    print(file[:10])\n",
        "    train_data_size = round(len(q_id_split)*0.8)\n",
        "    val_data_size = round(len(q_id_split) - train_data_size)\n",
        "    print(\"Training data size:{}\".format(train_data_size))\n",
        "    print(\"Validation data size:{}\".format(val_data_size))\n",
        "    return q_id_split,train_data_size,val_data_size,col_names\n",
        "\n",
        "def split_dataframes(q_id_split,train_data_size,col_names):\n",
        "    random.seed(3007)\n",
        "    random.shuffle(q_id_split)\n",
        "    train_data = q_id_split[:train_data_size]\n",
        "    val_data = q_id_split[train_data_size:]\n",
        "    f_values_train = np.concatenate(train_data, axis=0)\n",
        "    f_values_val = np.concatenate(val_data, axis=0)\n",
        "    print(len(train_data))\n",
        "    print(len(val_data))\n",
        "    train_df = pd.DataFrame(f_values_train, columns = col_names)\n",
        "    val_df = pd.DataFrame(f_values_val, columns = col_names)\n",
        "    return train_df,val_df\n",
        "\n",
        "def obtain_SU4labels_list(dataframe):\n",
        "    SU4_labels = dataframe['SU4_labels']\n",
        "    labels_list = list(SU4_labels)\n",
        "    print('Labels size:{}'.format(len(labels_list)))\n",
        "    return labels_list"
      ],
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "vPtuKVvyiCLB",
        "outputId": "72e7c24a-bdbb-4d6a-be4f-8ff2ea92ebb8"
      },
      "source": [
        "q_id_split, train_data_size, val_data_size, col_names = find_datasize(DATA_DIR,file)\n",
        "train_df, val_df = split_dataframes(q_id_split,train_data_size,col_names)\n",
        "train_labels = obtain_SU4labels_list(train_df)\n",
        "val_labels = obtain_SU4labels_list(val_df)\n"
      ],
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Index(['qid', 'pubmedid', 'sentid', 'N1', 'N2', 'L', 'S4', 'SU4',\n",
            "       'sentence text', 'SU4_labels', 'question'],\n",
            "      dtype='object')\n",
            "   qid  ...                                           question\n",
            "0    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "1    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "2    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "3    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "4    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "5    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "6    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "7    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "8    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "9    0  ...  Is Hirschsprung disease a mendelian or a multi...\n",
            "\n",
            "[10 rows x 11 columns]\n",
            "Training data size:2594\n",
            "Validation data size:648\n",
            "2594\n",
            "648\n",
            "Labels size:41583\n",
            "Labels size:10161\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "roHVSbVZoh_7"
      },
      "source": [
        "#Add special tokens\n",
        "def obtain_specialtokenized_list(dataframe):\n",
        "    sentences = dataframe['sentence text']\n",
        "    questions = dataframe['question']\n",
        "    sentences_list = list(sentences)\n",
        "    questions_list = list(questions)\n",
        "    question_sentence_list = []\n",
        "    for i in range(len(sentences_list)):\n",
        "        question_sentence_list = question_sentence_list + [\"[CLS] \" + questions_list[i] + \" [SEP] \" + sentences_list[i] + \" [SEP]\"]\n",
        "    return question_sentence_list\n",
        "\n",
        "#Tokenize texts\n",
        "def tokenize_sent(givenlist):\n",
        "    tokenized_texts = [tokenizer.tokenize(sent) for sent in givenlist]\n",
        "    return tokenized_texts\n",
        "\n",
        "#Remove commas from tokens\n",
        "def remove_token_commas(givenlist):\n",
        "  no_commas_tokenlist = []\n",
        "  for each in givenlist:\n",
        "    y = [i.replace(',',';') for i in each]\n",
        "    no_commas_tokenlist.append(y)\n",
        "  return no_commas_tokenlist\n",
        "\n",
        "#Create segment ids from tokens\n",
        "def segment_id(givenlist):\n",
        "  MAX_LEN = 512\n",
        "  segment_ids = []\n",
        "  for each in givenlist:\n",
        "    token_sent = ','.join(each)\n",
        "    d = \"[SEP]\"\n",
        "    ques_ans =  [token+d for token in token_sent.split(d) if token]\n",
        "    ques = [0]*len(ques_ans[0].split(','))\n",
        "    ans = [1]*(len(ques_ans[1].split(','))-1)\n",
        "    seg_ids_ques_ans = ques+ans\n",
        "    length = len(seg_ids_ques_ans)\n",
        "    if length >= MAX_LEN:\n",
        "      length_current = MAX_LEN\n",
        "      seg_ids_ques_ans = seg_ids_ques_ans[:length_current]\n",
        "      segment_ids.append(seg_ids_ques_ans)\n",
        "    else:\n",
        "      length_current = MAX_LEN - length\n",
        "      seg_ids_ques_ans += [0]*length_current\n",
        "      segment_ids.append(seg_ids_ques_ans)\n",
        "  return segment_ids\n",
        "\n",
        "#Convert tokenized sentences to respective token ids\n",
        "def token2ids(tokenized_texts):\n",
        "    MAX_LEN = 512\n",
        "    tokens_to_ids = [tokenizer.convert_tokens_to_ids(sent) for sent in tokenized_texts]\n",
        "    tokens_to_ids = pad_sequences(tokens_to_ids, maxlen=MAX_LEN, dtype=\"long\", truncating=\"post\", padding=\"post\")\n",
        "    return tokens_to_ids\n",
        "\n",
        "#Create masks for the tokenids\n",
        "def create_masks(token_ids):\n",
        "    attention_masks = []\n",
        "    for tid in token_ids:\n",
        "        tid_mask = [float(i>0) for i in tid]\n",
        "        attention_masks.append(tid_mask)\n",
        "    return attention_masks\n"
      ],
      "execution_count": 25,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "eAPahYO2osPD"
      },
      "source": [
        "train_specialtok_list = obtain_specialtokenized_list(train_df)  \n",
        "val_specialtok_list = obtain_specialtokenized_list(val_df)  \n",
        "\n",
        "train_tokenized = tokenize_sent(train_specialtok_list)\n",
        "val_tokenized = tokenize_sent(val_specialtok_list)\n",
        "\n",
        "train_tokenized_nocommas = remove_token_commas(train_tokenized)\n",
        "val_tokenized_nocommas = remove_token_commas(val_tokenized)\n",
        "\n",
        "train_token_type_ids =  segment_id(train_tokenized_nocommas)\n",
        "\n",
        "train_tokenids = token2ids(train_tokenized_nocommas)\n",
        "val_tokenids = token2ids(val_tokenized_nocommas)\n",
        "\n",
        "train_masks = create_masks(train_tokenids)\n",
        "val_masks = create_masks(val_tokenids)"
      ],
      "execution_count": 8,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rU8l-4B7pQ1K",
        "outputId": "197474dc-5138-40b3-8032-ba15ef741842"
      },
      "source": [
        "# Convert all of our data into torch tensors, the required datatype for our model\n",
        "train_inputs = torch.tensor(train_tokenids, dtype=torch.int64)\n",
        "validation_inputs = torch.tensor(val_tokenids, dtype=torch.int64)\n",
        "\n",
        "train_labels = torch.tensor(train_labels, dtype=torch.int64)\n",
        "validation_labels = torch.tensor(val_labels, dtype=torch.int64)\n",
        "\n",
        "train_masks = torch.tensor(train_masks, dtype=torch.float32)\n",
        "validation_masks = torch.tensor(val_masks, dtype=torch.float32)\n",
        "\n",
        "train_token_type_ids = torch.tensor(train_token_type_ids, dtype=torch.int64)\n",
        "\n",
        "print(type(train_inputs))\n",
        "print(type(train_masks))\n",
        "print(type(train_labels))\n",
        "print(type(train_token_type_ids))\n",
        "print('***************')\n",
        "print(train_inputs.dtype)\n",
        "print(train_masks.dtype)\n",
        "print(train_labels.dtype)\n",
        "print(train_token_type_ids.dtype)\n",
        "print('***************')\n",
        "print(validation_inputs.dtype)\n",
        "print(validation_masks.dtype)\n",
        "print(validation_labels.dtype)\n",
        "\n",
        "print('***************')\n",
        "print(train_inputs.shape)\n",
        "print(train_masks.shape)\n",
        "print(train_labels.shape)\n",
        "print(train_token_type_ids.shape)\n",
        "print('***************')\n",
        "print(train_inputs[0].shape)\n",
        "print(train_masks[0].shape)\n",
        "print(train_labels[0].shape)\n",
        "print(train_token_type_ids[0].shape)\n",
        "print('***************')\n",
        "print(train_inputs[0])\n",
        "print(train_masks[0])\n",
        "print(train_labels[0])\n",
        "print(train_token_type_ids[0])\n",
        "\n",
        "\n",
        "# Select a batch size for training. \n",
        "batch_size = 32\n",
        "\n",
        "# Create an iterator of our data with torch DataLoader \n",
        "train_data = TensorDataset(train_inputs, train_token_type_ids, train_masks, train_labels)\n",
        "train_sampler = RandomSampler(train_data)\n",
        "train_dataloader = DataLoader(train_data, sampler=train_sampler, batch_size=batch_size)\n"
      ],
      "execution_count": 11,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:5: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).\n",
            "  \"\"\"\n",
            "/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:8: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).\n",
            "  \n",
            "/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:11: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).\n",
            "  # This is added back by InteractiveShellApp.init_path()\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "<class 'torch.Tensor'>\n",
            "<class 'torch.Tensor'>\n",
            "<class 'torch.Tensor'>\n",
            "<class 'torch.Tensor'>\n",
            "***************\n",
            "torch.int64\n",
            "torch.float32\n",
            "torch.int64\n",
            "torch.int64\n",
            "***************\n",
            "torch.int64\n",
            "torch.float32\n",
            "torch.int64\n",
            "***************\n",
            "torch.Size([41583, 512])\n",
            "torch.Size([41583, 512])\n",
            "torch.Size([41583])\n",
            "torch.Size([41583, 512])\n",
            "***************\n",
            "torch.Size([512])\n",
            "torch.Size([512])\n",
            "torch.Size([])\n",
            "torch.Size([512])\n",
            "***************\n",
            "tensor([  101,  2054,  2828,  1997, 16221,  2003,  4786,  1996,  3919, 11463,\n",
            "         7088,  6491,  6887, 16515, 13874,  1999, 11565,  2098, 14885,  1029,\n",
            "          102,  2182,  2057,  2265,  2008,  1996, 16221,  2724,  3228,  4125,\n",
            "         2000,  3919, 11463,  7088,  6491,  1999,  3725,  2001,  1996, 23851,\n",
            "         1997,  1037,  2312,  1025, 18231,  2135,  5567,  1025,  9099,  6873,\n",
            "        19150,  5783,  2046,  1996,  2034, 17174,  2078,  1997,  1996,  4962,\n",
            "        17132,  1012,   102,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0,     0,     0,     0,     0,     0,     0,     0,     0,\n",
            "            0,     0])\n",
            "tensor([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
            "        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
            "        1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,\n",
            "        1., 1., 1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n",
            "        0., 0., 0., 0., 0., 0., 0., 0.])\n",
            "tensor(1)\n",
            "tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1,\n",
            "        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,\n",
            "        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,\n",
            "        0, 0, 0, 0, 0, 0, 0, 0])\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8mus3xuHfral",
        "outputId": "34d77c8f-655d-4052-f364-73cb0f29cde1"
      },
      "source": [
        "val_token_type_ids = segment_id(val_tokenized_nocommas)\n",
        "validation_token_type_ids = torch.tensor(val_token_type_ids, dtype=torch.int64)\n",
        "print(validation_token_type_ids.dtype)\n",
        "print(validation_token_type_ids.shape)\n",
        "\n",
        "validation_data = TensorDataset(validation_inputs, validation_token_type_ids, validation_masks, validation_labels)\n",
        "validation_sampler = SequentialSampler(validation_data)\n",
        "validation_dataloader = DataLoader(validation_data, sampler=validation_sampler, batch_size=batch_size)"
      ],
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "torch.int64\n",
            "torch.Size([10161, 512])\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "Gr7OCmvcqlZj"
      },
      "source": [
        "seed_val = 42\n",
        "\n",
        "random.seed(seed_val)\n",
        "np.random.seed(seed_val)\n",
        "torch.manual_seed(seed_val)\n",
        "torch.cuda.manual_seed_all(seed_val)\n",
        "\n",
        "def flat_accuracy(preds, labels):\n",
        "    pred_flat = np.argmax(preds, axis=1).flatten()\n",
        "    labels_flat = labels.flatten()\n",
        "    return np.sum(pred_flat == labels_flat) / len(labels_flat)\n",
        "\n",
        "def flat_f1score(preds, labels):\n",
        "    pred_flat = np.argmax(preds, axis=1).flatten()\n",
        "    labels_flat = labels.flatten()\n",
        "    return f1_score(labels_flat, pred_flat)"
      ],
      "execution_count": 12,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "fuD8uF9xthGq"
      },
      "source": [
        "#Define model\n",
        "def model(type,pretrainedmodel):\n",
        "    model = type.from_pretrained(pretrainedmodel, num_labels = 2)\n",
        "    for param in model.bert.parameters():\n",
        "        param.requires_grad = False\n",
        "    return model.cuda()"
      ],
      "execution_count": 18,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "u387HeE_tZ_s",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "119c151a-64f3-4ad9-df9e-7db13d7fac56"
      },
      "source": [
        "#Call the model\n",
        "from pytorch_pretrained_bert import BertAdam, BertForSequenceClassification\n",
        "model = model(BertForSequenceClassification,'bert-base-uncased')\n",
        "\n",
        "#Define epochs\n",
        "epochs = 4\n",
        "\n",
        "#Optimizer & Scheduler fine-tuning parameters\n",
        "lr=2e-5\n",
        "num_warmup_steps = 10\n",
        "num_training_steps = 1000\n",
        "\n",
        "#Optmizer and Scheduler\n",
        "optimizer = torch.optim.AdamW(model.parameters(), lr = lr)\n",
        "scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps = num_warmup_steps, num_training_steps = num_training_steps)\n"
      ],
      "execution_count": 19,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "100%|██████████| 407873900/407873900 [00:13<00:00, 30771582.80B/s]\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "7ln3ADK5qXMT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "7597d38b-be42-44d8-b046-3e11258e0ef8"
      },
      "source": [
        "# Store our loss and accuracy for plotting\n",
        "train_loss_set = []\n",
        "best_accuracy_val = 0\n",
        "best_epoch = -1\n",
        "best_epoch_weights = copy.deepcopy(model.state_dict())\n",
        "\n",
        "for epoch in trange(epochs, desc=\"Epoch\"):  \n",
        "  ## TRAINING\n",
        "  # Set our model to training mode\n",
        "  model.train()  \n",
        "  tr_loss = 0\n",
        "  tr_accuracy, tr_f1score = 0, 0\n",
        "  nb_tr_steps = 0\n",
        "  \n",
        "  # Train the data for one epoch\n",
        "  for step, batch in enumerate(train_dataloader):\n",
        "\n",
        "    batch = tuple(t.to(device) for t in batch)\n",
        "    b_input_ids, b_input_tokenids, b_input_masks, b_labels = batch\n",
        "\n",
        "    # Clear out the gradients (by default they accumulate)\n",
        "    optimizer.zero_grad()\n",
        "    \n",
        "    # Forward pass\n",
        "    logits = model(b_input_ids, token_type_ids = b_input_tokenids, attention_mask = b_input_masks, labels = None)\n",
        "\n",
        "    loss_fn = CrossEntropyLoss()\n",
        "    loss = loss_fn(logits.view(-1, 2), b_labels.view(-1))\n",
        "\n",
        "    loss.backward()\n",
        "\n",
        "    # Move logits and labels to CPU\n",
        "    logits = logits.detach().cpu().numpy()\n",
        "    label_ids = b_labels.to('cpu').numpy()\n",
        "    \n",
        "    optimizer.step()\n",
        "    scheduler.step()\n",
        "\n",
        "    tr_loss += loss.item()\n",
        "    tmp_tr_accuracy = flat_accuracy(logits, label_ids)\n",
        "    tmp_tr_f1score = flat_f1score(logits, label_ids)    \n",
        "    tr_accuracy += tmp_tr_accuracy\n",
        "    tr_f1score += tmp_tr_f1score\n",
        "    nb_tr_steps += 1\n",
        "\n",
        "  epoch_loss = tr_loss/nb_tr_steps\n",
        "  epoch_accuracy = tr_accuracy/nb_tr_steps\n",
        "  epoch_f1score = tr_f1score/nb_tr_steps\n",
        "  train_loss_set.append(epoch_loss)\n",
        "\n",
        "  print(\"Train loss: {}\".format(epoch_loss))\n",
        "  print(\"Training Accuracy for epoch: {}\".format(epoch_accuracy))\n",
        "  print(\"Training f1score: {}\".format(epoch_f1score))\n",
        "\n",
        "  if (epoch_accuracy > best_accuracy_val):\n",
        "    best_accuracy_val = epoch_accuracy\n",
        "    best_epoch = epoch \n",
        "    torch.save(model.state_dict(), os.path.join(DATA_DIR, 'Bert-8b_5b_epoch_{}.pth'.format(epoch)))\n",
        "\n",
        "print(\" Best epoch: {}\".format(best_epoch))\n",
        "print(\" Best Accuracy: {}\".format(best_accuracy_val))"
      ],
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "\n",
            "Epoch:   0%|          | 0/4 [00:00<?, ?it/s]\u001b[A"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Train loss: 0.6036675828007552\n",
            "Training Accuracy for epoch: 0.7125560897435897\n",
            "Training f1score: 0.005838001741847899\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "\n",
            "Epoch:  25%|██▌       | 1/4 [49:44<2:29:14, 2984.98s/it]\u001b[A\n",
            "Epoch:  50%|█████     | 2/4 [1:39:30<1:39:30, 2985.02s/it]\u001b[A"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Train loss: 0.6041521448355455\n",
            "Training Accuracy for epoch: 0.7123221153846153\n",
            "Training f1score: 0.006263866475404939\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "\n",
            "Epoch:  75%|███████▌  | 3/4 [2:29:13<49:44, 2984.59s/it]  \u001b[A"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Train loss: 0.6043102850593053\n",
            "Training Accuracy for epoch: 0.7122676282051281\n",
            "Training f1score: 0.00594399680259409\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "\n",
            "Epoch: 100%|██████████| 4/4 [3:19:12<00:00, 2988.14s/it]"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Train loss: 0.6043631578867252\n",
            "Training Accuracy for epoch: 0.7123189102564103\n",
            "Training f1score: 0.006214985868832024\n",
            " Best epoch: 0\n",
            " Best Accuracy: 0.7125560897435897\n"
          ],
          "name": "stdout"
        },
        {
          "output_type": "stream",
          "text": [
            "\n"
          ],
          "name": "stderr"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "VVoG0shXqdSC",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b36dddf8-cd5d-41f5-d3ac-01c666bd329b"
      },
      "source": [
        "model.eval()\n",
        "\n",
        "val_loss_set = []\n",
        "eval_loss, eval_accuracy, eval_f1score = 0, 0, 0\n",
        "nb_eval_steps = 0\n",
        "\n",
        "# Evaluate data for one epoch\n",
        "for batch in validation_dataloader:\n",
        "    batch = tuple(t.to(device) for t in batch)\n",
        "    \n",
        "    # Unpack the inputs from our dataloader\n",
        "    b_input_ids, b_input_tokenids, b_input_masks, b_labels = batch\n",
        "\n",
        "    # Telling the model not to compute or store gradients, saving memory and speeding up validation\n",
        "    with torch.no_grad():\n",
        "        logits = model(b_input_ids, token_type_ids = b_input_tokenids, attention_mask = b_input_masks, labels = None)\n",
        "          \n",
        "    loss_fn = CrossEntropyLoss()\n",
        "    loss = loss_fn(logits.view(-1, 2), b_labels.view(-1))\n",
        "\n",
        "    # Move logits and labels to CPU\n",
        "    logits = logits.detach().cpu().numpy()\n",
        "    label_ids = b_labels.to('cpu').numpy()\n",
        "\n",
        "    eval_loss += loss.item()\n",
        "    tmp_eval_accuracy = flat_accuracy(logits, label_ids)\n",
        "    tmp_eval_f1score = flat_f1score(logits, label_ids)\n",
        "    eval_accuracy += tmp_eval_accuracy\n",
        "    eval_f1score += tmp_eval_f1score\n",
        "    nb_eval_steps += 1\n",
        "\n",
        "    val_epoch_loss = eval_loss/nb_eval_steps\n",
        "    val_epoch_accuracy = eval_accuracy/nb_eval_steps\n",
        "    epoch_f1score = eval_f1score/nb_eval_steps\n",
        "    val_loss_set.append(val_epoch_loss)\n",
        "\n",
        "print(\"Validation loss: {}\".format(val_epoch_loss))\n",
        "print(\"Validation Accuracy: {}\".format(val_epoch_accuracy))\n",
        "print(\"Validation f1score: {}\".format(epoch_f1score))\n"
      ],
      "execution_count": 33,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1515: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 due to no true nor predicted samples. Use `zero_division` parameter to control this behavior.\n",
            "  average, \"true nor predicted\", 'F-score is', len(true_sum)\n"
          ],
          "name": "stderr"
        },
        {
          "output_type": "stream",
          "text": [
            "Validation loss: 0.6072351404893324\n",
            "Validation Accuracy: 0.7035180817610063\n",
            "Validation f1score: 0.003650793650793651\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "TS4RdNy_0ufk"
      },
      "source": [
        "#Code to load jsonfile\n",
        "\n",
        "from pandas.io.json import json_normalize \n",
        "\n",
        "with open(test_file, 'r') as json_file:\n",
        "    data = json.load(json_file)\n",
        "\n",
        "##Code to get questions list\n",
        "def get_testdatalists(data):\n",
        "    data_access = data['questions']\n",
        "    id_list = []\n",
        "    type_list = []\n",
        "    body_list = []\n",
        "    all_text_list = []\n",
        "    for i in range(len(data_access)):\n",
        "      text_list = []\n",
        "      id_list.append(data_access[i]['id'])\n",
        "      type_list.append(data_access[i]['type'])\n",
        "      body_list.append(data_access[i]['body'])\n",
        "      for j in data_access[i]['snippets']:\n",
        "          text_list.append(j['text'])\n",
        "      all_text_list.append(text_list)\n",
        "    return id_list, body_list, type_list, all_text_list\n",
        "\n",
        "#Make a dataframe for test data\n",
        "def get_dataframes(id_list, body_list, type_list, all_text_list):\n",
        "    test_df = pd.DataFrame()\n",
        "    test_df['id'] = id_list\n",
        "    test_df['body'] = body_list\n",
        "    test_df['type'] = type_list\n",
        "    test_df['sentences'] = all_text_list\n",
        "    print(test_df[:10])\n",
        "    return test_df\n",
        "\n",
        "#Gte list of questions and sentences\n",
        "def get_datalists(test_df):\n",
        "    qid_test_df = test_df['id']\n",
        "    type_test_df= test_df['type']\n",
        "    sentences_test_df = test_df['sentences']\n",
        "    questions_test_df = test_df['body']\n",
        "    qid_list_test_df = list(qid_test_df)\n",
        "    type_list_test_df = list(type_test_df)\n",
        "    sentences_list_test_df = list(sentences_test_df)\n",
        "    questions_list_test_df = list(questions_test_df)\n",
        "    return questions_list_test_df, sentences_list_test_df\n",
        "\n",
        "#Add special tokens\n",
        "def join_ques_sent(questions_list_test_df, sentences_list_test_df):\n",
        "    question_sentence_list_test_df = []\n",
        "    for i in range(len(questions_list_test_df)):\n",
        "      each_list = []\n",
        "      for j in sentences_list_test_df[i]:\n",
        "          each_list= each_list + [\"[CLS] \" + questions_list_test_df[i] + \" [SEP] \" + j + \" [SEP]\"]\n",
        "      question_sentence_list_test_df.append(each_list)\n",
        "    return question_sentence_list_test_df\n",
        "\n",
        "#Tokenize and create tokenids, typeids and masks\n",
        "def create_test_tokens_masks(question_sentence_list_test_df):\n",
        "    token_ids_test_df_all = []\n",
        "    type_ids_test_df_all = []\n",
        "    masks_test_df_all = []\n",
        "    for ques_ans_sent in question_sentence_list_test_df:\n",
        "        tokenized_test_df = tokenize_sent(ques_ans_sent)\n",
        "        no_commas_tokenlist_test_df = remove_token_commas(tokenized_test_df)\n",
        "        \n",
        "        token_ids_test_df = token2ids(no_commas_tokenlist_test_df)\n",
        "        masks_test_df = create_masks(token_ids_test_df)\n",
        "        type_ids_test_df = segment_id(no_commas_tokenlist_test_df)\n",
        "\n",
        "        token_ids_test_df_all.append(token_ids_test_df)\n",
        "        type_ids_test_df_all.append(type_ids_test_df)\n",
        "        masks_test_df_all.append(masks_test_df)\n",
        "    return token_ids_test_df_all, type_ids_test_df_all, masks_test_df_all"
      ],
      "execution_count": 39,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RLxUoHZQ3332",
        "outputId": "bc1d7635-bbba-47d4-f7f4-04f6ce769aeb"
      },
      "source": [
        "id_list, body_list, type_list, all_text_list = get_testdatalists(data)\n",
        "test_df = get_dataframes(id_list, body_list, type_list, all_text_list)\n",
        "\n",
        "questions_list_test_df, sentences_list_test_df = get_datalists(test_df)\n",
        "question_sentence_list_test_df = join_ques_sent(questions_list_test_df, sentences_list_test_df)\n",
        "\n",
        "token_ids_test_df_all, type_ids_test_df_all, masks_test_df_all = create_test_tokens_masks(question_sentence_list_test_df)"
      ],
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "                         id  ...                                          sentences\n",
            "0  5d387360a1e1595105000007  ...  [CaMK4-dependent activation of AKT/mTOR and CR...\n",
            "1  5e4949d36d0a277941000006  ...  [Genetic mapping studies reveal that mutations...\n",
            "2  5e4b540b6d0a27794100001c  ...  [Camrelizumab (SHR-1210) is a humanised anti-p...\n",
            "3  5e822615835f4e4777000034  ...  [Erythropoietic protoporphyria (EPP) is a rare...\n",
            "4  5e476da1d14c9f295d000002  ...  [The phase III trial in small cell lung cancer...\n",
            "5  5d386ed6a1e1595105000004  ...  [A TNFR2 3' flanking region polymorphism in sy...\n",
            "6  5e51de866d0a27794100003f  ...  [VISMapper: ultra-fast exhaustive cartography ...\n",
            "7  5e3c83c548dab47f26000001  ...  [Tauopathies are a group of neurodegenerative ...\n",
            "8  5e4b64126d0a277941000028  ...  [A phase III study failed for carboplatin, pac...\n",
            "9  5e3ab58db5b409ea5300001c  ...  [ Library of Integrated Network-based Cellular...\n",
            "\n",
            "[10 rows x 4 columns]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5gWH8YU3Aquv",
        "outputId": "2733c971-9ea8-4adb-896a-eba79c7caa95"
      },
      "source": [
        "print(len(token_ids_test_df_all))\n",
        "print(len(type_ids_test_df_all))\n",
        "print(len(masks_test_df_all))\n",
        "\n",
        "print(type(token_ids_test_df_all[0]))\n",
        "print(type(type_ids_test_df_all[0]))\n",
        "print(type(masks_test_df_all[0]))\n",
        "\n",
        "print(token_ids_test_df_all[0])\n",
        "print(type_ids_test_df_all[0])\n",
        "print(masks_test_df_all[0])"
      ],
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "100\n",
            "100\n",
            "100\n",
            "<class 'numpy.ndarray'>\n",
            "<class 'list'>\n",
            "<class 'list'>\n",
            "[[ 101 2054 2024 ...    0    0    0]\n",
            " [ 101 2054 2024 ...    0    0    0]\n",
            " [ 101 2054 2024 ...    0    0    0]\n",
            " [ 101 2054 2024 ...    0    0    0]\n",
            " [ 101 2054 2024 ...    0    0    0]\n",
            " [ 101 2054 2024 ...    0    0    0]]\n",
            "[[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]\n",
            "[[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]]\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4GzdNime34Kd",
        "outputId": "f61d1276-03b4-4f3a-d5ab-d7e2347c417c"
      },
      "source": [
        "topsentences_summary = []\n",
        "\n",
        "for i in range(len(token_ids_test_df_all)):\n",
        "  batch_size = len(token_ids_test_df_all[i])\n",
        "\n",
        "  test_inputs = torch.tensor(token_ids_test_df_all[i], dtype=torch.int64)\n",
        "  test_masks = torch.tensor(masks_test_df_all[i], dtype=torch.float32)\n",
        "  test_token_ids = torch.tensor(type_ids_test_df_all[i], dtype=torch.int64)\n",
        "\n",
        "  test_data = TensorDataset(test_inputs, test_token_ids, test_masks)\n",
        "  test_sampler = SequentialSampler(test_data)\n",
        "  test_dataloader = DataLoader(test_data, sampler=test_sampler, batch_size=batch_size)\n",
        "\n",
        "  logits_list_test_df = []\n",
        "\n",
        "  model.eval()\n",
        "\n",
        "  # Evaluate data for one epoch\n",
        "  for batch in test_dataloader:\n",
        "\n",
        "    # Add batch to GPU\n",
        "    batch = tuple(t.to(device) for t in batch)\n",
        "    \n",
        "    # Unpack the inputs from our dataloader\n",
        "    b_input_ids, b_input_tokenids, b_input_masks = batch\n",
        "    \n",
        "    # Telling the model not to compute or store gradients, saving memory and speeding up validation\n",
        "    with torch.no_grad():\n",
        "      \n",
        "      # Forward pass, calculate logit predictions\n",
        "      logits = model(b_input_ids, token_type_ids = b_input_tokenids, attention_mask = b_input_masks, labels = None)   \n",
        "      logits_list_test_df.append(logits)\n",
        "    \n",
        "    prediction = torch.sigmoid(logits)\n",
        "    print(\"Softmax:{}\".format(prediction))\n",
        "\n",
        "    if len(prediction) >= 5:\n",
        "      values, indices = torch.topk(prediction,5,dim=0)\n",
        "      print(\"top_indices:{}\".format(indices))\n",
        "      indexes = indices[:,0].tolist()\n",
        "      each_summ = []\n",
        "      for ind in indexes:\n",
        "        each_summ.append(test_df['sentences'][i][ind])\n",
        "      each_summ = ' '.join(map(str, each_summ))\n",
        "      topsentences_summary.append(each_summ)\n",
        "      print(each_summ)\n",
        "    else:\n",
        "      values, indices = torch.topk(prediction,len(prediction),dim=0)\n",
        "      print(\"top_indices:{}\".format(indices))\n",
        "      indexes = indices[:,0].tolist()\n",
        "      each_summ = []\n",
        "      for ind in indexes:\n",
        "        each_summ.append(test_df['sentences'][i][ind])\n",
        "      each_summ = ' '.join(map(str, each_summ))\n",
        "      topsentences_summary.append(each_summ)\n",
        "      print(each_summ)\n",
        "\n",
        "    # Move logits to CPU\n",
        "    logits = logits.detach().cpu().numpy()\n",
        "    print(\"Logits:{}\".format(logits))\n",
        "\n",
        "print(\"summary:{}\".format(topsentences_summary))\n"
      ],
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "stream",
          "text": [
            "Softmax:tensor([[0.6364, 0.4174],\n",
            "        [0.6392, 0.4241],\n",
            "        [0.6363, 0.4169],\n",
            "        [0.6370, 0.4170],\n",
            "        [0.6322, 0.4242],\n",
            "        [0.6364, 0.4200]], device='cuda:0')\n",
            "top_indices:tensor([[1, 4],\n",
            "        [3, 1],\n",
            "        [0, 5],\n",
            "        [5, 0],\n",
            "        [2, 3]], device='cuda:0')\n",
            " Here, we present evidence that the calcium/calmodulin-dependent protein kinase IV (CaMK4) is increased and required during Th17 cell differentiation. CaMK4 inhibition has potential as a therapeutic strategy for Th17-driven autoimmune diseases. CaMK4-dependent activation of AKT/mTOR and CREM-α underlies autoimmunity-associated Th17 imbalance CaMK4 inhibition significantly decreased the levels of glycolytic intermediates such as glucose-6-phosphate, fructose-6-phosphate, fructose-1,6-diphosphate, pyruvate, and lactate, whereas it did not affect the levels of the pentose phosphate pathway intermediates such as 6-Phospho-D-gluconate, ribulose-5-phosphate, ribose-5-phosphate and phosphoribosyl pyrophosphate Inhibition of CaMK4 reduced Il17 transcription through decreased activation of the cAMP response element modulator α (CREM-α) and reduced activation of the AKT/mTOR pathway, which is known to enhance Th17 differentiation.\n",
            "Logits:[[ 0.5598452  -0.33340815]\n",
            " [ 0.57202    -0.30587906]\n",
            " [ 0.55918723 -0.33557555]\n",
            " [ 0.56250596 -0.33519155]\n",
            " [ 0.54181075 -0.3055243 ]\n",
            " [ 0.5596087  -0.32269338]]\n",
            "Softmax:tensor([[0.6392, 0.4192],\n",
            "        [0.6403, 0.4196]], device='cuda:0')\n",
            "top_indices:tensor([[1, 1],\n",
            "        [0, 0]], device='cuda:0')\n",
            "A new study identifies homozygous missense mutations in SGOL1, which encodes a component of the cohesin complex, in a newly described disorder termed Chronic Atrial and Intestinal Dysrhythmia (CAID) syndrome. These findings implicate cohesin in the regulation of intrinsic cardiac and intestinal rhythm and further expand the growing group of disorders termed the cohesinopathies. Genetic mapping studies reveal that mutations in cohesion pathways are responsible for multispectrum developmental abnormalities termed cohesinopathies. These include Roberts syndrome (RBS), Cornelia de Lange Syndrome (CdLS), and Warsaw Breakage Syndrome (WABS). The cohesinopathies are characterized by overlapping phenotypes ranging from craniofacial deformities, limb defects, and mental retardation. Though these syndromes share a similar suite of phenotypes and arise due to mutations in a common cohesion pathway, the underlying mechanisms are currently believed to be distinct. Defects in mitotic failure and apoptosis i.e. trans DNA tethering events are believed to be the underlying cause of RBS, whereas the underlying cause of CdLS is largely modeled as occurring through defects in transcriptional processes i.e. cis DNA tethering events. Here, we review recent findings described primarily in zebrafish, paired with additional studies in other model systems, including human patient cells, which challenge the notion that cohesinopathies represent separate syndromes. \n",
            "Logits:[[ 0.57203645 -0.32588154]\n",
            " [ 0.57649475 -0.3242496 ]]\n",
            "Softmax:tensor([[0.6407, 0.4135],\n",
            "        [0.6413, 0.4140],\n",
            "        [0.6379, 0.4191],\n",
            "        [0.6407, 0.4135],\n",
            "        [0.6385, 0.4140],\n",
            "        [0.6375, 0.4155],\n",
            "        [0.6382, 0.4169],\n",
            "        [0.6379, 0.4191],\n",
            "        [0.6420, 0.4143],\n",
            "        [0.6407, 0.4135],\n",
            "        [0.6379, 0.4191],\n",
            "        [0.6420, 0.4143],\n",
            "        [0.6388, 0.4176],\n",
            "        [0.6401, 0.4154],\n",
            "        [0.6395, 0.4156],\n",
            "        [0.6445, 0.4124]], device='cuda:0')\n",
            "top_indices:tensor([[15, 10],\n",
            "        [11,  7],\n",
            "        [ 8,  2],\n",
            "        [ 1, 12],\n",
            "        [ 0,  6]], device='cuda:0')\n",
            "All patients were administered intravenous camrelizumab (SHR-1210), a novel anti-PD-1 antibody, at doses of 60 mg, 200 mg, or 400 mg (4-week interval after first dose followed by a 2-week schedule) until disease progression or intolerable toxicity. Recent phase 1-2 trials reported manageable safety profiles and promising antitumor activities of anti-PD-1 drugs (pembrolizumab, nivolumab, camrelizumab and JS001) with/without chemotherapy in recurrent/metastatic nasopharyngeal carcinoma (RM-NPC), however head-to-head comparison among these regimens is lacking. Recent phase 1-2 trials reported manageable safety profiles and promising antitumor activities of anti-PD-1 drugs (pembrolizumab, nivolumab, camrelizumab and JS001) with/without chemotherapy in recurrent/metastatic nasopharyngeal carcinoma (RM-NPC), however head-to-head comparison among these regimens is lacking. The PD-1 antibody camrelizumab was well tolerated in patients with nasopharyngeal carcinoma. Camrelizumab (SHR-1210) is a humanised anti-programmed death-1 (anti PD-1) antibody.\n",
            "Logits:[[ 0.5782651  -0.34971565]\n",
            " [ 0.58117133 -0.34761292]\n",
            " [ 0.5660982  -0.32650957]\n",
            " [ 0.5782651  -0.34971565]\n",
            " [ 0.56898093 -0.34737268]\n",
            " [ 0.5646283  -0.34112865]\n",
            " [ 0.56774867 -0.33562735]\n",
            " [ 0.5660982  -0.32650957]\n",
            " [ 0.58424634 -0.34625244]\n",
            " [ 0.5782651  -0.34971565]\n",
            " [ 0.5660982  -0.32650957]\n",
            " [ 0.58424634 -0.34625244]\n",
            " [ 0.5703439  -0.33247682]\n",
            " [ 0.57593524 -0.34184003]\n",
            " [ 0.57325065 -0.34084925]\n",
            " [ 0.5949714  -0.35422334]]\n",
            "Softmax:tensor([[0.6340, 0.4150],\n",
            "        [0.6339, 0.4152],\n",
            "        [0.6340, 0.4189]], device='cuda:0')\n",
            "top_indices:tensor([[2, 2],\n",
            "        [0, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            " an erythropoietic protoporphyria (EPP)-associated ferrochelatase (FECH) mutation  Erythropoietic protoporphyria (EPP) is a rare inherited disorder of the heme biosynthesis pathway resulting in the accumulation of protoporphyrins in the blood, erythrocytes, and other tissues. Because of a gene mutation in the FECH gene, ferrochelatase, the enzyme involved in the final step of heme synthesis, is deficient in these patients.   Erythropoietic protoporphyria (EPP) is the most common inherited porphyria in children and is diagnosed in most individuals after the onset of cutaneous manifestations.\n",
            "Logits:[[ 0.54933554 -0.34330678]\n",
            " [ 0.54906857 -0.34238765]\n",
            " [ 0.54936975 -0.32710463]]\n",
            "Softmax:tensor([[0.6379, 0.4137],\n",
            "        [0.6390, 0.4144],\n",
            "        [0.6418, 0.4145],\n",
            "        [0.6418, 0.4145],\n",
            "        [0.6379, 0.4137]], device='cuda:0')\n",
            "top_indices:tensor([[2, 2],\n",
            "        [3, 3],\n",
            "        [1, 1],\n",
            "        [4, 4],\n",
            "        [0, 0]], device='cuda:0')\n",
            "There were no significant differences in survival in a non-small cell lung cancer prinomastat study, and in a small cell lung cancer marimastat trial. There were no significant differences in survival in a non-small cell lung cancer prinomastat study , and in a small cell lung cancer marimastat trial.  CONCLUSION: Treatment with marimastat after induction therapy for SCLC did not result in improved survival and had a negative impact on quality of life. The phase III trial in small cell lung cancer was discontinued when the results of study 140 were released in February 2001 showing that marimastat was not significantly more effective than placebo in prolonging the survival of small cell lung cancer patients. The phase III trial in small cell lung cancer was discontinued when the results of study 140 were released in February 2001 showing that marimastat was not significantly more effective than placebo in prolonging the survival of small cell lung cancer patients.\n",
            "Logits:[[ 0.56608796 -0.34860918]\n",
            " [ 0.57086515 -0.34562892]\n",
            " [ 0.58325505 -0.34531298]\n",
            " [ 0.58325505 -0.34531298]\n",
            " [ 0.56608796 -0.34860918]]\n",
            "Softmax:tensor([[0.6330, 0.4177],\n",
            "        [0.6370, 0.4193],\n",
            "        [0.6401, 0.4184],\n",
            "        [0.6255, 0.4297],\n",
            "        [0.6416, 0.4171],\n",
            "        [0.6330, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[4, 3],\n",
            "        [2, 1],\n",
            "        [1, 2],\n",
            "        [5, 0],\n",
            "        [0, 5]], device='cuda:0')\n",
            "No transmission distortion was observed for TNFR2-196R allele.   In conclusion, the TNFR2 196R allele was found to be significantly associated with the susceptibility to SLE in the Japanese population. Further population and functional studies will be of particular importance to establish TNFR2 as one of the susceptibility genes to SLE. We therefore characterized the frequency of a genetic polymorphism in the 3' untranslated region of the TNFR2 gene in Caucasoid SLE patients and geographically matched controls. Lack of association between the Met196Arg polymorphism in the TNFR2 gene and autoimmune diseases accompanied by vasculitis including SLE in Japanese. A TNFR2 3' flanking region polymorphism in systemic lupus erythematosus.\n",
            "Logits:[[ 0.54497486 -0.33205166]\n",
            " [ 0.5625369  -0.32568762]\n",
            " [ 0.57562697 -0.32920802]\n",
            " [ 0.5129969  -0.2829932 ]\n",
            " [ 0.58210945 -0.3347167 ]\n",
            " [ 0.54510874 -0.33297473]]\n",
            "Softmax:tensor([[0.6343, 0.4181],\n",
            "        [0.6382, 0.4175],\n",
            "        [0.6362, 0.4187],\n",
            "        [0.6362, 0.4187],\n",
            "        [0.6362, 0.4187],\n",
            "        [0.6372, 0.4185],\n",
            "        [0.6374, 0.4180]], device='cuda:0')\n",
            "top_indices:tensor([[1, 4],\n",
            "        [6, 3],\n",
            "        [5, 2],\n",
            "        [3, 5],\n",
            "        [2, 0]], device='cuda:0')\n",
            "The possibility of integrating viral vectors to become a persistent part of the host genome makes them a crucial element of clinical gene therapy. However, viral integration has associated risks, such as the unintentional activation of oncogenes that can result in cancer. Therefore, the analysis of integration sites of retroviral vectors is a crucial step in developing safer vectors for therapeutic use.RESULTS: Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. VISMapper can be found at: http://vismapper.babelomics.org .CONCLUSIONS: Because it uses novel mapping algorithms VISMapper is remarkably faster than previous available programs. It also provides a useful graphical interface to analyze the integration sites found in the genomic context. Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS: Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS\n",
            "\n",
            "Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites.\n",
            "Logits:[[ 0.550612   -0.3306619 ]\n",
            " [ 0.5675574  -0.333192  ]\n",
            " [ 0.55906314 -0.32818472]\n",
            " [ 0.55906314 -0.32818472]\n",
            " [ 0.55906314 -0.32818472]\n",
            " [ 0.563059   -0.3287303 ]\n",
            " [ 0.5639868  -0.33082268]]\n",
            "Softmax:tensor([[0.6534, 0.4044],\n",
            "        [0.6454, 0.4145]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            "Tauopathies are a group of neurodegenerative disorders with accumulation of three-repeat (3R) or four-repeat (4R) Tau CTE is defined as a tauopathy with a distribution of tau-positive neurofibrillary tangles (NFTs) that is distinct from other tauopathies,\n",
            "Logits:[[ 0.6340682  -0.38706043]\n",
            " [ 0.59868276 -0.3452644 ]]\n",
            "Softmax:tensor([[0.6321, 0.4148],\n",
            "        [0.6328, 0.4149],\n",
            "        [0.6356, 0.4163],\n",
            "        [0.6337, 0.4139],\n",
            "        [0.6360, 0.4136],\n",
            "        [0.6337, 0.4139],\n",
            "        [0.6341, 0.4138],\n",
            "        [0.6335, 0.4131],\n",
            "        [0.6345, 0.4143],\n",
            "        [0.6360, 0.4138],\n",
            "        [0.6321, 0.4148],\n",
            "        [0.6356, 0.4163]], device='cuda:0')\n",
            "top_indices:tensor([[ 9,  2],\n",
            "        [ 4, 11],\n",
            "        [11,  1],\n",
            "        [ 2, 10],\n",
            "        [ 8,  0]], device='cuda:0')\n",
            "Two phase III trials of the anti-IGF-1R monoclonal antibody , figitumumab ( CP-751,871) , were discontinued in 2010 as it was considered unlikely either trial would meet their primary endpoints .  Two phase III trials of the anti-IGF-1R monoclonal antibody, figitumumab (CP-751,871), were discontinued in 2010 as it was considered unlikely either trial would meet their primary endpoints. In light of disappointing clinical data with figitumumab and other targeted agents, it is likely that the use of molecular markers will become important in predicting response to treatment.  Median OS was 8.6 months for figitumumab plus chemotherapy and 9.8 months for chemotherapy alone (hazard ratio [HR], 1.18; 95% CI, 0.99 to 1.40; P = .06); median progression-free survival was 4.7 months (95% CI, 4.2 to 5.4) and 4.6 months (95% CI, 4.2 to 5.4), respectively (HR, 1.10; P = .27); the objective response rates were 33% and 35%, respectively. Median OS was 8.6 months for figitumumab plus chemotherapy and 9.8 months for chemotherapy alone (hazard ratio [HR], 1.18; 95% CI, 0.99 to 1.40; P = .06); median progression-free survival was 4.7 months (95% CI, 4.2 to 5.4) and 4.6 months (95% CI, 4.2 to 5.4), respectively (HR, 1.10; P = .27); the objective response rates were 33% and 35%, respectively.  The insulin-like growth factor receptor ( IGF-1R ) monoclonal antibody figitumumab , while initially promising , appears to increase toxicity and death in combination with chemotherapy in the treatment of patients with NSCLC of squamous histology; therefore , clinical development of this class of agents will need to proceed with caution . \n",
            "Logits:[[ 0.5413739  -0.34405294]\n",
            " [ 0.5440945  -0.3436883 ]\n",
            " [ 0.5563077  -0.33794507]\n",
            " [ 0.5481164  -0.34804666]\n",
            " [ 0.5579961  -0.34899187]\n",
            " [ 0.5481164  -0.34804666]\n",
            " [ 0.54969496 -0.34827852]\n",
            " [ 0.54732865 -0.35099444]\n",
            " [ 0.55165637 -0.34608874]\n",
            " [ 0.55825216 -0.3483766 ]\n",
            " [ 0.5413739  -0.34405294]\n",
            " [ 0.5563077  -0.33794507]]\n",
            "Softmax:tensor([[0.6319, 0.4192],\n",
            "        [0.6339, 0.4177],\n",
            "        [0.6394, 0.4146],\n",
            "        [0.6361, 0.4184],\n",
            "        [0.6350, 0.4178],\n",
            "        [0.6324, 0.4186],\n",
            "        [0.6348, 0.4185]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [3, 5],\n",
            "        [4, 6],\n",
            "        [6, 3],\n",
            "        [1, 4]], device='cuda:0')\n",
            "The Library of Integrated Cellular Signatures (LINCS) project provides comprehensive transcriptome profiling of human cell lines before and after chemical and genetic perturbations.  Recently, resources such as the Library of Integrated Network-Based Cellular Signatures (LINCS) L1000 database provide gene expression profiles induced by various chemical and genetic perturbations The library of integrated network-based cellular signatures (LINCS) L1000 data set currently comprises of over a million gene expression profiles of chemically perturbed human cell lines. The GE data is from the Library of Integrated Network-based Cellular Signatures (LINCS) L1000 dataset that measured changes in GE before and after treatment of human cells with over 20 000 small-molecule compounds including most of the FDA-approved drugs. The Library of Integrated Network-based Cellular Signatures (LINCS) L1000 big data provide gene expression profiles induced by over 10 000 compounds, shRNAs, and kinase inhibitors using the L1000 platform.\n",
            "Logits:[[ 0.5405842  -0.32607397]\n",
            " [ 0.5489447  -0.33210433]\n",
            " [ 0.57281595 -0.34502167]\n",
            " [ 0.558487   -0.32935062]\n",
            " [ 0.5536759  -0.33176082]\n",
            " [ 0.5423465  -0.32843226]\n",
            " [ 0.55298877 -0.32891518]]\n",
            "Softmax:tensor([[0.6383, 0.4177],\n",
            "        [0.6367, 0.4177],\n",
            "        [0.6358, 0.4156],\n",
            "        [0.6300, 0.4230],\n",
            "        [0.6380, 0.4134],\n",
            "        [0.6358, 0.4182],\n",
            "        [0.6399, 0.4169],\n",
            "        [0.6308, 0.4226],\n",
            "        [0.6356, 0.4164],\n",
            "        [0.6328, 0.4183],\n",
            "        [0.6358, 0.4182],\n",
            "        [0.6300, 0.4230],\n",
            "        [0.6311, 0.4231]], device='cuda:0')\n",
            "top_indices:tensor([[ 6, 12],\n",
            "        [ 0, 11],\n",
            "        [ 4,  3],\n",
            "        [ 1,  7],\n",
            "        [ 5,  9]], device='cuda:0')\n",
            "The cytokine storm is an aggressive immune response characterized by the recruitment of inflammatory leukocytes and exaggerated levels of cytokines and chemokines at the site of infection .  Endotoxins and exotoxins are among the most potent bacterial inducers of cytokines. During infectious processes, the production of inflammatory cytokines including tumor necrosis factor (TNF), interleukin-1β (IL-1β), gamma interferon (IFNγ) and chemokines orchestrates the anti-infectious innate immune response. However, an overzealous production, leading up to a cytokine storm, can be deleterious and contributes to mortality consecutive to sepsis or toxic shock syndrome Ebola virus (EBOV) disease (EVD) results from an exacerbated immunological response that is highlighted by a burst in the production of inflammatory mediators known as a \"cytokine storm.\" undesirable elevation of cytokine levels, as may occur in response to a drug or a device, may lead to severe side effects such as systemic inflammatory response syndrome or cytokine storm. Cytokine storm defines a dysregulation of and an excessively exaggerated immune response most often accompanying selected viral infections and several autoimmune diseases.\n",
            "Logits:[[ 0.5677983  -0.33238125]\n",
            " [ 0.5611578  -0.3323372 ]\n",
            " [ 0.557012   -0.34099174]\n",
            " [ 0.5323932  -0.3103803 ]\n",
            " [ 0.5666792  -0.34985232]\n",
            " [ 0.55738825 -0.33023733]\n",
            " [ 0.57477444 -0.3356845 ]\n",
            " [ 0.5354597  -0.31194803]\n",
            " [ 0.5563561  -0.3374348 ]\n",
            " [ 0.5443739  -0.32957914]\n",
            " [ 0.55738825 -0.33023733]\n",
            " [ 0.5323932  -0.3103803 ]\n",
            " [ 0.5369724  -0.30999798]]\n",
            "Softmax:tensor([[0.6306, 0.4215],\n",
            "        [0.6383, 0.4140],\n",
            "        [0.6416, 0.4129],\n",
            "        [0.6459, 0.4116],\n",
            "        [0.6301, 0.4218],\n",
            "        [0.6416, 0.4129],\n",
            "        [0.6459, 0.4116]], device='cuda:0')\n",
            "top_indices:tensor([[3, 4],\n",
            "        [6, 0],\n",
            "        [5, 1],\n",
            "        [2, 5],\n",
            "        [1, 2]], device='cuda:0')\n",
            "SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. The database essentially consists of a collection of text files providing specific annotations for human single nucleotide polymorphisms (SNPs), namely whether they are predicted to abolish, create or change the affinity of one or several transcription factor (TF) binding sites. A SNP's effect on TF binding is estimated based on a position weight matrix (PWM) model for the binding specificity of the corresponding factor. These data files are regenerated at regular intervals by an automatic procedure that takes as input a reference genome, a comprehensive SNP catalogue and a collection of PWMs. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. Homepage: http://ccg.vital-it.ch/snp2tfbs/.\n",
            "Logits:[[ 0.534814   -0.3166501 ]\n",
            " [ 0.56791085 -0.34745762]\n",
            " [ 0.5825326  -0.35189655]\n",
            " [ 0.6008878  -0.35731506]\n",
            " [ 0.53252655 -0.31525752]\n",
            " [ 0.5825326  -0.35189655]\n",
            " [ 0.6008878  -0.35731506]]\n",
            "Softmax:tensor([[0.6307, 0.4206],\n",
            "        [0.6344, 0.4185],\n",
            "        [0.6346, 0.4213],\n",
            "        [0.6346, 0.4213],\n",
            "        [0.6301, 0.4210],\n",
            "        [0.6346, 0.4213],\n",
            "        [0.6351, 0.4209],\n",
            "        [0.6334, 0.4211]], device='cuda:0')\n",
            "top_indices:tensor([[6, 5],\n",
            "        [5, 3],\n",
            "        [2, 2],\n",
            "        [3, 7],\n",
            "        [1, 4]], device='cuda:0')\n",
            "To address this possible bias, phosphopeptides quantified in both enriched and non-enriched samples can be used to calculate factors that mitigate the bias.Results: We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics.Availability and implementation: The phosphonormalizer package is freely available under GPL ( > =2) license from Bioconductor (https://bioconductor.org/packages/phosphonormalizer).Contact: sohrab.saraei@utu.fi or laura.elo@utu.fi.Supplementary information: Supplementary data are available at Bioinformatics online. Results\n",
            "We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Results\n",
            "\n",
            "We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Results We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Global centering-based normalization is a commonly used normalization approach in mass spectrometry-based label-free proteomics. It scales the peptide abundances to have the same median intensities, based on an assumption that the majority of abundances remain the same across the samples. However, especially in phosphoproteomics, this assumption can introduce bias, as the samples are enriched during sample preparation which can mask the underlying biological changes. To address this possible bias, phosphopeptides quantified in both enriched and non-enriched samples can be used to calculate factors that mitigate the bias.Results: We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics.Availability and implementation: The phosphonormalizer package is freely available under GPL ( > =2) license from Bioconductor (https://bioconductor.org/packages/phosphonormalizer).\n",
            "Logits:[[ 0.535306   -0.32025525]\n",
            " [ 0.5510093  -0.32894665]\n",
            " [ 0.5520308  -0.31746468]\n",
            " [ 0.5520308  -0.31746468]\n",
            " [ 0.532476   -0.3186735 ]\n",
            " [ 0.5520308  -0.31746468]\n",
            " [ 0.55426985 -0.31891352]\n",
            " [ 0.5467972  -0.3181783 ]]\n",
            "Softmax:tensor([[0.6368, 0.4159],\n",
            "        [0.6363, 0.4177],\n",
            "        [0.6350, 0.4202],\n",
            "        [0.6360, 0.4160],\n",
            "        [0.6392, 0.4177]], device='cuda:0')\n",
            "top_indices:tensor([[4, 2],\n",
            "        [0, 1],\n",
            "        [1, 4],\n",
            "        [3, 3],\n",
            "        [2, 0]], device='cuda:0')\n",
            "So far, results, especially from oral immunotherapy studies, have shown good efficacy in achieving desensitization to peanut with a good safety profile. Currently, two forms of peanut immunotherapy, oral immunotherapy (OIT) and epicutaneous immunotherapy (EPIT), are in Phase III clinical trials and have shown promise to induce desensitization in many subjects This article presents an overview of potential treatments of food allergy, with an emphasis on various forms of immunotherapy (including oral immunotherapy, sublingual immunotherapy, epicutaneous immunotherapy, immunotherapy with modified food antigens, and immunotherapy with a recombinant peanut vaccine). Efforts have been made to develop a vaccine for peanut allergy. Recent advances in immunotherapy and vaccine development for peanut allergy.\n",
            "Logits:[[ 0.56145287 -0.33964834]\n",
            " [ 0.5595151  -0.33204865]\n",
            " [ 0.55354863 -0.32177368]\n",
            " [ 0.5581023  -0.3391424 ]\n",
            " [ 0.57177824 -0.3324037 ]]\n",
            "Softmax:tensor([[0.6361, 0.4158],\n",
            "        [0.6333, 0.4167],\n",
            "        [0.6358, 0.4152],\n",
            "        [0.6339, 0.4172],\n",
            "        [0.6318, 0.4193],\n",
            "        [0.6344, 0.4158],\n",
            "        [0.6320, 0.4181],\n",
            "        [0.6351, 0.4155],\n",
            "        [0.6343, 0.4183],\n",
            "        [0.6330, 0.4185],\n",
            "        [0.6328, 0.4179],\n",
            "        [0.6320, 0.4181],\n",
            "        [0.6339, 0.4172],\n",
            "        [0.6318, 0.4193],\n",
            "        [0.6361, 0.4172]], device='cuda:0')\n",
            "top_indices:tensor([[ 0,  4],\n",
            "        [14, 13],\n",
            "        [ 2,  9],\n",
            "        [ 7,  8],\n",
            "        [ 5,  6]], device='cuda:0')\n",
            "Feline infectious peritonitis (FIP) is a common and highly lethal coronavirus disease of domestic cats Feline coronaviruses (FCoV) exist as 2 biotypes: feline enteric coronavirus (FECV) and feline infectious peritonitis virus (FIPV). Feline coronavirus (FCoV) infection is very common in cats, usually causing only mild intestinal signs such as diarrhoea. Up to 10% of FCoV infections, however, result in the fatal disease feline infectious peritonitis (FIP).  Feline infectious peritonitis (FIP) is one of the most important infectious diseases in cats and is caused by feline coronavirus (FCoV). Feline infectious peritonitis (FIP) is a fatal disease caused by feline coronavirus (FCoV) infection.\n",
            "Logits:[[ 0.55849427 -0.3401734 ]\n",
            " [ 0.5461983  -0.33643967]\n",
            " [ 0.55702287 -0.34237313]\n",
            " [ 0.5491012  -0.33436885]\n",
            " [ 0.54000264 -0.3256946 ]\n",
            " [ 0.55107147 -0.339942  ]\n",
            " [ 0.54077464 -0.33041802]\n",
            " [ 0.554129   -0.3413609 ]\n",
            " [ 0.5507486  -0.32990602]\n",
            " [ 0.5451378  -0.32907194]\n",
            " [ 0.5443622  -0.3315762 ]\n",
            " [ 0.5408584  -0.33053094]\n",
            " [ 0.5491012  -0.33436885]\n",
            " [ 0.54000264 -0.3256946 ]\n",
            " [ 0.55847764 -0.3344615 ]]\n",
            "Softmax:tensor([[0.6505, 0.4056],\n",
            "        [0.6372, 0.4202],\n",
            "        [0.6392, 0.4153],\n",
            "        [0.6330, 0.4200]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [2, 3],\n",
            "        [1, 2],\n",
            "        [3, 0]], device='cuda:0')\n",
            "Macrophage activation syndrome (MAS) is a life-threatening hyperinflammatory state mediated by uncontrolled cytokine storm and haemophagocytosis. Reactive oxygen species (ROS) are signaling molecules responsible for the production of cytokines and chemokines that can mediate hyperactivation of the immune response called cytokine storm.  During infectious processes, the production of inflammatory cytokines including tumor necrosis factor (TNF), interleukin-1β (IL-1β), gamma interferon (IFNγ) and chemokines orchestrates the anti-infectious innate immune response. However, an overzealous production, leading up to a cytokine storm, can be deleterious and contributes to mortality consecutive to sepsis or toxic shock syndrome.  cytokine storm leading to severe organ damage\n",
            "Logits:[[ 0.6212046  -0.3820543 ]\n",
            " [ 0.56303024 -0.32207933]\n",
            " [ 0.5720371  -0.3421028 ]\n",
            " [ 0.5451393  -0.32265475]]\n",
            "Softmax:tensor([[0.6449, 0.4132],\n",
            "        [0.6387, 0.4148],\n",
            "        [0.6387, 0.4148]], device='cuda:0')\n",
            "top_indices:tensor([[0, 2],\n",
            "        [2, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            "Introduced monoclonal antibodies (infliximab, etanercept, adaluimumab, golimumab, rituximab), tested for efficacy in other pathologies associated with the formation of granulomas, have a limited application in patients with SA.  Although treatment was well tolerated, neither ustekinumab nor golimumab demonstrated efficacy in pulmonary sarcoidosis. Although treatment was well tolerated, neither ustekinumab nor golimumab demonstrated efficacy in pulmonary sarcoidosis. \n",
            "Logits:[[ 0.59655744 -0.35073707]\n",
            " [ 0.56964976 -0.34412116]\n",
            " [ 0.56964976 -0.34412116]]\n",
            "Softmax:tensor([[0.6363, 0.4133],\n",
            "        [0.6442, 0.4093],\n",
            "        [0.6387, 0.4148],\n",
            "        [0.6373, 0.4142],\n",
            "        [0.6363, 0.4140]], device='cuda:0')\n",
            "top_indices:tensor([[1, 2],\n",
            "        [2, 3],\n",
            "        [3, 4],\n",
            "        [0, 0],\n",
            "        [4, 1]], device='cuda:0')\n",
            "The viral spike glycoprotein (S) utilizes angiotensin-converting enzyme 2 (ACE2) as a host protein receptor and mediates fusion of the viral and host membranes, making S essential to viral entry into host cells and host species tropism. Cell entry studies demonstrated that three newly identified SARSr-CoVs with different S protein sequences are all able to use human ACE2 as the receptor Angiotensin-converting enzyme 2 (ACE2), a relatively new member of the RAS, has drawn extensive attention since 2003, because of the findings that ACE2 is the receptor for SARS Corona virus and that maintenance of normal ACE2 levels in the lung is beneficial for the host to combat inflammatory lung disease. The trimeric SARS coronavirus (SARS-CoV) surface spike (S) glycoprotein consisting of three S1-S2 heterodimers binds the cellular receptor angiotensin-converting enzyme 2 (ACE2) and mediates fusion of the viral and cellular membranes through a pre- to postfusion conformation transition The infection of target cells by the SARS CoV is mediated through the interaction of the viral Spike (S) protein (1255 amino acids) and its cellular receptor, angiotensin-converting enzyme 2 (ACE2).\n",
            "Logits:[[ 0.55942017 -0.35026276]\n",
            " [ 0.59360814 -0.36681598]\n",
            " [ 0.5696951  -0.34408185]\n",
            " [ 0.5636298  -0.3467706 ]\n",
            " [ 0.5593748  -0.34731254]]\n",
            "Softmax:tensor([[0.6392, 0.4114]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "On May 9 2011, the US FDA approved Sanofi Pasteur's Fluzone(®) Intradermal influenza vaccine, the first influenza vaccine licensed in the USA that uses a new microinjection system for intradermal delivery of vaccines (Soluvia™, Becton Dickinson). \n",
            "Logits:[[ 0.5720883 -0.3580319]]\n",
            "Softmax:tensor([[0.6346, 0.4140],\n",
            "        [0.6333, 0.4160],\n",
            "        [0.6350, 0.4150],\n",
            "        [0.6350, 0.4150],\n",
            "        [0.6346, 0.4143],\n",
            "        [0.6335, 0.4155],\n",
            "        [0.6334, 0.4158],\n",
            "        [0.6334, 0.4158],\n",
            "        [0.6332, 0.4174],\n",
            "        [0.6350, 0.4150],\n",
            "        [0.6445, 0.4178]], device='cuda:0')\n",
            "top_indices:tensor([[10, 10],\n",
            "        [ 3,  8],\n",
            "        [ 9,  1],\n",
            "        [ 2,  7],\n",
            "        [ 4,  6]], device='cuda:0')\n",
            "Whilst intact in all FECVs, the 3c gene was mutated in the majority (71.4 %) of FIPVs, but not in all, implying that mutation in 3c is not the (single) cause of FIP. Feline enteric coronavirus (FECV) causes inapparent or mild enteritis in cats, but a highly fatal disease, called feline infectious peritonitis (FIP), can arise through mutation of FECV to FIP virus (FIPV). Feline infectious peritonitis virus (FIPV) was presumed to arise from mutations in the 3c of a ubiquitous and largely nonpathogenic feline enteric coronavirus (FECV). Feline infectious peritonitis virus (FIPV) was presumed to arise from mutations in the 3c of a ubiquitous and largely nonpathogenic feline enteric coronavirus (FECV). Feline infectious peritonitis (FIP) is a lethal systemic disease caused by FIP virus (FIPV), a virulent mutant of apathogenic feline enteric coronavirus (FECV).\n",
            "Logits:[[ 0.551854   -0.34749946]\n",
            " [ 0.5463052  -0.33927882]\n",
            " [ 0.55371493 -0.34325922]\n",
            " [ 0.55387473 -0.3432388 ]\n",
            " [ 0.5520207  -0.3463079 ]\n",
            " [ 0.5472966  -0.341228  ]\n",
            " [ 0.54678285 -0.34020522]\n",
            " [ 0.54678285 -0.34020522]\n",
            " [ 0.54582655 -0.33357698]\n",
            " [ 0.55371493 -0.34325922]\n",
            " [ 0.59501106 -0.3319782 ]]\n",
            "Softmax:tensor([[0.6356, 0.4163],\n",
            "        [0.6420, 0.4151],\n",
            "        [0.6331, 0.4168],\n",
            "        [0.6328, 0.4188],\n",
            "        [0.6361, 0.4194]], device='cuda:0')\n",
            "top_indices:tensor([[1, 4],\n",
            "        [4, 3],\n",
            "        [0, 2],\n",
            "        [2, 0],\n",
            "        [3, 1]], device='cuda:0')\n",
            " Its onset preceded the rearrangement of TCR genes, as Xlr expression was conserved in thymus cells from RAG1(0/0) mice.  Our results provide a novel framework for understanding ASE function and demonstrate a novel role for SATB1 as a regulator of Rag locus organization and gene expression in DP thymocytes. High level expression of the Xlr nuclear protein in immature thymocytes and colocalization with the matrix-associated region-binding SATB1 protein An anti-silencer- and SATB1-dependent chromatin hub regulates Rag1 and Rag2 gene expression during thymocyte development.  SATB1 binds to the ASE and Rag promoters, facilitating inclusion of Rag2 in the chromatin hub and the loading of RNA polymerase II to both the Rag1 and Rag2 promoters.\n",
            "Logits:[[ 0.556285   -0.3379014 ]\n",
            " [ 0.5839369  -0.3430176 ]\n",
            " [ 0.5455239  -0.3361138 ]\n",
            " [ 0.54435605 -0.3277753 ]\n",
            " [ 0.5585281  -0.32536137]]\n",
            "Softmax:tensor([[0.6318, 0.4218],\n",
            "        [0.6369, 0.4169],\n",
            "        [0.6412, 0.4160],\n",
            "        [0.6377, 0.4163],\n",
            "        [0.6365, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [3, 4],\n",
            "        [1, 1],\n",
            "        [4, 3],\n",
            "        [0, 2]], device='cuda:0')\n",
            " Like normal nascent chromatin, chromatin labeled for brief periods (0.5-1 min) in the presence of butyrate was more sensitive to digestion with DNase I and micrococcal nuclease than control bulk chromatin. Incubation of mature chromatin in butyrate for 1 h did not induce DNase I sensitivity: therefore, the presence of sodium butyrate was required during replication to preserve the increased digestibility of nascent chromatin DNA this class of nascent chromatin exhibits a shortened repeat length of approximately 165 bp, as opposed to the 288-bp repeat of bulk chromatin. Within 10 min of DNA synthesis, the spacing of mature chromatin is established; the spacing maturation can occur in the absence of protein synthesis. The second class of nascent DNA is distinguished from the nucleosomal component by its insolubility, lack of discernible nucleosomal organization, and dependence on protein synthesis to attain typical subunit structure In a previous study (Perry and Annunziato, Nucleic Acids Res. 17, 4275 [1989]) it was shown that when replication occurs in the presence of sodium butyrate (thereby inhibiting histone deacetylation), nascent chromatin fails to mature fully and instead remains preferentially sensitive to DNaseI, more soluble in magnesium, and depleted of histone H1 (relative to mature chromatin). Maturation of nucleosomal and nonnucleosomal components of nascent chromatin: differential requirements for concurrent protein synthesis.\n",
            "Logits:[[ 0.540099   -0.31554723]\n",
            " [ 0.56213975 -0.33538672]\n",
            " [ 0.5805599  -0.33904016]\n",
            " [ 0.5652474  -0.338075  ]\n",
            " [ 0.5603689  -0.3330971 ]]\n",
            "Softmax:tensor([[0.6368, 0.4154],\n",
            "        [0.6425, 0.4162],\n",
            "        [0.6345, 0.4185],\n",
            "        [0.6340, 0.4168],\n",
            "        [0.6343, 0.4227],\n",
            "        [0.6304, 0.4216]], device='cuda:0')\n",
            "top_indices:tensor([[1, 4],\n",
            "        [0, 5],\n",
            "        [2, 2],\n",
            "        [4, 3],\n",
            "        [3, 1]], device='cuda:0')\n",
            "Here, we show that promoter classes are significantly differentiated by nucleosome organization and chromatin structure. Dispersed promoters display higher associations with well-positioned nucleosomes downstream of the TSS and a more clearly defined nucleosome free region upstream, while focused promoters have a less organized nucleosome structure, yet higher presence of RNA polymerase II. Nucleosome depletion at 5'-HS4 was dependent on interaction of the insulator protein CCCTC-binding factor (CTCF) and was required for enhancer blocking.  hese differences extend to histone variants (H2A.Z) and marks (H3K4 methylation), as well as insulator binding (such as CTCF), independent of the expression levels of affected genes. This general architectural change correlates with enhanced binding of CTCF and cohesins and more pronounced insulation of contacts across chromatin boundaries in lineage-committed cells.  nucleosome occupancy at nucleosome-free regions (NFRs), many of which are located at sites occupied by the multivalent factors Ctcf and cohesin. \n",
            "Logits:[[ 0.56159985 -0.34175986]\n",
            " [ 0.5863569  -0.33820346]\n",
            " [ 0.551402   -0.32906905]\n",
            " [ 0.5493492  -0.33594912]\n",
            " [ 0.5505863  -0.3118458 ]\n",
            " [ 0.53401715 -0.31622094]]\n",
            "Softmax:tensor([[0.6315, 0.4173],\n",
            "        [0.6333, 0.4156],\n",
            "        [0.6322, 0.4155],\n",
            "        [0.6330, 0.4168],\n",
            "        [0.6320, 0.4169],\n",
            "        [0.6351, 0.4149],\n",
            "        [0.6321, 0.4185],\n",
            "        [0.6326, 0.4165],\n",
            "        [0.6341, 0.4145],\n",
            "        [0.6327, 0.4156]], device='cuda:0')\n",
            "top_indices:tensor([[5, 6],\n",
            "        [8, 0],\n",
            "        [1, 4],\n",
            "        [3, 3],\n",
            "        [9, 7]], device='cuda:0')\n",
            "The proteasome inhibitor bortezomib (also known as Velcade and PS-341) is a clinically effective antineoplastic drug that is FDA approved for treatment of hematologic malignancies such as multiple myeloma and mantle cell lymphoma. Bortezomib (Velcade™) is a reversible proteasome inhibitor that is approved for the treatment of multiple myeloma (MM). The proteasome inhibitor bortezomib is effective for a variety of tumors, but not for GBM.  The proteasome inhibitor bortezomib, registered for Multiple Myeloma treatment, is currently explored for activity in solid tumors including non-small cell lung cancer (NSCLC). The proteasome inhibitor Bortezomib is used to treat multiple myeloma (MM).\n",
            "Logits:[[ 0.5387543  -0.33381963]\n",
            " [ 0.5461986  -0.340891  ]\n",
            " [ 0.5416322  -0.34140205]\n",
            " [ 0.54495454 -0.33581135]\n",
            " [ 0.54083073 -0.3356298 ]\n",
            " [ 0.5540507  -0.34360275]\n",
            " [ 0.5411197  -0.328771  ]\n",
            " [ 0.54345423 -0.33736247]\n",
            " [ 0.54996526 -0.34524143]\n",
            " [ 0.5437052  -0.34103075]]\n",
            "Softmax:tensor([[0.6311, 0.4217],\n",
            "        [0.6293, 0.4232],\n",
            "        [0.6290, 0.4208],\n",
            "        [0.6316, 0.4179],\n",
            "        [0.6327, 0.4229],\n",
            "        [0.6337, 0.4212],\n",
            "        [0.6298, 0.4194]], device='cuda:0')\n",
            "top_indices:tensor([[5, 1],\n",
            "        [4, 4],\n",
            "        [3, 0],\n",
            "        [0, 5],\n",
            "        [6, 2]], device='cuda:0')\n",
            " We found that although the post-SET zinc finger and the KRAB domains are not essential for the methyltransferase activity of PRDM9 in cell culture, the KRAB domain mutant mice show only residual PRDM9 methyltransferase activity and undergo meiotic arrest. In aggregate, our data indicate that domains typically involved in regulation of gene expression do not serve that role in PRDM9, but are likely involved in setting the proper chromatin environment for initiation and completion of homologous recombination. In many eukaryotes, sites of meiotic recombination, also called hotspots, are regions of accessible chromatin, but in many vertebrates, their location follows a distinct pattern and is specified by PR domain-containing protein 9 (PRDM9).  PRDM9 (PR domain-containing protein 9) is a meiosis-specific protein that trimethylates H3K4 and controls the activation of recombination hot spots. It is an essential enzyme in the progression of early meiotic prophase. Our findings do not identify the nature of the underlying DNA sequences, but argue against the proposed role of Prdm9 as an essential transcription factor in mouse meiosis PRDM9 Methyltransferase Activity Is Essential for Meiotic DNA Double-Strand Break Formation at Its Binding Sites.\n",
            "Logits:[[ 0.5367672  -0.31596223]\n",
            " [ 0.5293758  -0.30959487]\n",
            " [ 0.52778405 -0.31962422]\n",
            " [ 0.5390105  -0.3315216 ]\n",
            " [ 0.54397374 -0.31069663]\n",
            " [ 0.5482429  -0.31802425]\n",
            " [ 0.5314381  -0.32528338]]\n",
            "Softmax:tensor([[0.6315, 0.4213],\n",
            "        [0.6321, 0.4199],\n",
            "        [0.6333, 0.4183]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [1, 1],\n",
            "        [0, 2]], device='cuda:0')\n",
            "The gene Fxy (also known as MID1 [7]) spans the pseudoautosomal boundary (PAB) in the laboratory mouse (Mus musculus domesticus, C57BL/6) such that the 5' three exons of the gene are located on the X chromosome but the seven exons encoding the carboxy-terminal two-thirds of the protein are located within the PAR and are therefore present on both the X and Y chromosomes The Fxy gene in mice is also located on the X chromosome but spans the pseudoautosomal boundary in this species. Here we describe a gene closely related to FXY/MID1, called FXY2, which also maps to the X chromosome within Xq22 We have previously described a gene, Fxy , that spans the pseudoautosomal boundary in mice such that the first three exons of the gene are located on the X chromosome, but the remainder of the gene is located on both X and Y chromosomes. \n",
            "Logits:[[ 0.53869337 -0.31736168]\n",
            " [ 0.541261   -0.3233676 ]\n",
            " [ 0.5466104  -0.3297053 ]]\n",
            "Softmax:tensor([[0.6409, 0.4172],\n",
            "        [0.6363, 0.4154],\n",
            "        [0.6421, 0.4159]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [0, 2],\n",
            "        [1, 1]], device='cuda:0')\n",
            "RESULTS: At the end of the study, after 547 days of follow-up, 103 patients (51.7%) in the pentoxifylline group and 120 (59.7%) in the placebo group were alive (unadjusted risk 1.28, p = 0.107; adjusted risk 1.43, p = 0.02). .RESULTS: At the end of the study, after 547 days of follow-up, 103 patients (51.7%) in the pentoxifylline group and 120 (59.7%) in the placebo group were alive (unadjusted risk 1.28, p = 0.107; adjusted risk 1.43, p = 0.02). In contrast, analysis of secondary outcome functional variables did not show the same negative effect of the drug.  CONCLUSIONS: Pentoxifylline is not beneficial in ALS and should be avoided in patients treated with riluzole. \n",
            "Logits:[[ 0.57922614 -0.33446136]\n",
            " [ 0.559293   -0.34179956]\n",
            " [ 0.5846299  -0.33979368]]\n",
            "Softmax:tensor([[0.6345, 0.4185],\n",
            "        [0.6452, 0.4169],\n",
            "        [0.6334, 0.4185],\n",
            "        [0.6337, 0.4180],\n",
            "        [0.6347, 0.4174],\n",
            "        [0.6347, 0.4174]], device='cuda:0')\n",
            "top_indices:tensor([[1, 2],\n",
            "        [5, 0],\n",
            "        [4, 3],\n",
            "        [0, 5],\n",
            "        [3, 4]], device='cuda:0')\n",
            "Mapping brain metabolic connectivity in awake rats with μPET and optogenetic stimulation. PURPOSE\n",
            "Positron emission tomography (PET) is a useful imaging modality that quantifies the physiological distributions of radiolabeled tracers in vivo in humans and animals. PURPOSE\n",
            "\n",
            "Positron emission tomography (PET) is a useful imaging modality that quantifies the physiological distributions of radiolabeled tracers in vivo in humans and animals. Positron emission tomography (PET) with [(18)F]2-fluoro-2-deoxy-D-glucose was used to measure changes in regional brain glucose metabolism Positron emission tomography (PET) allows the quantitative measurement of regional cerebral flow (rCBF) in humans in quantitative terms.\n",
            "Logits:[[ 0.5516148  -0.32890218]\n",
            " [ 0.5981774  -0.33559254]\n",
            " [ 0.54676723 -0.32881337]\n",
            " [ 0.5479634  -0.3308012 ]\n",
            " [ 0.5525375  -0.33364016]\n",
            " [ 0.5525375  -0.33364016]]\n",
            "Softmax:tensor([[0.6323, 0.4203],\n",
            "        [0.6341, 0.4202],\n",
            "        [0.6331, 0.4209],\n",
            "        [0.6324, 0.4197],\n",
            "        [0.6331, 0.4209]], device='cuda:0')\n",
            "top_indices:tensor([[1, 2],\n",
            "        [4, 4],\n",
            "        [2, 0],\n",
            "        [3, 1],\n",
            "        [0, 3]], device='cuda:0')\n",
            "A significant association was observed in the discovery cohorts for the single-nucleotide polymorphism (SNP) rs17518584 (discovery P-value=3.12 × 10(-8)) and in the joint discovery and replication meta-analysis (P-value=3.28 × 10(-9) after adjustment for age, gender and education) in an intron of the gene cell adhesion molecule 2 (CADM2) for performance on the LDST/DSST. Rs17518584 is located about 170 kb upstream of the transcription start site of the major transcript for the CADM2 gene, but is within an intron of a variant transcript that includes an alternative first exon. The variant is associated with expression of CADM2 in the cingulate cortex (P-value=4 × 10(-4)). The protein encoded by CADM2 is involved in glutamate signaling (P-value=7.22 × 10(-15)), gamma-aminobutyric acid (GABA) transport (P-value=1.36 × 10(-11)) and neuron cell-cell adhesion (P-value=1.48 × 10(-13)). Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. The CADM2 gene is associated with processing speed performance - evidence among elderly with type 2 diabetes. GWAS for executive function and processing speed suggests involvement of the CADM2 gene.\n",
            "Logits:[[ 0.5422782  -0.32174265]\n",
            " [ 0.5499867  -0.32211182]\n",
            " [ 0.5453669  -0.31898075]\n",
            " [ 0.54248375 -0.32420793]\n",
            " [ 0.5453669  -0.31898075]]\n",
            "Softmax:tensor([[0.6339, 0.4202],\n",
            "        [0.6454, 0.4095],\n",
            "        [0.6331, 0.4202]], device='cuda:0')\n",
            "top_indices:tensor([[1, 0],\n",
            "        [0, 2],\n",
            "        [2, 1]], device='cuda:0')\n",
            "The exponential reduction in cost of genome sequencing has resulted in a rapid growth of genomic data. Most of the entropy of short read data lies not in the sequence of read bases themselves but in their Quality Scores-the confidence measurement that each base has been sequenced correctly. Lossless compression methods are now close to their theoretical limits and hence there is a need for lossy methods that further reduce the complexity of these data without impacting downstream analyses.RESULTS: We here propose GeneCodeq, a Bayesian method inspired by coding theory for adjusting quality scores to improve the compressibility of quality scores without adversely impacting genotyping accuracy. Our model leverages a corpus of k-mers to reduce the entropy of the quality scores and thereby the compressibility of these data (in FASTQ or SAM/BAM/CRAM files), resulting in compression ratios that significantly exceeds those of other methods. Our approach can also be combined with existing lossy compression schemes to further reduce entropy and allows the user to specify a reference panel of expected sequence variations to improve the model accuracy. In addition to extensive empirical evaluation, we also derive novel theoretical insights that explain the empirical performance and pitfalls of corpus-based quality score compression schemes in general. Finally, we show that as a positive side effect of compression, the model can lead to improved genotyping accuracy.AVAILABILITY AND IMPLEMENTATION: GeneCodeq is available at: github.com/genecodeq/eval GeneCodeq: quality score compression and improved genotyping using a Bayesian framework. GeneCodeq: quality score compression and improved genotyping using a Bayesian framework\n",
            "Logits:[[ 0.54905933 -0.3220404 ]\n",
            " [ 0.5988616  -0.36597112]\n",
            " [ 0.54544854 -0.3221372 ]]\n",
            "Softmax:tensor([[0.6333, 0.4141],\n",
            "        [0.6289, 0.4201],\n",
            "        [0.6352, 0.4129],\n",
            "        [0.6342, 0.4157],\n",
            "        [0.6338, 0.4159]], device='cuda:0')\n",
            "top_indices:tensor([[2, 1],\n",
            "        [3, 4],\n",
            "        [4, 3],\n",
            "        [0, 0],\n",
            "        [1, 2]], device='cuda:0')\n",
            "programmed cell death protein 1 (PD-1)-blocking antibodies nivolumab or pembrolizumab  An improvement in the understanding of the role of the immune system in tumor immunosurveillance has led to the development of the programmed death-1 ( PD-1 ) immune checkpoint inhibitor nivolumab ( Opdivo) .  Nivolumab (Opdivo(®); Nivolumab BMS™) was the first programmed death (PD)-1 immune checkpoint inhibitor to be approved for use in advanced, squamous non-small cell lung cancer (NSCLC) following prior chemotherapy. Fatal Myocarditis Following Treatment with the PD-1 Inhibitor Nivolumab  PD-1 inhibitor nivolumab (Opdivo)\n",
            "Logits:[[ 0.5464323  -0.34696695]\n",
            " [ 0.52745366 -0.32255054]\n",
            " [ 0.5546662  -0.3520968 ]\n",
            " [ 0.55046225 -0.34048805]\n",
            " [ 0.5487553  -0.3396904 ]]\n",
            "Softmax:tensor([[0.6353, 0.4180],\n",
            "        [0.6366, 0.4174],\n",
            "        [0.6332, 0.4182],\n",
            "        [0.6323, 0.4179],\n",
            "        [0.6404, 0.4177],\n",
            "        [0.6335, 0.4190],\n",
            "        [0.6359, 0.4159],\n",
            "        [0.6353, 0.4161],\n",
            "        [0.6333, 0.4190],\n",
            "        [0.6350, 0.4167],\n",
            "        [0.6353, 0.4161],\n",
            "        [0.6346, 0.4171],\n",
            "        [0.6369, 0.4171]], device='cuda:0')\n",
            "top_indices:tensor([[ 4,  8],\n",
            "        [12,  5],\n",
            "        [ 1,  2],\n",
            "        [ 6,  0],\n",
            "        [ 7,  3]], device='cuda:0')\n",
            "FINDINGS: ALSFRS-R score deterioration was faster in the minocycline group than in the placebo group (-1.30 vs -1.04 units/month, 95% CI for difference -0.44 to -0.08; p=0.005). Patients on minocycline also had non-significant tendencies towards faster decline in FVC (-3.48 vs -3.01, -1.03 to 0.11; p=0.11) and MMT score (-0.30 vs -0.26, -0.08 to 0.01; p=0.11), and greater mortality during the 9-month treatment phase (hazard ratio=1.32, 95% CI 0.83 to 2.10; p=0.23) than did patients on placebo. Quality-of-life scores did not differ between the treatment groups.  Patients on minocycline also had non-significant tendencies towards faster decline in FVC (-3.48 vs -3.01, -1.03 to 0.11; p=0.11) and MMT score (-0.30 vs -0.26, -0.08 to 0.01; p=0.11), and greater mortality during the 9-month treatment phase (hazard ratio=1.32, 95% CI 0.83 to 2.10; p=0.23) than did patients on placebo. This pilot study shows that minocycline and riluzole can be taken safely together. Further trials are needed to assess efficacy of such treatment. A recent publication of the results of a clinical trial of minocycline in 412 ALS patient has aroused considerable controversy in the ALS scientific community. As on previous occasions, the results obtained in the laboratory are not reproduced in clinical practice. A recent publication of the results of a clinical trial of minocycline in 412 ALS patient has aroused considerable controversy in the ALS scientific community.\n",
            "Logits:[[ 0.5548228  -0.33083257]\n",
            " [ 0.56046975 -0.33364472]\n",
            " [ 0.54604936 -0.32999796]\n",
            " [ 0.542266   -0.33124724]\n",
            " [ 0.57695836 -0.3324129 ]\n",
            " [ 0.5471189  -0.32707742]\n",
            " [ 0.55760866 -0.33949208]\n",
            " [ 0.55499434 -0.33894005]\n",
            " [ 0.5464949  -0.32698384]\n",
            " [ 0.5535314  -0.33621523]\n",
            " [ 0.55499434 -0.33894005]\n",
            " [ 0.5522072  -0.33473262]\n",
            " [ 0.56181216 -0.33488044]]\n",
            "Softmax:tensor([[0.6418, 0.4178],\n",
            "        [0.6381, 0.4221],\n",
            "        [0.6319, 0.4202],\n",
            "        [0.6357, 0.4221],\n",
            "        [0.6395, 0.4181]], device='cuda:0')\n",
            "top_indices:tensor([[0, 3],\n",
            "        [4, 1],\n",
            "        [1, 2],\n",
            "        [3, 4],\n",
            "        [2, 0]], device='cuda:0')\n",
            "Extracellular vesicles (EVs) are membrane-bound vesicles released into the extracellular space by almost all types of cells. EVs can cross the physiological barriers, and a variety of biological fluids are enriched in them. EVs are a heterogeneous population of vesicles, including exosomes, microvesicles, and apoptotic bodies.  Extracellular vesicles are cell-derived membrane particles ranging from 30 to 5,000 nm in size, including exosomes, microvesicles, and apoptotic bodies. They are released under physiological conditions, but also upon cellular activation, senescence, and apoptosis. They play an important role in intercellular communication.  Extracellular vesicles (EVs), microparticles, exosomes, and apoptotic bodies, originated by different cell types are emerging as a novel mean of cell-to-cell communication in physiology and pathology and represent a new way to convey fundamental information between cells.   In treated cells, the apoptotic hallmarks such as formation of apoptotic bodies,  Apoptotic bodies are generated on apoptotic cell shrinkage and death.\n",
            "Logits:[[ 0.5833644  -0.33174235]\n",
            " [ 0.56694186 -0.31431434]\n",
            " [ 0.5404104  -0.32202238]\n",
            " [ 0.5567299  -0.31411085]\n",
            " [ 0.57297975 -0.33071902]]\n",
            "Softmax:tensor([[0.6354, 0.4144]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "Despite discontinuation of the first inhalable insulin, Exubera®, due to suboptimal market acceptance, development of orally inhaled insulin delivery systems has been galvanized by the recent approval of Afrezza® and several others awaiting approval.\n",
            "Logits:[[ 0.55535656 -0.34590742]]\n",
            "Softmax:tensor([[0.6310, 0.4182],\n",
            "        [0.6351, 0.4177],\n",
            "        [0.6313, 0.4217],\n",
            "        [0.6320, 0.4220],\n",
            "        [0.6348, 0.4192]], device='cuda:0')\n",
            "top_indices:tensor([[1, 3],\n",
            "        [4, 2],\n",
            "        [3, 4],\n",
            "        [2, 0],\n",
            "        [0, 1]], device='cuda:0')\n",
            "In vitro, the three known yeast nucleosome spacing enzymes (CHD1, ISW1 and ISW2) form arrays with different spacing. Isw1b acts in conjunction with Chd1 to regulate chromatin structure by preventing trans-histone exchange from taking place over coding regions. In this way, Isw1b and Chd1 are important in maintaining chromatin integrity during transcription elongation by RNAPII. We propose that CHD1 directs short spacing, resulting in eviction of H1 and chromatin unfolding, whereas ISW1 directs longer spacing, allowing H1 to bind and condense the chromatin CHD1 and ISW1 compete to set the spacing on most genes, such that CHD1 dominates genes with shorter spacing and ISW1 dominates genes with longer spacing The ISW1 and CHD1 ATP-dependent chromatin remodelers compete to set nucleosome spacing in vivo\n",
            "Logits:[[ 0.5364266  -0.33032012]\n",
            " [ 0.5540094  -0.33208072]\n",
            " [ 0.53787875 -0.315606  ]\n",
            " [ 0.5407589  -0.31474173]\n",
            " [ 0.5530132  -0.32590386]]\n",
            "Softmax:tensor([[0.6371, 0.4159],\n",
            "        [0.6365, 0.4160],\n",
            "        [0.6384, 0.4156],\n",
            "        [0.6365, 0.4161],\n",
            "        [0.6376, 0.4135],\n",
            "        [0.6364, 0.4144],\n",
            "        [0.6355, 0.4186],\n",
            "        [0.6399, 0.4145],\n",
            "        [0.6404, 0.4136],\n",
            "        [0.6388, 0.4137],\n",
            "        [0.6339, 0.4168],\n",
            "        [0.6336, 0.4182],\n",
            "        [0.6439, 0.4118]], device='cuda:0')\n",
            "top_indices:tensor([[12,  6],\n",
            "        [ 8, 11],\n",
            "        [ 7, 10],\n",
            "        [ 9,  3],\n",
            "        [ 2,  1]], device='cuda:0')\n",
            "Gamma-ray irradiation introduces single and/or double strand breaks into the DNA molecule of the cells. RESULTS BRCA2-defective cells were unable to repair the double-strand DNA breaks induced by ionizing radiation. Ionizing radiation and radiomimetic drugs such as bleomycin, calichieamycin, neocarzinostatin chromophore, and other synthetic agents can produce both single and double strand breaks in DNA. BACKGROUND Induction of DNA double strand breaks and alterations in the repair of these breaks is implicated in breast carcinogenesis. The induction of DNA interstrand cross-links by ionizing radiation has been largely ignored in favour of studies on double-strand break formation and repair.\n",
            "Logits:[[ 0.5626608  -0.33972925]\n",
            " [ 0.56004584 -0.33921704]\n",
            " [ 0.56834483 -0.34068665]\n",
            " [ 0.56010944 -0.33867943]\n",
            " [ 0.56503844 -0.34941718]\n",
            " [ 0.55961925 -0.34560055]\n",
            " [ 0.55582213 -0.3285474 ]\n",
            " [ 0.5748941  -0.3453083 ]\n",
            " [ 0.5769584  -0.34920213]\n",
            " [ 0.5703623  -0.34861356]\n",
            " [ 0.54910153 -0.3358483 ]\n",
            " [ 0.5477576  -0.32998157]\n",
            " [ 0.59249604 -0.35653603]]\n",
            "Softmax:tensor([[0.6364, 0.4127],\n",
            "        [0.6372, 0.4118],\n",
            "        [0.6480, 0.4053],\n",
            "        [0.6343, 0.4167],\n",
            "        [0.6349, 0.4104],\n",
            "        [0.6352, 0.4129],\n",
            "        [0.6348, 0.4166]], device='cuda:0')\n",
            "top_indices:tensor([[2, 3],\n",
            "        [1, 6],\n",
            "        [0, 5],\n",
            "        [5, 0],\n",
            "        [4, 1]], device='cuda:0')\n",
            " This study provides strong evidence of positive associations between protracted low-dose radiation exposure and leukaemia.FU e highest risk of radiation- induced breast cancer is evidenced in the sub-population of female patients who have undergone radiotherapy for either malignant or non-malignant diseases Because latency period for different nosological forms of radiation-induced malignant tumors varies widely, profound attention in further studies should be drawn not only to thyroid, breast cancers and leukemia, but also to malignancies with longer latent period: lung, stomach, colon, ovary, urinary bladder, kidney cancer and multiple myeloma. Breast cancer ( BC ) is the most common type of malignancy in female patients and radio-treatment is the conventional therapy even if a great number of studies reported that enhanced sensitivity to ionizing radiation as measured as chromosome effects is present in a significant proportion of cancer patients , including breast cancer ones .  Ionizing radiation exposure is a risk factor for development of esophageal squamous cell carcinoma , a histological subtype of esophageal cancer that is highly aggressive and is associated with poor patient prognosis . \n",
            "Logits:[[ 0.559967   -0.35280707]\n",
            " [ 0.5632906  -0.35665306]\n",
            " [ 0.61035365 -0.38328815]\n",
            " [ 0.55087936 -0.33615702]\n",
            " [ 0.5533621  -0.36218902]\n",
            " [ 0.55464095 -0.35209876]\n",
            " [ 0.5530201  -0.33679965]]\n",
            "Softmax:tensor([[0.6335, 0.4163],\n",
            "        [0.6349, 0.4203],\n",
            "        [0.6368, 0.4175],\n",
            "        [0.6349, 0.4203],\n",
            "        [0.6368, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[2, 1],\n",
            "        [4, 3],\n",
            "        [3, 4],\n",
            "        [1, 2],\n",
            "        [0, 0]], device='cuda:0')\n",
            "KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies. Acute myeloid leukemia (AML) is an aggressive cancer with a poor prognosis, for which mainstream treatments have not changed for decades. To identify additional therapeutic targets in AML, we optimize a genome-wide clustered regularly interspaced short palindromic repeats (CRISPR) screening platform and use it to identify genetic vulnerabilities in AML cells. We identify 492 AML-specific cell-essential genes, including several established therapeutic targets such as DOT1L, BCL2, and MEN1, and many other genes including clinically actionable candidates. We validate selected genes using genetic and pharmacological inhibition, and chose KAT2A as a candidate for downstream study. KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies.\n",
            "Logits:[[ 0.54711854 -0.33810478]\n",
            " [ 0.5533539  -0.32152572]\n",
            " [ 0.5616139  -0.33288684]\n",
            " [ 0.5533539  -0.32152572]\n",
            " [ 0.5616139  -0.33288684]]\n",
            "Softmax:tensor([[0.6331, 0.4173],\n",
            "        [0.6337, 0.4171],\n",
            "        [0.6353, 0.4151],\n",
            "        [0.6329, 0.4187],\n",
            "        [0.6333, 0.4185],\n",
            "        [0.6372, 0.4143],\n",
            "        [0.6393, 0.4150],\n",
            "        [0.6377, 0.4159],\n",
            "        [0.6375, 0.4157],\n",
            "        [0.6372, 0.4142],\n",
            "        [0.6361, 0.4163],\n",
            "        [0.6333, 0.4185],\n",
            "        [0.6344, 0.4161],\n",
            "        [0.6338, 0.4162]], device='cuda:0')\n",
            "top_indices:tensor([[ 6,  3],\n",
            "        [ 7, 11],\n",
            "        [ 8,  4],\n",
            "        [ 5,  0],\n",
            "        [ 9,  1]], device='cuda:0')\n",
            "BACKGROUND\n",
            "\n",
            "Idiopathic toe walking (ITW) is an exclusionary diagnosis given to healthy children who persist in walking on their toes after they should typically have achieved a heel-toe gait. BACKGROUND\n",
            "\n",
            "Idiopathic toe walking is a diagnosis of exclusion characterized by a persistent toe-toe gait pattern after three years of age. Idiopathic toe walking is a diagnosis of exclusion characterized by a persistent toe-toe gait pattern after three years of age.  Idiopathic toe walking (ITW) is a pathological gait pattern in which children walk on their tip toes with no orthopedic or neurological reason.  The diagnosis of idiopathic toe walking is a diagnosis of exclusion used for children with persistent toe walking and no associated medical condition . \n",
            "Logits:[[ 0.54566914 -0.33374998]\n",
            " [ 0.54815084 -0.3348493 ]\n",
            " [ 0.5548866  -0.34299153]\n",
            " [ 0.54484385 -0.32811943]\n",
            " [ 0.54632384 -0.32907408]\n",
            " [ 0.5632093  -0.34641087]\n",
            " [ 0.57240164 -0.34323022]\n",
            " [ 0.5654792  -0.33980903]\n",
            " [ 0.56444913 -0.34042072]\n",
            " [ 0.56315976 -0.34649354]\n",
            " [ 0.5584258  -0.33788934]\n",
            " [ 0.54632384 -0.32907408]\n",
            " [ 0.5510013  -0.33886757]\n",
            " [ 0.548661   -0.33842286]]\n",
            "Softmax:tensor([[0.6357, 0.4197],\n",
            "        [0.6379, 0.4144],\n",
            "        [0.6411, 0.4180],\n",
            "        [0.6376, 0.4176],\n",
            "        [0.6382, 0.4178],\n",
            "        [0.6396, 0.4167],\n",
            "        [0.6518, 0.4101],\n",
            "        [0.6380, 0.4153],\n",
            "        [0.6337, 0.4212],\n",
            "        [0.6379, 0.4144],\n",
            "        [0.6337, 0.4212],\n",
            "        [0.6372, 0.4143],\n",
            "        [0.6379, 0.4144],\n",
            "        [0.6402, 0.4193],\n",
            "        [0.6337, 0.4212]], device='cuda:0')\n",
            "top_indices:tensor([[ 6, 14],\n",
            "        [ 2, 10],\n",
            "        [13,  8],\n",
            "        [ 5,  0],\n",
            "        [ 4, 13]], device='cuda:0')\n",
            "3'AmNic-rEPA recipients with the highest serum antinicotine Ab response (top 30% by area under the curve (AUC)) were significantly more likely than the placebo recipients (24.6% vs. 12.0%, P = 0.024, odds ratio (OR) = 2.69, 95% confidence interval (CI), 1.14-6.37) to attain 8 weeks of continuous abstinence from weeks 19 through 26.  FINDINGS: There was no difference in abstinence rates between NicVAX and placebo from weeks 9 to 52 [27.7 versus 30.0%, odds ratio (OR) = 0.89, 95% confidence interval (CI) = 0.62-1.29] or weeks 37 to 52 (33.8 versus 33.2%, OR = 1.03, 95% CI = 0.73-1.46). The top 30% antibody responders, compared to the placebo group, showed a non-significant tendency towards higher abstinence rates from weeks 37 to 52 (42.2 versus 33.2%, OR = 1.47, 95% CI = 0.89-2.42) FINDINGS\n",
            "There was no difference in abstinence rates between NicVAX and placebo from weeks 9 to 52 [27.7 versus 30.0%, odds ratio (OR) = 0.89, 95% confidence interval (CI) = 0.62-1.29] or weeks 37 to 52 (33.8 versus 33.2%, OR = 1.03, 95% CI = 0.73-1.46). AUTHORS' CONCLUSIONS: There is currently no evidence that nicotine vaccines enhance long-term smoking cessation. The RR for 12 month cessation in active and placebo groups was 1.35 (95% Confidence Interval (CI) 0.82 to 2.22) in the trial of NIC002 and 1.74 (95% CI 0.73 to 4.18) in one NicVAX trial. Two Phase III NicVAX trials, for which full results were not available, reported similar quit rates of approximately 11% in both groups.\n",
            "Logits:[[ 0.5568899  -0.32404435]\n",
            " [ 0.56630725 -0.3457498 ]\n",
            " [ 0.5802987  -0.33094156]\n",
            " [ 0.5651519  -0.3327692 ]\n",
            " [ 0.567666   -0.33184165]\n",
            " [ 0.5736281  -0.33648345]\n",
            " [ 0.62675846 -0.363546  ]\n",
            " [ 0.5669091  -0.34230283]\n",
            " [ 0.5482912  -0.3179558 ]\n",
            " [ 0.56630725 -0.3457498 ]\n",
            " [ 0.5482912  -0.3179558 ]\n",
            " [ 0.563038   -0.34640273]\n",
            " [ 0.56630725 -0.3457498 ]\n",
            " [ 0.57619923 -0.32571554]\n",
            " [ 0.5482912  -0.3179558 ]]\n",
            "Softmax:tensor([[0.6345, 0.4178],\n",
            "        [0.6359, 0.4167],\n",
            "        [0.6338, 0.4182],\n",
            "        [0.6373, 0.4123],\n",
            "        [0.6338, 0.4182],\n",
            "        [0.6359, 0.4167],\n",
            "        [0.6340, 0.4162]], device='cuda:0')\n",
            "top_indices:tensor([[3, 2],\n",
            "        [5, 4],\n",
            "        [1, 0],\n",
            "        [0, 5],\n",
            "        [6, 1]], device='cuda:0')\n",
            "Natalizumab did not achieve a statistically significant primary composite disability outcome in a trial of 887 patients with secondary progressive MS , but it did demonstrate a benefit on a prespecified component of the 9-Hole Peg Test .  In this review, we summarize the pathophysiological mechanisms involved in the development of SPMS and the rationale and clinical potential for natalizumab, which is currently approved for the treatment of relapsing forms of MS, to exert beneficial effects in reducing disease progression unrelated to relapses in SPMS. In this review, we summarize the pathophysiological mechanisms involved in the development of SPMS and the rationale and clinical potential for natalizumab, which is currently approved for the treatment of relapsing forms of MS, to exert beneficial effects in reducing disease progression unrelated to relapses in SPMS.  INTERPRETATION: Natalizumab treatment for secondary progressive multiple sclerosis did not reduce progression on the primary multicomponent disability endpoint in part 1, but it did reduce progression on its upper-limb component. Natalizumab treatment for secondary progressive multiple sclerosis did not reduce progression on the primary multicomponent disability endpoint in part 1, but it did reduce progression on its upper-limb component.\n",
            "Logits:[[ 0.5516818  -0.3317218 ]\n",
            " [ 0.5574371  -0.33645087]\n",
            " [ 0.5484092  -0.3303643 ]\n",
            " [ 0.56383437 -0.3545433 ]\n",
            " [ 0.5484092  -0.3303643 ]\n",
            " [ 0.5574371  -0.33645087]\n",
            " [ 0.5492612  -0.3383925 ]]\n",
            "Softmax:tensor([[0.6385, 0.4121],\n",
            "        [0.6343, 0.4178],\n",
            "        [0.6489, 0.4129],\n",
            "        [0.6392, 0.4111],\n",
            "        [0.6398, 0.4160],\n",
            "        [0.6381, 0.4183],\n",
            "        [0.6323, 0.4153]], device='cuda:0')\n",
            "top_indices:tensor([[2, 5],\n",
            "        [4, 1],\n",
            "        [3, 4],\n",
            "        [0, 6],\n",
            "        [5, 2]], device='cuda:0')\n",
            "his hypothesis may add a new dimension to the understanding of MS etiology and help to design novel therapeutic agents against potential targets that might be discovered. If this hypothesis proves to be true, tau phosphorylation inhibitors can be potential candidates for MS drug development. The amyloid-forming proteins tau, αB crystallin, and amyloid P protein are all found in lesions of multiple sclerosis (MS). Alzheimer's disease (AD) is the most common cause of progressive dementia in the elderly. It is characterized by a progressive and irreversible loss of cognitive abilities and formation of senile plaques, composed mainly of amyloid β (Aβ), and neurofibrillary tangles (NFTs), composed of tau protein, in the hippocampus and cortex of afflicted human Abnormally hyperphosphorylated tau is the major protein constituent of neurofibrillary tangles (NFTs) in the brain of Alzheimer disease (AD) patients Alzheimer's disease (AD), progressive supranuclear palsy (PSP), frontotemporal dementia and parkinsonism linked to chromosome 17 (FTDP-17), and Pick's disease (PiD) are commonly known as tauopathies.\n",
            "Logits:[[ 0.5688906  -0.35512114]\n",
            " [ 0.55085707 -0.33187   ]\n",
            " [ 0.6140674  -0.3519751 ]\n",
            " [ 0.57176614 -0.3592749 ]\n",
            " [ 0.5743427  -0.3393356 ]\n",
            " [ 0.5672761  -0.3295833 ]\n",
            " [ 0.54190946 -0.34196535]]\n",
            "Softmax:tensor([[0.6398, 0.4159]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "Our goal was to compare the safety and immunogenicity of a combination vaccine (DTaP(5)-IPV-Hib; Pentacel) with that of its separately administered, US-licensed equivalent vaccines (diphtheria, tetanus, 5-component acellular pertussis vaccine [DTaP(5); Daptacel], inactivated poliovirus vaccine [IPV; IPOL], and Haemophilus influenzae type b [Hib] vaccine [ActHIB]), when administered to infants and toddlers concomitantly with other routinely recommended vaccines and to assess antibody persistence from the fourth dose in toddlers to the fifth (preschool) DTaP(5) dose.\n",
            "Logits:[[ 0.57453394 -0.33981362]]\n",
            "Softmax:tensor([[0.6373, 0.4183],\n",
            "        [0.6409, 0.4207],\n",
            "        [0.6371, 0.4192],\n",
            "        [0.6356, 0.4203],\n",
            "        [0.6383, 0.4216],\n",
            "        [0.6371, 0.4192],\n",
            "        [0.6409, 0.4207],\n",
            "        [0.6415, 0.4210]], device='cuda:0')\n",
            "top_indices:tensor([[7, 4],\n",
            "        [6, 7],\n",
            "        [1, 6],\n",
            "        [4, 1],\n",
            "        [0, 3]], device='cuda:0')\n",
            "Combining in vivo and in vitro findings, we identified nine AhR agonists, six of which are marketed therapeutics and have been approved by the U.S. Food and Drug Administration, including leflunomide, flutamide, and nimodipine. However, recent discoveries of new AhR ligands with potential therapeutic applications have been reported, inviting reconsideration of this policy. However, recent discoveries of new AhR ligands with potential therapeutic applications have been reported, inviting reconsideration of this policy. However , recent discoveries of new AhR ligands with potential therapeutic applications have been reported , inviting reconsideration of this policy  Based on our review of the data, there is little evidence to support the indiscriminate exclusion of AhR activators/Cyp1a1 inducers from early drug developmental pipelines. We also found no evidence that short-term treatment with RMAhRLs produce \"dioxin-like toxicity\"\n",
            "Logits:[[ 0.5636848  -0.32980475]\n",
            " [ 0.57944673 -0.32000256]\n",
            " [ 0.56294763 -0.32592642]\n",
            " [ 0.5565281  -0.32145244]\n",
            " [ 0.5679059  -0.316248  ]\n",
            " [ 0.56294763 -0.32592642]\n",
            " [ 0.57944673 -0.32000256]\n",
            " [ 0.5819823  -0.318691  ]]\n",
            "Softmax:tensor([[0.6321, 0.4206],\n",
            "        [0.6330, 0.4181],\n",
            "        [0.6350, 0.4184],\n",
            "        [0.6351, 0.4177],\n",
            "        [0.6294, 0.4232],\n",
            "        [0.6336, 0.4193]], device='cuda:0')\n",
            "top_indices:tensor([[3, 4],\n",
            "        [2, 0],\n",
            "        [5, 5],\n",
            "        [1, 2],\n",
            "        [0, 1]], device='cuda:0')\n",
            "Emerging lines of evidence indicate that histone variants (H2AX and H2A.Z), histone post-translational modifications (acetylation, phosphorylation, methylation and ubiquitination) and chromatin-remodeling complexes (INO80, SWR1, SWI/SNF, RSC and NuRD) are important and direct players in the DNA double-strand break (DSB) response as well. In contrast, H2A.Z deposition is dispensable for nucleosome positioning.  Accordingly, the absence of SWR-C or histone H2A.Z results in compromised chromatin remodeling and impaired gene expression in the absence of RSC and H3K4 methylation. In RSC-depleted cells, NFRs shrink such that the average positions of flanking nucleosomes move toward predicted sites. In contrast, the upstream nucleosome which covers the TATA box under repressed conditions is shifted approximately 50 bp further upstream by the ATP-dependent chromatin remodeler RSC upon activation. It is marked with the histone variant H2A.Z and H4K16 acetylation in active state\n",
            "Logits:[[ 0.5412975  -0.32040367]\n",
            " [ 0.54514617 -0.33065188]\n",
            " [ 0.5539287  -0.32937217]\n",
            " [ 0.55411315 -0.33206585]\n",
            " [ 0.5295525  -0.30956355]\n",
            " [ 0.54758173 -0.32567772]]\n",
            "Softmax:tensor([[0.6599, 0.3550],\n",
            "        [0.6475, 0.4080],\n",
            "        [0.6583, 0.4083],\n",
            "        [0.6406, 0.4160],\n",
            "        [0.6356, 0.4223],\n",
            "        [0.6524, 0.4007],\n",
            "        [0.6597, 0.3883],\n",
            "        [0.6509, 0.4121]], device='cuda:0')\n",
            "top_indices:tensor([[0, 4],\n",
            "        [6, 3],\n",
            "        [2, 7],\n",
            "        [5, 2],\n",
            "        [7, 1]], device='cuda:0')\n",
            "Amifostine is the first FDA approved cytoprotective and chemoprotective agent in the treatment of cancer.   Tea polyphenols (TPs) have been shown to reduce radiation-induced damage in multiple studies, CAPE was found to act both as radioprotector and radiosensitizer. Metformin as a radiation modifier; implications to normal tissue protection and tumor sensitization. Recent studies have shown that alpha-2-macroglobulin (α2M) possesses radioprotective effects. \n",
            "Logits:[[ 0.6630386  -0.5972838 ]\n",
            " [ 0.60822845 -0.37219045]\n",
            " [ 0.65590554 -0.37110254]\n",
            " [ 0.57779366 -0.3393522 ]\n",
            " [ 0.55652654 -0.3132416 ]\n",
            " [ 0.62980807 -0.40256238]\n",
            " [ 0.6618312  -0.45446956]\n",
            " [ 0.622973   -0.35540372]]\n",
            "Softmax:tensor([[0.6387, 0.4169],\n",
            "        [0.6349, 0.4163],\n",
            "        [0.6354, 0.4161],\n",
            "        [0.6360, 0.4166]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0],\n",
            "        [3, 3],\n",
            "        [2, 1],\n",
            "        [1, 2]], device='cuda:0')\n",
            "ur studies indicate that both nuclear matrix association and DNA binding are required for optimal SATB1-mediated repression of the integrated MMTV promoter and may allow insulation from cellular regulatory elements. knock-out of Satb1 significantly inhibited cell viability and migration, and promoted Schwann cells apoptosis. SATB1 is essential for maintaining TCR responsiveness during the induction and effector phases and may provide a novel therapeutic target for T cell-mediated autoimmune diseases.  T cells from SATB1 conditional knockout (SATB1cKO) mice, in which the Satb1 gene is deleted from hematopoietic cells, impair phosphorylation of signaling molecules in response to T cell receptor (TCR) crosslinking\n",
            "Logits:[[ 0.56985885 -0.33546114]\n",
            " [ 0.55324966 -0.337937  ]\n",
            " [ 0.5555056  -0.3387985 ]\n",
            " [ 0.5580108  -0.33670416]]\n",
            "Softmax:tensor([[0.6397, 0.4063]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "This system was successfully tested in a Phase 2 clinical trial for the treatment of post-menopausal women with osteoporosis.\n",
            "Logits:[[ 0.5739062 -0.3791465]]\n",
            "Softmax:tensor([[0.6347, 0.4197],\n",
            "        [0.6300, 0.4181],\n",
            "        [0.6392, 0.4190],\n",
            "        [0.6391, 0.4176],\n",
            "        [0.6392, 0.4190],\n",
            "        [0.6391, 0.4176]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [4, 4],\n",
            "        [5, 2],\n",
            "        [3, 1],\n",
            "        [0, 3]], device='cuda:0')\n",
            "Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. The identification and characterization of signal pleckstrin homology (PH)-domain-binding motifs established critical sites for breast cancer growth in PAR1&2. The identification and characterization of signal pleckstrin homology (PH)-domain-binding motifs established critical sites for breast cancer growth in PAR1&2. Although emerging roles of protease-activated receptor1&2 (PAR1&2) in cancer are recognized, their underlying signalling events are poorly understood. Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. This occurs via the association of the pleckstrin homology (PH) domain with Akt/PKB as a key signalling event of PARs. Other PH-domain signal-proteins such as Etk/Bmx and Vav3 also associate with PAR1 and PAR2 through their PH domains. PAR1 and PAR2 bind with priority to Etk/Bmx. A point mutation in PAR2, H349A, but not in R352A, abrogates PH-protein association and is sufficient to markedly reduce PAR2-instigated breast tumour growth in vivo and placental extravillous trophoblast (EVT) invasion in vitro. Similarly, the PAR1 mutant hPar1-7A, which is unable to bind the PH domain, reduces mammary tumours and EVT invasion, endowing these motifs with physiological significance and underscoring the importance of these previously unknown PAR1 and PAR2 PH-domain-binding motifs in both pathological and physiological invasion processes.\n",
            "Logits:[[ 0.5524723  -0.32394165]\n",
            " [ 0.5321624  -0.33068272]\n",
            " [ 0.57184803 -0.32706922]\n",
            " [ 0.5714694  -0.33252007]\n",
            " [ 0.57184803 -0.32706922]\n",
            " [ 0.5714694  -0.33252007]]\n",
            "Softmax:tensor([[0.6407, 0.4171],\n",
            "        [0.6383, 0.4176],\n",
            "        [0.6426, 0.4104],\n",
            "        [0.6342, 0.4237]], device='cuda:0')\n",
            "top_indices:tensor([[2, 3],\n",
            "        [0, 1],\n",
            "        [1, 0],\n",
            "        [3, 2]], device='cuda:0')\n",
            "Delays in vaccination can stymie the development of herd immunity, and a large proportion of children in the U.S. are known not to receive vaccines on time. Vaccines are very effective in providing individual and community (herd) immunity against a range of diseases.  We argue that individuals who have access to vaccines and for whom vaccination is not medically contraindicated have a moral obligation to contribute to the realisation of herd immunity by being vaccinated. is to drive population control of disease beyond those who are vaccinated (i.e. through herd immunity). \n",
            "Logits:[[ 0.5785791  -0.33464453]\n",
            " [ 0.5681066  -0.3327727 ]\n",
            " [ 0.58686    -0.3624816 ]\n",
            " [ 0.5504347  -0.30755475]]\n",
            "Softmax:tensor([[0.6339, 0.4194],\n",
            "        [0.6337, 0.4189],\n",
            "        [0.6342, 0.4188],\n",
            "        [0.6337, 0.4189],\n",
            "        [0.6342, 0.4188],\n",
            "        [0.6342, 0.4188]], device='cuda:0')\n",
            "top_indices:tensor([[5, 0],\n",
            "        [4, 3],\n",
            "        [2, 1],\n",
            "        [0, 4],\n",
            "        [1, 2]], device='cuda:0')\n",
            "Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. There was no difference between the multipotent adult progenitor cell group and placebo groups in global stroke recovery at day 90 (odds ratio 1·08 [95% CI 0·55-2·09], p=0·83).INTERPRETATION: Administration of multipotent adult progenitor cells was safe and well tolerated in patients with acute ischaemic stroke. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. INTERPRETATION\n",
            "\n",
            "Administration of multipotent adult progenitor cells was safe and well tolerated in patients with acute ischaemic stroke.\n",
            "Logits:[[ 0.5487964  -0.32541618]\n",
            " [ 0.5483246  -0.3274055 ]\n",
            " [ 0.5501418  -0.32754147]\n",
            " [ 0.5483246  -0.3274055 ]\n",
            " [ 0.5501418  -0.32754147]\n",
            " [ 0.5501418  -0.32754147]]\n",
            "Softmax:tensor([[0.6467, 0.4105],\n",
            "        [0.6365, 0.4160],\n",
            "        [0.6465, 0.4132]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [2, 2],\n",
            "        [1, 0]], device='cuda:0')\n",
            " MS and hospitalized were included and analyzed after which they were divided into the primary progressive MS A and B groups, the relapsing-remitting MS (RRMS) C and D groups  relapsing-remitting MS, primary and secondary progressive MS)  Multiple sclerosis presents with different phenotypes, most commonly a relapsing-remitting course and, less frequently, a progressive accumulation of disability from disease onset (primary progressive multiple sclerosis). The majority of people with relapsing-remitting multiple sclerosis, after a variable time, switch to a stage characterised by gradual neurological worsening known as secondary progressive multiple sclerosis.\n",
            "Logits:[[ 0.60475814 -0.3619507 ]\n",
            " [ 0.5601394  -0.33925632]\n",
            " [ 0.60377586 -0.35059822]]\n",
            "Softmax:tensor([[0.6398, 0.4153],\n",
            "        [0.6340, 0.4161],\n",
            "        [0.6393, 0.4117],\n",
            "        [0.6537, 0.3801],\n",
            "        [0.6343, 0.4168],\n",
            "        [0.6360, 0.4125],\n",
            "        [0.6356, 0.4184],\n",
            "        [0.6392, 0.4156],\n",
            "        [0.6356, 0.4184],\n",
            "        [0.6360, 0.4125],\n",
            "        [0.6356, 0.4184],\n",
            "        [0.6360, 0.4125],\n",
            "        [0.6393, 0.4117],\n",
            "        [0.6356, 0.4184]], device='cuda:0')\n",
            "top_indices:tensor([[ 3,  8],\n",
            "        [ 0,  6],\n",
            "        [12, 10],\n",
            "        [ 2, 13],\n",
            "        [ 7,  4]], device='cuda:0')\n",
            " The anti-LINGO-1 trial showed that the drug is safe and tolerable. A future phase II trial will provide more insights regarding the compound. Based on our knowledge of pathophysiology, three therapeutic strategies are proposed: anti-inflammatory (ocrelizumab, siponimod…); remyelinating (opicinumab); and neuroprotective (high-dose biotin, ibudilast, simvastatin…). Nevertheless, despite recent promising positive clinical trials, new methodological approaches for therapeutic protocols with adaptable outcomes to assess progression are still needed. Anti-LINGO-1 (opicinumab) is the first investigational product that achieved phase I trial with the aim of remyelination and axonal protection and/or repair in MS. Anti-LINGO-1 (opicinumab) is the first investigational product that achieved phase I trial with the aim of remyelination and axonal protection and/or repair in MS.  Participants (aged 18-58 years) with relapsing multiple sclerosis (relapsing-remitting multiple sclerosis and secondary progressive multiple sclerosis with relapses) were randomised in a 1:2:2:2:2 ratio by an interactive voice and web response system to opicinumab 3 mg/kg, 10 mg/kg, 30 mg/kg, or 100 mg/kg, or placebo.\n",
            "Logits:[[ 0.5745962  -0.34212306]\n",
            " [ 0.5495543  -0.33898452]\n",
            " [ 0.57223684 -0.35684383]\n",
            " [ 0.6352943  -0.4889736 ]\n",
            " [ 0.550678   -0.33605805]\n",
            " [ 0.55814534 -0.35384512]\n",
            " [ 0.5564494  -0.32939065]\n",
            " [ 0.5717569  -0.3408537 ]\n",
            " [ 0.5564494  -0.32939065]\n",
            " [ 0.55814534 -0.35384512]\n",
            " [ 0.5564494  -0.32939065]\n",
            " [ 0.55814534 -0.35384512]\n",
            " [ 0.57223684 -0.35684383]\n",
            " [ 0.5564494  -0.32939065]]\n",
            "Softmax:tensor([[0.6348, 0.4158],\n",
            "        [0.6439, 0.4018],\n",
            "        [0.6368, 0.4160],\n",
            "        [0.6378, 0.4175],\n",
            "        [0.6358, 0.4163],\n",
            "        [0.6353, 0.4152],\n",
            "        [0.6353, 0.4172],\n",
            "        [0.6361, 0.4153],\n",
            "        [0.6378, 0.4161],\n",
            "        [0.6353, 0.4152],\n",
            "        [0.6377, 0.4138],\n",
            "        [0.6344, 0.4163],\n",
            "        [0.6439, 0.4018],\n",
            "        [0.6439, 0.4018],\n",
            "        [0.6378, 0.4175],\n",
            "        [0.6358, 0.4163],\n",
            "        [0.6377, 0.4138],\n",
            "        [0.6378, 0.4161],\n",
            "        [0.6378, 0.4161],\n",
            "        [0.6353, 0.4152]], device='cuda:0')\n",
            "top_indices:tensor([[13,  3],\n",
            "        [12, 14],\n",
            "        [ 1,  6],\n",
            "        [14, 11],\n",
            "        [ 3,  4]], device='cuda:0')\n",
            "In this respect , several B cell-targeted therapies emerged , including anti-CD20 antibodies ( rituximab , ocrelizumab , and ofatumumab) , anti-CD19 antibody ( inebilizumab) , and agents targeting the BAFF/APRIL signaling pathway ( atacicept , belimumab , and LY2127399) .  In this respect, several B cell-targeted therapies emerged, including anti-CD20 antibodies (rituximab, ocrelizumab, and ofatumumab), anti-CD19 antibody (inebilizumab), and agents targeting the BAFF/APRIL signaling pathway (atacicept, belimumab, and LY2127399). In this respect, several B cell-targeted therapies emerged, including anti-CD20 antibodies (rituximab, ocrelizumab, and ofatumumab), anti-CD19 antibody (inebilizumab), and agents targeting the BAFF/APRIL signaling pathway (atacicept, belimumab, and LY2127399).  In NMO , though there have yet to be any approved monoclonal antibodies , rituximab , anti-complement C5 ( eculizumab) , anti-IL-6 receptor ( tocilizumab) , anti-CD19 ( inebilizumab ) and non-pathogenic anti-aquaporin 4 ( aquaporumab ) have been suggested to be effective , and some of these are now under clinical trials .  In NMO, though there have yet to be any approved monoclonal antibodies, rituximab, anti-complement C5 (eculizumab), anti-IL-6 receptor (tocilizumab), anti-CD19 (inebilizumab) and non-pathogenic anti-aquaporin 4 (aquaporumab) have been suggested to be effective, and some of these are now under clinical trials.\n",
            "Logits:[[ 0.55298525 -0.34016907]\n",
            " [ 0.5922437  -0.39804623]\n",
            " [ 0.5615136  -0.339285  ]\n",
            " [ 0.56600213 -0.33310804]\n",
            " [ 0.55722934 -0.33804473]\n",
            " [ 0.5548085  -0.3424095 ]\n",
            " [ 0.5549511  -0.33419177]\n",
            " [ 0.55826616 -0.3421018 ]\n",
            " [ 0.5656267  -0.33888048]\n",
            " [ 0.5548085  -0.3424095 ]\n",
            " [ 0.565202   -0.34835586]\n",
            " [ 0.5509603  -0.337994  ]\n",
            " [ 0.5922437  -0.39804623]\n",
            " [ 0.5922437  -0.39804623]\n",
            " [ 0.56600213 -0.33310804]\n",
            " [ 0.55722934 -0.33804473]\n",
            " [ 0.565202   -0.34835586]\n",
            " [ 0.5656267  -0.33888048]\n",
            " [ 0.5656267  -0.33888048]\n",
            " [ 0.5548085  -0.3424095 ]]\n",
            "Softmax:tensor([[0.6347, 0.4177],\n",
            "        [0.6374, 0.4155],\n",
            "        [0.6373, 0.4169],\n",
            "        [0.6351, 0.4186],\n",
            "        [0.6370, 0.4180],\n",
            "        [0.6357, 0.4162],\n",
            "        [0.6351, 0.4186]], device='cuda:0')\n",
            "top_indices:tensor([[1, 3],\n",
            "        [2, 6],\n",
            "        [4, 4],\n",
            "        [5, 0],\n",
            "        [3, 2]], device='cuda:0')\n",
            "At week 4, the proportion of the patients able to walk independently (functional grade ≤2) was 61% (90% CI 42-78; n=14) in the eculizumab group, and 45% (20-73; n=5) in the placebo group. Adverse events occurred in all 34 patients. Three patients had serious adverse events: two in the eculizumab group (anaphylaxis in one patient and intracranial haemorrhage and abscess in another patient) and one in the placebo group (depression). For the primary efficacy outcome at 4 weeks after recruitment, two of two placebo- and two of five eculizumab-treated subjects had improved by one or more grades on the GBS DS. At week 4, the proportion of the patients able to walk independently (functional grade ≤2) was 61% (90% CI 42-78; n=14) in the eculizumab group, and 45% (20-73; n=5) in the placebo group. A recent Japanese randomized controlled trial with eculizumab, a monoclonal antibody against the complement C5, indicated that eculizumab might improve the outcomes of GBS patients at six months from onset. However, because this is a small study without statistical comparison with the placebo group, the efficacy and safety of eculizumab could be investigated in larger, randomised controlled trials.\n",
            "Logits:[[ 0.55231196 -0.33219227]\n",
            " [ 0.5640086  -0.34130323]\n",
            " [ 0.5635747  -0.3353976 ]\n",
            " [ 0.55434126 -0.32859865]\n",
            " [ 0.56237936 -0.33101195]\n",
            " [ 0.55668324 -0.33847547]\n",
            " [ 0.55434126 -0.32859865]]\n",
            "Softmax:tensor([[0.6351, 0.4171],\n",
            "        [0.6348, 0.4195],\n",
            "        [0.6325, 0.4186],\n",
            "        [0.6356, 0.4155],\n",
            "        [0.6327, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[3, 1],\n",
            "        [0, 2],\n",
            "        [1, 4],\n",
            "        [4, 0],\n",
            "        [2, 3]], device='cuda:0')\n",
            "Mcl-1 inhibitor S63845 The recent description of S63845 as the first specific and potent MCL-1 inhibitor represents an important therapeutic advance  We tested the efficacy of the BH3 mimetic combination of A-1210477 (an MCL-1 inhibitor) A-1210477, a selective MCL-1 inhibitor a highly specific MCL-1 inhibitor, S63845,\n",
            "Logits:[[ 0.5543485  -0.33462894]\n",
            " [ 0.5530721  -0.3248018 ]\n",
            " [ 0.54287744 -0.32857668]\n",
            " [ 0.5562132  -0.34111142]\n",
            " [ 0.5437016  -0.33309954]]\n",
            "Softmax:tensor([[0.6402, 0.4153],\n",
            "        [0.6382, 0.4119],\n",
            "        [0.6421, 0.4085],\n",
            "        [0.6400, 0.4078],\n",
            "        [0.6415, 0.4101],\n",
            "        [0.6370, 0.4125],\n",
            "        [0.6390, 0.4116],\n",
            "        [0.6370, 0.4125],\n",
            "        [0.6415, 0.4101],\n",
            "        [0.6403, 0.4103],\n",
            "        [0.6390, 0.4113]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [8, 7],\n",
            "        [4, 5],\n",
            "        [9, 1],\n",
            "        [0, 6]], device='cuda:0')\n",
            "Autophagy is a cellular survival pathway that is necessary for the degradation of cellular constituents such as long-lived proteins and damaged organelles. Autophagy is an evolutionarily conserved cellular process in which intracellular components are eliminated via lysosomal degradation to supply nutrients for organelle biogenesis and metabolic homeostasis. Autophagy is an evolutionarily conserved cellular process in which intracellular components are eliminated via lysosomal degradation to supply nutrients for organelle biogenesis and metabolic homeostasis. Autophagy is a homeostatic process involved in the turnover or elimination of cytoplasmic components, damaged organelles, and protein aggregates via a lysosomal degradation mechanism. In this study, we demonstrate that autophagy is a critical mediator of the viral degradation pathway and that this pathway is not HIV-1 specific.\n",
            "Logits:[[ 0.5761771  -0.34197342]\n",
            " [ 0.5677551  -0.35608327]\n",
            " [ 0.5843852  -0.37015978]\n",
            " [ 0.5751979  -0.37313583]\n",
            " [ 0.58167434 -0.36345732]\n",
            " [ 0.56243026 -0.35366434]\n",
            " [ 0.57093644 -0.35725412]\n",
            " [ 0.56243026 -0.35366434]\n",
            " [ 0.58167434 -0.36345732]\n",
            " [ 0.5765498  -0.3628804 ]\n",
            " [ 0.57086915 -0.35839057]]\n",
            "Softmax:tensor([[0.6307, 0.4195],\n",
            "        [0.6331, 0.4202]], device='cuda:0')\n",
            "top_indices:tensor([[1, 1],\n",
            "        [0, 0]], device='cuda:0')\n",
            "MPE-seq provides a unique and straightforward means for the genome-wide analysis of chromatin structure with minimal DNA sequence bias.  methidiumpropyl-EDTA sequencing (MPE-seq), a method for the genome-wide characterization of chromatin that involves the digestion of nuclei withMPE-Fe(II) followed by massively parallel sequencing. \n",
            "Logits:[[ 0.53531224 -0.32471353]\n",
            " [ 0.5457441  -0.3219714 ]]\n",
            "Softmax:tensor([[0.6345, 0.4150]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "Two rotavirus vaccines, RotaTeq and Rotarix, are licensed for global use; however, the protection they confer to unvaccinated individuals through indirect effects remains unknown. \n",
            "Logits:[[ 0.5514116 -0.3433547]]\n",
            "Softmax:tensor([[0.6399, 0.4159],\n",
            "        [0.6366, 0.4162],\n",
            "        [0.6376, 0.4186],\n",
            "        [0.6378, 0.4149]], device='cuda:0')\n",
            "top_indices:tensor([[0, 2],\n",
            "        [3, 1],\n",
            "        [2, 0],\n",
            "        [1, 3]], device='cuda:0')\n",
            "This revealed a 637-kb tandem duplication that in addition to DAX1 includes the four MAGEB genes, the hypothetical gene CXorf21, GK, and part of the MAP3K7IP3 gene Examination of X-linked DEGs, such as GTPBP6, TAF9L, and CXORF21, that show verbal cognition-gene expression correlations may establish a causal link between these genes, neurodevelopment, and language function.  For this, we selected five SNPs (rs1801274 in FCGR2A and rs2286672 in PLD2, rs887369 in CXorf21, rs9782955 in LYST, and rs3794060 in NADSYN1) Among statin users, diabetes cases had marginal but insignificantly different expression of ZNF532 (up-regulated 15%, Q-value=0.0584), CXORF21 (up-regulated 11%, Q-value=0.0584), and ZNHIT3 (up-regulated 19%, Q-value=0.0959), compared with controls.\n",
            "Logits:[[ 0.57483095 -0.3396503 ]\n",
            " [ 0.5606458  -0.33836028]\n",
            " [ 0.5649181  -0.32866317]\n",
            " [ 0.56579316 -0.34380785]]\n",
            "Softmax:tensor([[0.6430, 0.4217],\n",
            "        [0.6417, 0.4161]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0],\n",
            "        [1, 1]], device='cuda:0')\n",
            "In the context of genome-wide association studies (GWAS), there is a variety of statistical techniques in order to conduct the analysis, but, in most cases, the underlying genetic model is usually unknown. Under these circumstances, the classical Cochran-Armitage trend test (CATT) is suboptimal. Robust procedures that maximize the power and preserve the nominal type I error rate are preferable. Moreover, performing a meta-analysis using robust procedures is of great interest and has never been addressed in the past. The primary goal of this work is to implement several robust methods for analysis and meta-analysis in the statistical package Stata and subsequently to make the software available to the scientific community.Results: The CATT under a recessive, additive and dominant model of inheritance as well as robust methods based on the Maximum Efficiency Robust Test statistic, the MAX statistic and the MIN2 were implemented in Stata. Concerning MAX and MIN2, we calculated their asymptotic null distributions relying on numerical integration resulting in a great gain in computational time without losing accuracy. All the aforementioned approaches were employed in a fixed or a random effects meta-analysis setting using summary data with weights equal to the reciprocal of the combined cases and controls. Overall, this is the first complete effort to implement procedures for analysis and meta-analysis in GWAS using Stata.Availability and Implementation: A Stata program and a web-server are freely available for academic users at http://www.compgen.org/tools/GWAR. Availability and Implementation\n",
            "\n",
            "A Stata program and a web-server are freely available for academic users at http://www.compgen.org/tools/GWAR.\n",
            "Logits:[[ 0.58851844 -0.3159349 ]\n",
            " [ 0.58254284 -0.33863628]]\n",
            "Softmax:tensor([[0.6322, 0.4189],\n",
            "        [0.6346, 0.4173],\n",
            "        [0.6313, 0.4222],\n",
            "        [0.6306, 0.4226],\n",
            "        [0.6324, 0.4185],\n",
            "        [0.6313, 0.4222]], device='cuda:0')\n",
            "top_indices:tensor([[1, 3],\n",
            "        [4, 5],\n",
            "        [0, 2],\n",
            "        [5, 0],\n",
            "        [2, 4]], device='cuda:0')\n",
            "Single-cell RNA-sequencing (scRNA-seq) allows studying heterogeneity in gene expression in large cell populations. Such heterogeneity can arise due to technical or biological factors, making decomposing sources of variation difficult. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity. Our model jointly estimates the relevance of individual factors, refines gene set annotations, and infers factors without annotation. In applications to multiple scRNA-seq datasets, we find that f-scLVM robustly decomposes scRNA-seq datasets into interpretable components, thereby facilitating the identification of novel subpopulations. f-scLVM: scalable and versatile factor analysis for single-cell RNA-seq f-scLVM: scalable and versatile factor analysis for single-cell RNA-seq. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity.\n",
            "Logits:[[ 0.5417826  -0.32742932]\n",
            " [ 0.55217004 -0.33378536]\n",
            " [ 0.5378431  -0.31372982]\n",
            " [ 0.53474224 -0.31227073]\n",
            " [ 0.5423483  -0.3291124 ]\n",
            " [ 0.5378431  -0.31372982]]\n",
            "Softmax:tensor([[0.6331, 0.4167],\n",
            "        [0.6380, 0.4167],\n",
            "        [0.6319, 0.4168]], device='cuda:0')\n",
            "top_indices:tensor([[1, 2],\n",
            "        [0, 1],\n",
            "        [2, 0]], device='cuda:0')\n",
            "Animal promoters initiate transcription either at precise positions (narrow promoters) or dispersed regions (broad promoters), a distinction referred to as promoter shape. Although highly conserved, the functional properties of promoters with different shapes and the genetic basis of their evolution remain unclear. Here we used natural genetic variation across a panel of 81 Drosophila lines to measure changes in transcriptional start site (TSS) usage, identifying thousands of genetic variants affecting transcript levels (strength) or the distribution of TSSs within a promoter (shape). Our results identify promoter shape as a molecular trait that can evolve independently of promoter strength. Broad promoters typically harbor shape-associated variants, with signatures of adaptive selection. Single-cell measurements demonstrate that variants modulating promoter shape often increase expression noise, whereas heteroallelic interactions with other promoter variants alleviate these effects. These results uncover new functional properties of natural promoters and suggest the minimization of expression noise as an important factor in promoter evolution. Promoter shape varies across populations and affects promoter evolution and expression noise. Promoter shape varies across populations and affects promoter evolution and expression noise\n",
            "Logits:[[ 0.5456322  -0.3364193 ]\n",
            " [ 0.5666995  -0.3362009 ]\n",
            " [ 0.54050416 -0.33609083]]\n",
            "Softmax:tensor([[0.6357, 0.4188],\n",
            "        [0.6366, 0.4175],\n",
            "        [0.6373, 0.4186],\n",
            "        [0.6337, 0.4231]], device='cuda:0')\n",
            "top_indices:tensor([[2, 3],\n",
            "        [1, 0],\n",
            "        [0, 2],\n",
            "        [3, 1]], device='cuda:0')\n",
            " Long noncoding RNAs (lncRNAs) are more than 200 nucleotides in length and lack transcriptional ability. lncRNAs are the nonprotein coding RNAs that have a size longer than 200 nucleotides.   lncRNAs are defined as RNA transcripts longer than 200 nucleotides that are not transcribed into proteins  long noncoding RNA (lncRNA, >200 nucleotides)\n",
            "Logits:[[ 0.5567899  -0.3276133 ]\n",
            " [ 0.56078285 -0.3331835 ]\n",
            " [ 0.5635429  -0.32868552]\n",
            " [ 0.54795086 -0.31004953]]\n",
            "Softmax:tensor([[0.6393, 0.4176],\n",
            "        [0.6327, 0.4187]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            "This phase 2, randomised, double-blind, placebo-controlled, single-centre, crossover trial assessed the effectiveness of an oral neurokinin 3 receptor antagonist (MLE4901) on menopausal hot flushes.  Treatment with a neurokinin 3 receptor antagonist (MLE4901) could be practice changing as it safely and effectively relieves hot flush symptoms without the need for oestrogen exposure. \n",
            "Logits:[[ 0.5722696  -0.33279032]\n",
            " [ 0.54382825 -0.3280216 ]]\n",
            "Softmax:tensor([[0.6369, 0.4149],\n",
            "        [0.6369, 0.4155],\n",
            "        [0.6391, 0.4197],\n",
            "        [0.6356, 0.4168],\n",
            "        [0.6408, 0.4115]], device='cuda:0')\n",
            "top_indices:tensor([[4, 2],\n",
            "        [2, 3],\n",
            "        [1, 1],\n",
            "        [0, 0],\n",
            "        [3, 4]], device='cuda:0')\n",
            "Chloroquine is a potent inhibitor of SARS coronavirus infection and spread. we investigated how chloroquine (CQ), a commonly employed disease-modifying antirheumatic drug, Historically chloroquine was used to treat the most deadly form of malaria, caused by the parasite Plasmodium falciparum.  Chloroquine (CQ) has been used for decades as the primary chemotherapeutic drug for the treatment of malaria.  Hydroxychloroquine (HCQ), a 4-aminoquinolone antimalarial, is regarded as the oral therapy of choice for cutaneous and systemic lupus erythematosus (SLE). It is also licensed for rheumatoid arthritis (RA).\n",
            "Logits:[[ 0.5617216  -0.34373888]\n",
            " [ 0.5620285  -0.34110743]\n",
            " [ 0.5713816  -0.32396147]\n",
            " [ 0.55647856 -0.33599445]\n",
            " [ 0.5786957  -0.35795742]]\n",
            "Softmax:tensor([[0.6400, 0.4139],\n",
            "        [0.6451, 0.4120],\n",
            "        [0.6383, 0.4182],\n",
            "        [0.6416, 0.4177],\n",
            "        [0.6430, 0.4150],\n",
            "        [0.6433, 0.4145],\n",
            "        [0.6443, 0.4131],\n",
            "        [0.6416, 0.4177],\n",
            "        [0.6433, 0.4145],\n",
            "        [0.6424, 0.4173]], device='cuda:0')\n",
            "top_indices:tensor([[1, 2],\n",
            "        [6, 7],\n",
            "        [8, 3],\n",
            "        [5, 9],\n",
            "        [4, 4]], device='cuda:0')\n",
            "The medications comprised vitamin E, baclofen, riluzole, L-threonine, xaliproden, indinavir, and memantine. Six studies assessed cramps as an adverse event. The medications comprised creatine, gabapentin, dextromethorphan, quinidine, and lithium. In all 20 studies no favourable effect for the treatment of cramps in ALS/MND could be demonstrated, but many studies were underpowered to draw a definite conclusion. An effect of xaliproden on functional parameters, especially VC, was noted. The six months intent-to-treat analysis showed no statistically significant effect but a trend in favour of 2 mg xaliproden compared to placebo for reduction in the rate of deterioration of FVC, limbs functional score, and manual muscle testing score (MMT). The six months intent-to-treat analysis showed no statistically significant effect but a trend in favour of 2 mg xaliproden compared to placebo for reduction in the rate of deterioration of FVC, limbs functional score, and manual muscle testing score (MMT). An effect of xaliproden on functional parameters, especially VC, was noted. Although this effect did not reach statistical significance, xaliproden had a small effect on clinically noteworthy aspects of disease progression in ALS.\n",
            "Logits:[[ 0.5753404  -0.34801626]\n",
            " [ 0.59752285 -0.35578233]\n",
            " [ 0.5679873  -0.32998747]\n",
            " [ 0.5825043  -0.33237717]\n",
            " [ 0.5885378  -0.3435079 ]\n",
            " [ 0.5896948  -0.34552187]\n",
            " [ 0.594034   -0.35131112]\n",
            " [ 0.5825043  -0.33237717]\n",
            " [ 0.5896948  -0.34552187]\n",
            " [ 0.5856354  -0.33398688]]\n",
            "Softmax:tensor([[0.6371, 0.4183],\n",
            "        [0.6334, 0.4197],\n",
            "        [0.6331, 0.4207],\n",
            "        [0.6369, 0.4164],\n",
            "        [0.6390, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[4, 2],\n",
            "        [0, 1],\n",
            "        [3, 0],\n",
            "        [1, 4],\n",
            "        [2, 3]], device='cuda:0')\n",
            "Telangiectasia (macroscopically visible dilated skin vessels) occurring primarily on the hands and face, are a prominent feature in scleroderma and are present in the majority of patients. Telangiectasias are prominent small vessels (venules, capillaries or arterioles) that are visible as small red-purple focal lesions in the skin and mucous membranes. Hereditary hemorrhagic telangiectasia is a dominant disorder characterized by epistaxis, visceral arteriovenous malformations, and telangiectasias. Telangiectasias are small focal red macules and papules created by abnormally prominent capillaries, venules, and arterioles  Telangiectasia (macroscopically visible dilated skin vessels)\n",
            "Logits:[[ 0.56264204 -0.3296505 ]\n",
            " [ 0.54685074 -0.3239435 ]\n",
            " [ 0.5455875  -0.3200847 ]\n",
            " [ 0.56178033 -0.3376081 ]\n",
            " [ 0.5711088  -0.33318225]]\n",
            "Softmax:tensor([[0.6302, 0.4187],\n",
            "        [0.6320, 0.4193],\n",
            "        [0.6350, 0.4172],\n",
            "        [0.6312, 0.4178]], device='cuda:0')\n",
            "top_indices:tensor([[2, 1],\n",
            "        [1, 0],\n",
            "        [3, 3],\n",
            "        [0, 2]], device='cuda:0')\n",
            "Immunofluorescence and immunoblotting investigations revealed the presence of cathepsin L in the nuclear compartment in addition to its expected endo-lysosomal localization in colorectal carcinoma cells. endosomal cathepsin L cleavage by the endosomal/lysosomal protease cathepsin L Cathepsin L in the Late Endosome/Lysosome\n",
            "Logits:[[ 0.53317034 -0.32801333]\n",
            " [ 0.5406064  -0.32560948]\n",
            " [ 0.5537885  -0.33428782]\n",
            " [ 0.53748196 -0.33199638]]\n",
            "Softmax:tensor([[0.6401, 0.4197],\n",
            "        [0.6457, 0.4114],\n",
            "        [0.6412, 0.4141],\n",
            "        [0.6350, 0.4166],\n",
            "        [0.6416, 0.4140]], device='cuda:0')\n",
            "top_indices:tensor([[1, 0],\n",
            "        [4, 3],\n",
            "        [2, 2],\n",
            "        [0, 4],\n",
            "        [3, 1]], device='cuda:0')\n",
            "ACE2 has been identified in various tissues and is supposed to be a modulator of cardiovascular function. Decreases in ACE2 expression and activity have been reported in models of hypertension, heart failure, atherosclerosis, diabetic nephropathy and others.  Abundant ACE2 immunostaining was found in lung, kidney, heart, and islets of pancreas, but not in hepatocytes ACE2 is expressed at high level in the primary target cells of SARS-CoV, namely pneumocytes and surface enterocytes of the small intestine. in keeping with the variability of ACE2 expression we have observed previously endothelial cells, which express ACE2 to a high level,\n",
            "Logits:[[ 0.57596666 -0.32387546]\n",
            " [ 0.6000179  -0.35822275]\n",
            " [ 0.5805513  -0.34704903]\n",
            " [ 0.5539088  -0.33656773]\n",
            " [ 0.58216774 -0.34756795]]\n",
            "Softmax:tensor([[0.6345, 0.4194],\n",
            "        [0.6353, 0.4168],\n",
            "        [0.6394, 0.4181],\n",
            "        [0.6385, 0.4184],\n",
            "        [0.6394, 0.4181]], device='cuda:0')\n",
            "top_indices:tensor([[2, 0],\n",
            "        [4, 3],\n",
            "        [3, 4],\n",
            "        [1, 2],\n",
            "        [0, 1]], device='cuda:0')\n",
            "We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). We selected one clone , LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange ( RMCE Transgenesis of human pluripotent stem cells (hPSCs) can enable and empower a variety of studies in stem cell research, including lineage tracing and functional genetics studies. While in recent years much progress has been made in the development of tools for gene targeting, little attention has been given to the identification of sites in the human genome where transgenes can be inserted and reliably expressed. In order to find human genomic sites capable of supporting long-term and high-level transgene expression in hPSCs, we performed a lentiviral screen in human induced pluripotent stem cells (iPSCs). We isolated 40 iPSC clones each harboring a single vector copy and characterized the level of transgene expression afforded by each unique integration site. We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). The LiPS-A3 line can greatly facilitate the insertion of reporter and other genes in hPSCs. Targeting transgenes in the LiPS-A3S genomic locus can find broad applications in stem cell research and possibly cell and gene therapy. LiPS-A3S, a human genomic site for robust expression of inserted transgenes.\n",
            "Logits:[[ 0.5514111  -0.32541886]\n",
            " [ 0.5550981  -0.33602312]\n",
            " [ 0.57264084 -0.3305723 ]\n",
            " [ 0.5689877  -0.32929245]\n",
            " [ 0.57264084 -0.3305723 ]]\n",
            "Softmax:tensor([[0.6355, 0.4153],\n",
            "        [0.6315, 0.4169],\n",
            "        [0.6473, 0.4117],\n",
            "        [0.6329, 0.4170]], device='cuda:0')\n",
            "top_indices:tensor([[2, 3],\n",
            "        [0, 1],\n",
            "        [3, 0],\n",
            "        [1, 2]], device='cuda:0')\n",
            "A random effects meta-analytical model was used to compare adverse events and functional outcome.RESULTS: Thirty comparative studies comparing J, W, S and K pouch designs were included Formation of a colonic J-pouch with anastomosis to the rectal stump is an accepted form of reconstruction after low anterior resection (LAR) for rectal carcinoma Ileal pouch-anal anastomosis, or J pouch, surgery has become the procedure of choice for treatment of medically refractory ulcerative colitis and familial adenomatous polyposis. total proctocolectomy with outcomes after ileal J-pouch anal anastomosis (IPAA) at a single institution\n",
            "Logits:[[ 0.55587685 -0.3419915 ]\n",
            " [ 0.5388471  -0.33541766]\n",
            " [ 0.6071006  -0.35677123]\n",
            " [ 0.544605   -0.33498254]]\n",
            "Softmax:tensor([[0.6316, 0.4185],\n",
            "        [0.6294, 0.4193],\n",
            "        [0.6328, 0.4180],\n",
            "        [0.6336, 0.4172]], device='cuda:0')\n",
            "top_indices:tensor([[3, 1],\n",
            "        [2, 0],\n",
            "        [0, 2],\n",
            "        [1, 3]], device='cuda:0')\n",
            "SLC12A1 (NKCC2) Na-K-Cl cotransporter 2 (NKCC2) 2 chloride co-transporter (NKCC2) gene Na-K-2Cl cotransporter (NKCC2) function \n",
            "Logits:[[ 0.5389452  -0.3289737 ]\n",
            " [ 0.5294477  -0.3258499 ]\n",
            " [ 0.54426515 -0.3310613 ]\n",
            " [ 0.54765826 -0.33441952]]\n",
            "Softmax:tensor([[0.6380, 0.4181],\n",
            "        [0.6350, 0.4178],\n",
            "        [0.6344, 0.4186],\n",
            "        [0.6359, 0.4181],\n",
            "        [0.6403, 0.4168],\n",
            "        [0.6403, 0.4168],\n",
            "        [0.6344, 0.4186],\n",
            "        [0.6375, 0.4173],\n",
            "        [0.6348, 0.4189],\n",
            "        [0.6356, 0.4177]], device='cuda:0')\n",
            "top_indices:tensor([[4, 8],\n",
            "        [5, 6],\n",
            "        [0, 2],\n",
            "        [7, 3],\n",
            "        [3, 0]], device='cuda:0')\n",
            "RESULTS\n",
            "\n",
            "Celecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. RESULTS\n",
            "Celecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. In conclusion, the celecoxib-creatine combination was selected as preferable to the minocycline-creatine combination for further evaluation.  RESULTS: Celecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. INTERPRETATION\n",
            "\n",
            "At the dosage studied, celecoxib did not have a beneficial effect on research subjects with ALS, and it was safe.\n",
            "Logits:[[ 0.56685466 -0.33070648]\n",
            " [ 0.553706   -0.33178145]\n",
            " [ 0.55116117 -0.32843038]\n",
            " [ 0.55757475 -0.33052817]\n",
            " [ 0.5766494  -0.33592495]\n",
            " [ 0.5766494  -0.33592495]\n",
            " [ 0.55116117 -0.32843038]\n",
            " [ 0.5644614  -0.33373687]\n",
            " [ 0.55303085 -0.32723796]\n",
            " [ 0.55647045 -0.3323055 ]]\n",
            "Softmax:tensor([[0.6349, 0.4190],\n",
            "        [0.6344, 0.4165],\n",
            "        [0.6339, 0.4159]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0],\n",
            "        [1, 1],\n",
            "        [2, 2]], device='cuda:0')\n",
            "increased expression of anti-apoptotic proteins (Bcl-xL, Mcl-1 and XIAP)  repression of anti-apoptotic proteins (Mcl-1, Bcl-xl and XIAP) anti-apoptotic BCL-2 family members, such as BCL-2, BCL-XL or MCL-1\n",
            "Logits:[[ 0.5533882  -0.32702902]\n",
            " [ 0.5511888  -0.33728674]\n",
            " [ 0.54878426 -0.3397365 ]]\n",
            "Softmax:tensor([[0.6385, 0.4111],\n",
            "        [0.6366, 0.4170],\n",
            "        [0.6385, 0.4111],\n",
            "        [0.6366, 0.4170],\n",
            "        [0.6375, 0.4126],\n",
            "        [0.6357, 0.4174],\n",
            "        [0.6346, 0.4147],\n",
            "        [0.6353, 0.4142],\n",
            "        [0.6362, 0.4143],\n",
            "        [0.6385, 0.4111],\n",
            "        [0.6353, 0.4142]], device='cuda:0')\n",
            "top_indices:tensor([[9, 5],\n",
            "        [2, 3],\n",
            "        [0, 1],\n",
            "        [4, 6],\n",
            "        [1, 8]], device='cuda:0')\n",
            "Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib. Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib. Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib.  Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib We show that the coadministration of G1T28 (trilaciclib), which is a small-molecule inhibitor of cyclin-dependent kinases 4 and 6 (CDK4/6), contemporaneously with cytotoxic chemotherapy protects murine hematopoietic stem cells (HSCs) from chemotherapy-induced exhaustion in a serial 5-fluorouracil treatment model. \n",
            "Logits:[[ 0.5690513  -0.3593568 ]\n",
            " [ 0.56067675 -0.3351997 ]\n",
            " [ 0.5690513  -0.3593568 ]\n",
            " [ 0.56067675 -0.3351997 ]\n",
            " [ 0.5646161  -0.35315117]\n",
            " [ 0.55672747 -0.33335277]\n",
            " [ 0.5517913  -0.34444386]\n",
            " [ 0.55505496 -0.3468239 ]\n",
            " [ 0.5587013  -0.34629178]\n",
            " [ 0.5690513  -0.3593568 ]\n",
            " [ 0.55505496 -0.3468239 ]]\n",
            "Softmax:tensor([[0.6344, 0.4184]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "To evaluate the clinical PK/PD of PTH(1-34) delivered by a novel transdermal drug-coated microneedle patch system (ZP-PTH) for the treatment of osteoporosis.\n",
            "Logits:[[ 0.5509612  -0.32929155]]\n",
            "Softmax:tensor([[0.6669, 0.3752],\n",
            "        [0.6353, 0.4159]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            "Parathyroid hormone (1-34)-coated microneedle patch system: clinical pharmacokinetics and pharmacodynamics for treatment of osteoporosis. To evaluate the clinical PK/PD of PTH(1-34) delivered by a novel transdermal drug-coated microneedle patch system (ZP-PTH) for the treatment of osteoporosis.\n",
            "Logits:[[ 0.6940189  -0.50999916]\n",
            " [ 0.554903   -0.33976054]]\n",
            "Softmax:tensor([[0.6355, 0.4172],\n",
            "        [0.6324, 0.4156],\n",
            "        [0.6294, 0.4179],\n",
            "        [0.6347, 0.4152]], device='cuda:0')\n",
            "top_indices:tensor([[0, 2],\n",
            "        [3, 0],\n",
            "        [1, 1],\n",
            "        [2, 3]], device='cuda:0')\n",
            "The highly variable phenotypes observed in patients with classic Bartter's syndrome (BS) remain unsatisfactorily explained. The wide spectrum of functional severity of CLCNKB mutations may contribute to the phenotypic variability, and the genotype-phenotype association has not been established.  Classic Bartter's syndrome has been demonstrated to result from defective chloride transport across the basolateral membrane in the distal nephron due to mutations in the chloride channel gene CLCNKB. Progressive renal failure in patients with classic Bartter's syndrome (cBS) due to inactivating mutations in CLCNKB gene is extraordinarily rare classic Bartter's syndrome by mutations of ClC-Kb\n",
            "Logits:[[ 0.5556814  -0.33422145]\n",
            " [ 0.54268324 -0.34085932]\n",
            " [ 0.52949333 -0.3312184 ]\n",
            " [ 0.55256647 -0.34248686]]\n",
            "Softmax:tensor([[0.6258, 0.4218],\n",
            "        [0.6281, 0.4234],\n",
            "        [0.6254, 0.4222],\n",
            "        [0.6310, 0.4226]], device='cuda:0')\n",
            "top_indices:tensor([[3, 1],\n",
            "        [1, 3],\n",
            "        [0, 2],\n",
            "        [2, 0]], device='cuda:0')\n",
            "We developed a search system named UC2 (Unique Connectivity of Uncharged Compounds), where compounds are tentatively neutralized into uncharged states and stored on the basis of their unique connectivity of atoms after removing their stereochemical information using the first block in the hash of the IUPAC International Chemical Identifier, by which false-positive hits are remarkably reduced, both charged and uncharged compounds are properly searched in a single query and records having a unique connectivity are compiled in a single search result.Availability and implementation: The UC2 search tool is available free of charge as a REST web service (http://webs2.kazusa.or.jp/mfsearcher) and a Java-based GUI tool.Contact: sakurai@kazusa.or.jp.Supplementary information: Supplementary data are available at Bioinformatics online. For metabolite annotation in metabolomics, variations in the registered states of compounds (charged molecules and multiple components, such as salts) and their redundancy among compound databases could be the cause of misannotations and hamper immediate recognition of the uniqueness of metabolites while searching by mass values measured using mass spectrometry. We developed a search system named UC2 (Unique Connectivity of Uncharged Compounds), where compounds are tentatively neutralized into uncharged states and stored on the basis of their unique connectivity of atoms after removing their stereochemical information using the first block in the hash of the IUPAC International Chemical Identifier, by which false-positive hits are remarkably reduced, both charged and uncharged compounds are properly searched in a single query and records having a unique connectivity are compiled in a single search result.Availability and implementation: The UC2 search tool is available free of charge as a REST web service (http://webs2.kazusa.or.jp/mfsearcher) and a Java-based GUI tool. UC2 search: using unique connectivity of uncharged compounds for metabolite annotation by database searching in mass spectrometry-based metabolomics. UC2 search: using unique connectivity of uncharged compounds for metabolite annotation by database searching in mass spectrometry-based metabolomics\n",
            "Logits:[[ 0.5141942  -0.31554243]\n",
            " [ 0.5239043  -0.30894014]\n",
            " [ 0.5124844  -0.31387392]\n",
            " [ 0.5366947  -0.31228593]]\n",
            "Softmax:tensor([[0.6332, 0.4166]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "Administration of senktide, highly potent and selective NK3R agonist, resulted in increase of serum LH concentration, induction of VMS, increase in heart rate, and skin temperature in postmenopausal women.\n",
            "Logits:[[ 0.54595107 -0.33665678]]\n",
            "Softmax:tensor([[0.6350, 0.4147],\n",
            "        [0.6379, 0.4184],\n",
            "        [0.6354, 0.4165]], device='cuda:0')\n",
            "top_indices:tensor([[1, 1],\n",
            "        [2, 2],\n",
            "        [0, 0]], device='cuda:0')\n",
            "Either angiotensin converting enzyme inhibitor (ACEI) or angiotensin receptor 1 blocker (ARB) attenuates cardiac remodeling. However, the overall molecular modulation of the reversing remodeling process in response to the ACEI or ARB treatment is not yet well determined. In this study, we examined whether gene expressions are modulated by ACEI (temocapril), ARB (olmesartan)  ARB olmesartan (OLM) Olmesartan (OL) is the pharmacologically active metabolite of Olmesartan medoxomil (OM), an FDA-approved angiotensin II receptor antagonist for administrating cardiovascular diseases.\n",
            "Logits:[[ 0.5538729  -0.34462786]\n",
            " [ 0.5664436  -0.32938385]\n",
            " [ 0.55536366 -0.3370226 ]]\n",
            "Softmax:tensor([[0.6334, 0.4208],\n",
            "        [0.6349, 0.4163],\n",
            "        [0.6318, 0.4183]], device='cuda:0')\n",
            "top_indices:tensor([[1, 0],\n",
            "        [0, 2],\n",
            "        [2, 1]], device='cuda:0')\n",
            "X-inactivation escape of the TLR7 gene was investigated in monoclonal B cell lines and, independently, in pDCs after cell sorting and single-cell picking, indicating regular silencing of one TLR7 allele in females.  TLR7 is encoded by an X chromosome locus, and we examined here whether the TLR7 gene evades silencing by X chromosome inactivation in immune cells from women and Klinefelter syndrome males Female predisposition to TLR7-driven autoimmunity: gene dosage and the escape from X chromosome inactivation\n",
            "Logits:[[ 0.5467671  -0.3193594 ]\n",
            " [ 0.55324036 -0.33810794]\n",
            " [ 0.5399901  -0.32982534]]\n",
            "Softmax:tensor([[0.6324, 0.4182],\n",
            "        [0.6380, 0.4152],\n",
            "        [0.6362, 0.4162],\n",
            "        [0.6404, 0.4175],\n",
            "        [0.6362, 0.4162],\n",
            "        [0.6404, 0.4175]], device='cuda:0')\n",
            "top_indices:tensor([[3, 0],\n",
            "        [5, 5],\n",
            "        [1, 3],\n",
            "        [4, 4],\n",
            "        [2, 2]], device='cuda:0')\n",
            "We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. Recent technological advances have enabled DNA methylation to be assayed at single-cell resolution. However, current protocols are limited by incomplete CpG coverage and hence methods to predict missing methylation states are critical to enable genome-wide analyses. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells. We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. DeepCpG yields substantially more accurate predictions than previous methods. Additionally, we show that the model parameters can be interpreted, thereby providing insights into how sequence composition affects methylation variability. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells.\n",
            "Logits:[[ 0.5423294  -0.330351  ]\n",
            " [ 0.56651765 -0.34259307]\n",
            " [ 0.55886126 -0.33836368]\n",
            " [ 0.57716167 -0.33293673]\n",
            " [ 0.55886126 -0.33836368]\n",
            " [ 0.57716167 -0.33293673]]\n",
            "Softmax:tensor([[0.6365, 0.4143]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "After a decade in obscurity it (Vaxchora) has resurfaced again, now produced in the U.S. and equipped with a U.S. FDA license (June 10, 2016). \n",
            "Logits:[[ 0.560286   -0.34620133]]\n",
            "Softmax:tensor([[0.6379, 0.4179],\n",
            "        [0.6418, 0.4145]], device='cuda:0')\n",
            "top_indices:tensor([[1, 0],\n",
            "        [0, 1]], device='cuda:0')\n",
            "We evaluated the efficacy and tolerability of 3- and 6-month leuprorelin acetate (LA) depot formulations (Eligard®, Astellas Pharma GmbH) in patients with advanced prostate cancer treated in routine clinical practice in Germany. Efficacy and Tolerability of Leuprorelin Acetate (Eligard®) in Daily Practice in Germany: Pooled Data from 2 Prospective, Non-Interventional Studies with 3- or 6-Month Depot Formulations in Patients with Advanced Prostate Cancer.\n",
            "Logits:[[ 0.5663972  -0.3312261 ]\n",
            " [ 0.5830489  -0.34538233]]\n",
            "Softmax:tensor([[0.6422, 0.4148]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "We evaluated the efficacy and tolerability of 3- and 6-month leuprorelin acetate (LA) depot formulations (Eligard®, Astellas Pharma GmbH) in patients with advanced prostate cancer treated in routine clinical practice in Germany.\n",
            "Logits:[[ 0.585069  -0.3439844]]\n",
            "Softmax:tensor([[0.6365, 0.4175],\n",
            "        [0.6365, 0.4165],\n",
            "        [0.6352, 0.4212],\n",
            "        [0.6352, 0.4212],\n",
            "        [0.6352, 0.4212],\n",
            "        [0.6352, 0.4205],\n",
            "        [0.6360, 0.4199]], device='cuda:0')\n",
            "top_indices:tensor([[0, 4],\n",
            "        [1, 3],\n",
            "        [6, 2],\n",
            "        [5, 5],\n",
            "        [2, 6]], device='cuda:0')\n",
            "XenofilteR: computational deconvolution of mouse and human reads in tumor xenograft sequence data. Mouse xenografts from (patient-derived) tumors (PDX) or tumor cell lines are widely used as models to study various biological and preclinical aspects of cancer. However, analyses of their RNA and DNA profiles are challenging, because they comprise reads not only from the grafted human cancer but also from the murine host. The reads of murine origin result in false positives in mutation analysis of DNA samples and obscure gene expression levels when sequencing RNA. However, currently available algorithms are limited and improvements in accuracy and ease of use are necessary.RESULTS: We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. To assess the accuracy of XenofilteR, we generated sequence data by in silico mixing of mouse and human DNA sequence data. These analyses revealed that XenofilteR removes > 99.9% of sequence reads of mouse origin while retaining human sequences. This allowed for mutation analysis of xenograft samples with accurate variant allele frequencies, and retrieved all non-synonymous somatic tumor mutations.CONCLUSIONS: XenofilteR accurately dissects RNA and DNA sequences from mouse and human origin, thereby outperforming currently available tools. XenofilteR is open source and available at https://github.com/PeeperLab/XenofilteR . We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. RESULTS: We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. RESULTS\n",
            "\n",
            "We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome.\n",
            "Logits:[[ 0.5604196  -0.33310166]\n",
            " [ 0.5601018  -0.3371821 ]\n",
            " [ 0.5544778  -0.31797966]\n",
            " [ 0.5544778  -0.31797966]\n",
            " [ 0.5544778  -0.31797966]\n",
            " [ 0.5548007  -0.32055235]\n",
            " [ 0.55811006 -0.32323578]]\n",
            "Softmax:tensor([[0.6353, 0.4182],\n",
            "        [0.6328, 0.4193]], device='cuda:0')\n",
            "top_indices:tensor([[0, 1],\n",
            "        [1, 0]], device='cuda:0')\n",
            "Here, we show that the elephant genome encodes 20 copies of the tumor suppressor gene TP53 and that the increase in TP53 copy number occurred coincident with the evolution of large body sizes, the evolution of extreme sensitivity to genotoxic stress, and a hyperactive TP53 signaling pathway in the elephant (Proboscidean) lineage.  While humans have 1 copy (2 alleles) of TP53, African elephants have at least 20 copies (40 alleles), including 19 retrogenes (38 alleles) with evidence of transcriptional activity measured by reverse transcription polymerase chain reaction. \n",
            "Logits:[[ 0.554827   -0.33002272]\n",
            " [ 0.5442862  -0.32563323]]\n",
            "Softmax:tensor([[0.6371, 0.4120]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "MannKind Corporation has developed a powder formulation of insulin that allows for a high percentage of the administered insulin to be absorbed via the lung. Their product, AFREZZA (Technosphere insulin), is currently under review by the FDA for use in patients with diabetes. \n",
            "Logits:[[ 0.5628218  -0.35551354]]\n",
            "Softmax:tensor([[0.6327, 0.4199],\n",
            "        [0.6399, 0.4163],\n",
            "        [0.6470, 0.4166],\n",
            "        [0.6431, 0.4179],\n",
            "        [0.6486, 0.4139],\n",
            "        [0.6411, 0.4163],\n",
            "        [0.6472, 0.4144],\n",
            "        [0.6384, 0.4184],\n",
            "        [0.6470, 0.4166],\n",
            "        [0.6411, 0.4163],\n",
            "        [0.6472, 0.4144],\n",
            "        [0.6431, 0.4179],\n",
            "        [0.6486, 0.4139],\n",
            "        [0.6406, 0.4171],\n",
            "        [0.6472, 0.4144],\n",
            "        [0.6486, 0.4139],\n",
            "        [0.6384, 0.4184],\n",
            "        [0.6431, 0.4179],\n",
            "        [0.6486, 0.4139],\n",
            "        [0.6411, 0.4163],\n",
            "        [0.6472, 0.4144]], device='cuda:0')\n",
            "top_indices:tensor([[12,  0],\n",
            "        [ 4, 16],\n",
            "        [15,  7],\n",
            "        [18, 11],\n",
            "        [ 6,  3]], device='cuda:0')\n",
            "Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. DeepSynergy significantly outperformed the other methods with an improvement of 7.2% over the second best method at the prediction of novel drug combinations within the space of explored drugs and cell lines.\n",
            "Logits:[[ 0.54379797 -0.3230034 ]\n",
            " [ 0.57489985 -0.33813825]\n",
            " [ 0.60571814 -0.3365605 ]\n",
            " [ 0.58898664 -0.3314405 ]\n",
            " [ 0.612802   -0.34768608]\n",
            " [ 0.58002263 -0.3380181 ]\n",
            " [ 0.60680914 -0.34574154]\n",
            " [ 0.56858516 -0.32918876]\n",
            " [ 0.60571814 -0.3365605 ]\n",
            " [ 0.58002263 -0.3380181 ]\n",
            " [ 0.60680914 -0.34574154]\n",
            " [ 0.58898664 -0.3314405 ]\n",
            " [ 0.612802   -0.34768608]\n",
            " [ 0.57817316 -0.3348443 ]\n",
            " [ 0.60680914 -0.34574154]\n",
            " [ 0.612802   -0.34768608]\n",
            " [ 0.56858516 -0.32918876]\n",
            " [ 0.58898664 -0.3314405 ]\n",
            " [ 0.612802   -0.34768608]\n",
            " [ 0.58002263 -0.3380181 ]\n",
            " [ 0.60680914 -0.34574154]]\n",
            "Softmax:tensor([[0.6383, 0.4153],\n",
            "        [0.6350, 0.4151],\n",
            "        [0.6354, 0.4148],\n",
            "        [0.6383, 0.4153],\n",
            "        [0.6354, 0.4148],\n",
            "        [0.6383, 0.4153]], device='cuda:0')\n",
            "top_indices:tensor([[5, 5],\n",
            "        [3, 3],\n",
            "        [0, 0],\n",
            "        [4, 1],\n",
            "        [2, 2]], device='cuda:0')\n",
            "The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12). The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12). The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12).  INTERPRETATION Ozanezumab did not show efficacy compared with placebo in patients with ALS. INTERPRETATION\n",
            "\n",
            "Ozanezumab did not show efficacy compared with placebo in patients with ALS.\n",
            "Logits:[[ 0.5678295  -0.34220353]\n",
            " [ 0.55385995 -0.34296545]\n",
            " [ 0.55528635 -0.3442411 ]\n",
            " [ 0.5678295  -0.34220353]\n",
            " [ 0.55528635 -0.3442411 ]\n",
            " [ 0.5678295  -0.34220353]]\n",
            "Softmax:tensor([[0.6462, 0.4467],\n",
            "        [0.6333, 0.4169],\n",
            "        [0.6061, 0.4526],\n",
            "        [0.6333, 0.4169],\n",
            "        [0.6354, 0.4198],\n",
            "        [0.6333, 0.4169],\n",
            "        [0.6333, 0.4169],\n",
            "        [0.6519, 0.4313],\n",
            "        [0.5843, 0.4692]], device='cuda:0')\n",
            "top_indices:tensor([[7, 8],\n",
            "        [0, 2],\n",
            "        [4, 0],\n",
            "        [3, 7],\n",
            "        [1, 4]], device='cuda:0')\n",
            "At treatment end, no significant differences were seen for change in MADRS total score with TC-5214 versus placebo. At treatment end, no significant differences were seen for change in MADRS total score with TC-5214 versus placebo. Furthermore, there were no significant differences in any of the secondary endpoints.  No notable differences were observed between dexmecamylamine and placebo for any secondary end point. TC-5214 (dexmecamylamine) is a nicotinic channel modulator that has previously been evaluated for treatment of major depression disorder (MDD) and is currently being evaluated by Targacept as a treatment for overactive bladder. TC-5214 (dexmecamylamine) is a nicotinic channel modulator that has previously been evaluated for treatment of major depression disorder (MDD) and is currently being evaluated by Targacept as a treatment for overactive bladder. \n",
            "Logits:[[ 0.60233885 -0.21411355]\n",
            " [ 0.54652715 -0.33564442]\n",
            " [ 0.43097177 -0.19019215]\n",
            " [ 0.54652715 -0.33564442]\n",
            " [ 0.55524546 -0.32360235]\n",
            " [ 0.54652715 -0.33564442]\n",
            " [ 0.54652715 -0.33564442]\n",
            " [ 0.62730366 -0.27662304]\n",
            " [ 0.34030774 -0.1235389 ]]\n",
            "Softmax:tensor([[0.6368, 0.4200],\n",
            "        [0.6413, 0.4168],\n",
            "        [0.6373, 0.4224],\n",
            "        [0.6417, 0.4166],\n",
            "        [0.6426, 0.4153],\n",
            "        [0.6360, 0.4197],\n",
            "        [0.6406, 0.4173],\n",
            "        [0.6356, 0.4231],\n",
            "        [0.6412, 0.4154],\n",
            "        [0.6417, 0.4166],\n",
            "        [0.6373, 0.4224],\n",
            "        [0.6426, 0.4153],\n",
            "        [0.6368, 0.4106]], device='cuda:0')\n",
            "top_indices:tensor([[ 4,  7],\n",
            "        [11, 10],\n",
            "        [ 9,  2],\n",
            "        [ 3,  0],\n",
            "        [ 1,  5]], device='cuda:0')\n",
            "easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. The ever-growing availability of high-quality genotypes for a multitude of species has enabled researchers to explore the underlying genetic architecture of complex phenotypes at an unprecedented level of detail using genome-wide association studies (GWAS). The systematic comparison of results obtained from GWAS of different traits opens up new possibilities, including the analysis of pleiotropic effects. Other advantages that result from the integration of multiple GWAS are the ability to replicate GWAS signals and to increase statistical power to detect such signals through meta-analyses. In order to facilitate the simple comparison of GWAS results, we present easyGWAS, a powerful, species-independent online resource for computing, storing, sharing, annotating, and comparing GWAS. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. We demonstrate the potential of easyGWAS with a case study of the model organism Arabidopsis thaliana, using flowering and growth-related traits.\n",
            "Logits:[[ 0.5614632  -0.32260108]\n",
            " [ 0.5809323  -0.33590227]\n",
            " [ 0.56367606 -0.31278664]\n",
            " [ 0.58255696 -0.3366149 ]\n",
            " [ 0.5866623  -0.34216428]\n",
            " [ 0.55796427 -0.32416362]\n",
            " [ 0.57812476 -0.3337101 ]\n",
            " [ 0.55626184 -0.30988482]\n",
            " [ 0.5805531  -0.34149715]\n",
            " [ 0.58255696 -0.3366149 ]\n",
            " [ 0.56367606 -0.31278664]\n",
            " [ 0.5866623  -0.34216428]\n",
            " [ 0.56162435 -0.36152455]]\n",
            "Softmax:tensor([[0.6369, 0.4177],\n",
            "        [0.6384, 0.4174],\n",
            "        [0.6359, 0.4170]], device='cuda:0')\n",
            "top_indices:tensor([[1, 0],\n",
            "        [0, 1],\n",
            "        [2, 2]], device='cuda:0')\n",
            "Data suggesting this vaccine may improve OS for SS and MRCL patients is exciting but early, and on-going work is testing the impact of CMB305 on patient outcomes. CMB305 induces NY-ESO-1 specific T cell responses in both SS and MRC patients and these patients had excellent overall survival (OS) outcomes in the initial phase I study. The potential of the CMB305 vaccine regimen to target NY-ESO-1 and improve outcomes for synovial sarcoma and myxoid/round cell liposarcoma patients.\n",
            "Logits:[[ 0.56211686 -0.33215985]\n",
            " [ 0.5685988  -0.33335656]\n",
            " [ 0.55748034 -0.33508855]]\n",
            "Softmax:tensor([[0.6365, 0.4174],\n",
            "        [0.6367, 0.4188]], device='cuda:0')\n",
            "top_indices:tensor([[1, 1],\n",
            "        [0, 0]], device='cuda:0')\n",
            " In a randomized, controlled, phase 3, pivotal trial, Quadracel proved to be as efficacious and safe as Daptacel (diphtheria and tetanus toxoids and acellular pertussis vaccine adsorbed, Sanofi Pasteur Inc.) and IPOL (poliovirus vaccine inactivated, Sanofi Pasteur Inc.), given separately, to children between the ages of 4 and 6 years Quadracel (diphtheria and tetanus toxoids and acellular pertussis adsorbed and inactivated poliovirus vaccine, Sanofi Pasteur Inc.) is a new vaccination developed to condense the last dose of both DTaP and IPV so they do not have to be given separately, thus reducing the total number of vaccinations required.\n",
            "Logits:[[ 0.5602998 -0.3335134]\n",
            " [ 0.561188  -0.3278364]]\n",
            "Softmax:tensor([[0.6363, 0.4123]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "On May 9 2011, the US FDA approved Sanofi Pasteur's Fluzone(®) Intradermal influenza vaccine, the first influenza vaccine licensed in the USA that uses a new microinjection system for intradermal delivery of vaccines (Soluvia™, Becton Dickinson). \n",
            "Logits:[[ 0.5591409  -0.35430518]]\n",
            "Softmax:tensor([[0.6376, 0.4127]], device='cuda:0')\n",
            "top_indices:tensor([[0, 0]], device='cuda:0')\n",
            "Despite discontinuation of the first inhalable insulin, Exubera®, due to suboptimal market acceptance, development of orally inhaled insulin delivery systems has been galvanized by the recent approval of Afrezza® and several others awaiting approval.\n",
            "Logits:[[ 0.56477183 -0.35281992]]\n",
            "Softmax:tensor([[0.6314, 0.4221],\n",
            "        [0.6312, 0.4225],\n",
            "        [0.6535, 0.3957],\n",
            "        [0.6314, 0.4188],\n",
            "        [0.6783, 0.3790],\n",
            "        [0.6345, 0.4191]], device='cuda:0')\n",
            "top_indices:tensor([[4, 1],\n",
            "        [2, 0],\n",
            "        [5, 5],\n",
            "        [0, 3],\n",
            "        [3, 2]], device='cuda:0')\n",
            "These results strongly suggest that recombination is the primary determinant of the isochore organization of mammalian genomes. Thus, chromosomal position can directly affect the rate of evolution of a gene. Such an accelerated rate of evolution might be due to factors other than natural selection, in particular GC-biased gene conversion. This is true of neutral sequences, but also of constrained sequences, which can be illustrated using the mouse Fxy gen We have previously described a gene, Fxy , that spans the pseudoautosomal boundary in mice such that the first three exons of the gene are located on the X chromosome, but the remainder of the gene is located on both X and Y chromosomes. Therefore, this gene might be in a state of transition between pseudoautosomal and X-unique locations.  The mouse Fxy gene was translocated into the highly recombining pseudoautosomal region comparatively recently in evolutionary terms. This event resulted in a rapid increase of GC content.\n",
            "Logits:[[ 0.53830343 -0.31408814]\n",
            " [ 0.53740275 -0.3124816 ]\n",
            " [ 0.63455665 -0.4233466 ]\n",
            " [ 0.5382272  -0.3275702 ]\n",
            " [ 0.7457602  -0.4939097 ]\n",
            " [ 0.5515263  -0.32633826]]\n",
            "Softmax:tensor([[0.6378, 0.4190],\n",
            "        [0.6445, 0.4099],\n",
            "        [0.6394, 0.4182],\n",
            "        [0.6405, 0.4214],\n",
            "        [0.6339, 0.4204],\n",
            "        [0.6417, 0.4206],\n",
            "        [0.6276, 0.4237]], device='cuda:0')\n",
            "top_indices:tensor([[1, 6],\n",
            "        [5, 3],\n",
            "        [3, 5],\n",
            "        [2, 4],\n",
            "        [0, 0]], device='cuda:0')\n",
            "The insertion is generated by non-allelic homologous recombination between a 548 bp LTR6B repeat within the Y chromosome PAR1 and a second LTR6B repeat located 105 kb from the PAR boundary on the X chromosome.   The four genes within PAR2 have been proposed to constitute two zones with different base ratios and transcription, one of which was added recently to the X chromosome. In contrast, so far only 4 genes have been discovered in the PAR2 region. The pseudoautosomal regions (PAR1 and PAR2) of the human X and Y chromosomes pair and recombine during meiosis. Thus genes in this region are not inherited in a strictly sex-linked fashion. PAR1 is located at the terminal region of the short arms and PAR2 at the tips of the long arms of these chromosomes. To date, 24 genes have been assigned to the PAR1 region. The human sex chromosomes differ in sequence, except for the pseudoautosomal regions (PAR) at the terminus of the short and the long arms, denoted as PAR1 and PAR2. The boundary between PAR1 and the unique X and Y sequences was established during the divergence of the great apes. \n",
            "Logits:[[ 0.56581587 -0.32707986]\n",
            " [ 0.5948975  -0.36455768]\n",
            " [ 0.5729293  -0.33016947]\n",
            " [ 0.5774761  -0.31688234]\n",
            " [ 0.5487866  -0.32102183]\n",
            " [ 0.5825881  -0.32050064]\n",
            " [ 0.52205575 -0.30758154]]\n",
            "summary:[' Here, we present evidence that the calcium/calmodulin-dependent protein kinase IV (CaMK4) is increased and required during Th17 cell differentiation. CaMK4 inhibition has potential as a therapeutic strategy for Th17-driven autoimmune diseases. CaMK4-dependent activation of AKT/mTOR and CREM-α underlies autoimmunity-associated Th17 imbalance CaMK4 inhibition significantly decreased the levels of glycolytic intermediates such as glucose-6-phosphate, fructose-6-phosphate, fructose-1,6-diphosphate, pyruvate, and lactate, whereas it did not affect the levels of the pentose phosphate pathway intermediates such as 6-Phospho-D-gluconate, ribulose-5-phosphate, ribose-5-phosphate and phosphoribosyl pyrophosphate Inhibition of CaMK4 reduced Il17 transcription through decreased activation of the cAMP response element modulator α (CREM-α) and reduced activation of the AKT/mTOR pathway, which is known to enhance Th17 differentiation.', 'A new study identifies homozygous missense mutations in SGOL1, which encodes a component of the cohesin complex, in a newly described disorder termed Chronic Atrial and Intestinal Dysrhythmia (CAID) syndrome. These findings implicate cohesin in the regulation of intrinsic cardiac and intestinal rhythm and further expand the growing group of disorders termed the cohesinopathies. Genetic mapping studies reveal that mutations in cohesion pathways are responsible for multispectrum developmental abnormalities termed cohesinopathies. These include Roberts syndrome (RBS), Cornelia de Lange Syndrome (CdLS), and Warsaw Breakage Syndrome (WABS). The cohesinopathies are characterized by overlapping phenotypes ranging from craniofacial deformities, limb defects, and mental retardation. Though these syndromes share a similar suite of phenotypes and arise due to mutations in a common cohesion pathway, the underlying mechanisms are currently believed to be distinct. Defects in mitotic failure and apoptosis i.e. trans DNA tethering events are believed to be the underlying cause of RBS, whereas the underlying cause of CdLS is largely modeled as occurring through defects in transcriptional processes i.e. cis DNA tethering events. Here, we review recent findings described primarily in zebrafish, paired with additional studies in other model systems, including human patient cells, which challenge the notion that cohesinopathies represent separate syndromes. ', 'All patients were administered intravenous camrelizumab (SHR-1210), a novel anti-PD-1 antibody, at doses of 60 mg, 200 mg, or 400 mg (4-week interval after first dose followed by a 2-week schedule) until disease progression or intolerable toxicity. Recent phase 1-2 trials reported manageable safety profiles and promising antitumor activities of anti-PD-1 drugs (pembrolizumab, nivolumab, camrelizumab and JS001) with/without chemotherapy in recurrent/metastatic nasopharyngeal carcinoma (RM-NPC), however head-to-head comparison among these regimens is lacking. Recent phase 1-2 trials reported manageable safety profiles and promising antitumor activities of anti-PD-1 drugs (pembrolizumab, nivolumab, camrelizumab and JS001) with/without chemotherapy in recurrent/metastatic nasopharyngeal carcinoma (RM-NPC), however head-to-head comparison among these regimens is lacking. The PD-1 antibody camrelizumab was well tolerated in patients with nasopharyngeal carcinoma. Camrelizumab (SHR-1210) is a humanised anti-programmed death-1 (anti PD-1) antibody.', ' an erythropoietic protoporphyria (EPP)-associated ferrochelatase (FECH) mutation  Erythropoietic protoporphyria (EPP) is a rare inherited disorder of the heme biosynthesis pathway resulting in the accumulation of protoporphyrins in the blood, erythrocytes, and other tissues. Because of a gene mutation in the FECH gene, ferrochelatase, the enzyme involved in the final step of heme synthesis, is deficient in these patients.   Erythropoietic protoporphyria (EPP) is the most common inherited porphyria in children and is diagnosed in most individuals after the onset of cutaneous manifestations.', 'There were no significant differences in survival in a non-small cell lung cancer prinomastat study, and in a small cell lung cancer marimastat trial. There were no significant differences in survival in a non-small cell lung cancer prinomastat study , and in a small cell lung cancer marimastat trial.  CONCLUSION: Treatment with marimastat after induction therapy for SCLC did not result in improved survival and had a negative impact on quality of life. The phase III trial in small cell lung cancer was discontinued when the results of study 140 were released in February 2001 showing that marimastat was not significantly more effective than placebo in prolonging the survival of small cell lung cancer patients. The phase III trial in small cell lung cancer was discontinued when the results of study 140 were released in February 2001 showing that marimastat was not significantly more effective than placebo in prolonging the survival of small cell lung cancer patients.', \"No transmission distortion was observed for TNFR2-196R allele.   In conclusion, the TNFR2 196R allele was found to be significantly associated with the susceptibility to SLE in the Japanese population. Further population and functional studies will be of particular importance to establish TNFR2 as one of the susceptibility genes to SLE. We therefore characterized the frequency of a genetic polymorphism in the 3' untranslated region of the TNFR2 gene in Caucasoid SLE patients and geographically matched controls. Lack of association between the Met196Arg polymorphism in the TNFR2 gene and autoimmune diseases accompanied by vasculitis including SLE in Japanese. A TNFR2 3' flanking region polymorphism in systemic lupus erythematosus.\", 'The possibility of integrating viral vectors to become a persistent part of the host genome makes them a crucial element of clinical gene therapy. However, viral integration has associated risks, such as the unintentional activation of oncogenes that can result in cancer. Therefore, the analysis of integration sites of retroviral vectors is a crucial step in developing safer vectors for therapeutic use.RESULTS: Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. VISMapper can be found at: http://vismapper.babelomics.org .CONCLUSIONS: Because it uses novel mapping algorithms VISMapper is remarkably faster than previous available programs. It also provides a useful graphical interface to analyze the integration sites found in the genomic context. Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS: Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS Here we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites. RESULTS\\n\\nHere we present VISMapper, a vector integration site analysis web server, to analyze next-generation sequencing data for retroviral vector integration sites.', 'Tauopathies are a group of neurodegenerative disorders with accumulation of three-repeat (3R) or four-repeat (4R) Tau CTE is defined as a tauopathy with a distribution of tau-positive neurofibrillary tangles (NFTs) that is distinct from other tauopathies,', 'Two phase III trials of the anti-IGF-1R monoclonal antibody , figitumumab ( CP-751,871) , were discontinued in 2010 as it was considered unlikely either trial would meet their primary endpoints .  Two phase III trials of the anti-IGF-1R monoclonal antibody, figitumumab (CP-751,871), were discontinued in 2010 as it was considered unlikely either trial would meet their primary endpoints. In light of disappointing clinical data with figitumumab and other targeted agents, it is likely that the use of molecular markers will become important in predicting response to treatment.  Median OS was 8.6 months for figitumumab plus chemotherapy and 9.8 months for chemotherapy alone (hazard ratio [HR], 1.18; 95% CI, 0.99 to 1.40; P = .06); median progression-free survival was 4.7 months (95% CI, 4.2 to 5.4) and 4.6 months (95% CI, 4.2 to 5.4), respectively (HR, 1.10; P = .27); the objective response rates were 33% and 35%, respectively. Median OS was 8.6 months for figitumumab plus chemotherapy and 9.8 months for chemotherapy alone (hazard ratio [HR], 1.18; 95% CI, 0.99 to 1.40; P = .06); median progression-free survival was 4.7 months (95% CI, 4.2 to 5.4) and 4.6 months (95% CI, 4.2 to 5.4), respectively (HR, 1.10; P = .27); the objective response rates were 33% and 35%, respectively.  The insulin-like growth factor receptor ( IGF-1R ) monoclonal antibody figitumumab , while initially promising , appears to increase toxicity and death in combination with chemotherapy in the treatment of patients with NSCLC of squamous histology; therefore , clinical development of this class of agents will need to proceed with caution . ', 'The Library of Integrated Cellular Signatures (LINCS) project provides comprehensive transcriptome profiling of human cell lines before and after chemical and genetic perturbations.  Recently, resources such as the Library of Integrated Network-Based Cellular Signatures (LINCS) L1000 database provide gene expression profiles induced by various chemical and genetic perturbations The library of integrated network-based cellular signatures (LINCS) L1000 data set currently comprises of over a million gene expression profiles of chemically perturbed human cell lines. The GE data is from the Library of Integrated Network-based Cellular Signatures (LINCS) L1000 dataset that measured changes in GE before and after treatment of human cells with over 20 000 small-molecule compounds including most of the FDA-approved drugs. The Library of Integrated Network-based Cellular Signatures (LINCS) L1000 big data provide gene expression profiles induced by over 10\\u2009000 compounds, shRNAs, and kinase inhibitors using the L1000 platform.', 'The cytokine storm is an aggressive immune response characterized by the recruitment of inflammatory leukocytes and exaggerated levels of cytokines and chemokines at the site of infection .  Endotoxins and exotoxins are among the most potent bacterial inducers of cytokines. During infectious processes, the production of inflammatory cytokines including tumor necrosis factor (TNF), interleukin-1β (IL-1β), gamma interferon (IFNγ) and chemokines orchestrates the anti-infectious innate immune response. However, an overzealous production, leading up to a cytokine storm, can be deleterious and contributes to mortality consecutive to sepsis or toxic shock syndrome Ebola virus (EBOV) disease (EVD) results from an exacerbated immunological response that is highlighted by a burst in the production of inflammatory mediators known as a \"cytokine storm.\" undesirable elevation of cytokine levels, as may occur in response to a drug or a device, may lead to severe side effects such as systemic inflammatory response syndrome or cytokine storm. Cytokine storm defines a dysregulation of and an excessively exaggerated immune response most often accompanying selected viral infections and several autoimmune diseases.', \"SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. SNP2TFBS is a computational resource intended to support researchers investigating the molecular mechanisms underlying regulatory variation in the human genome. The database essentially consists of a collection of text files providing specific annotations for human single nucleotide polymorphisms (SNPs), namely whether they are predicted to abolish, create or change the affinity of one or several transcription factor (TF) binding sites. A SNP's effect on TF binding is estimated based on a position weight matrix (PWM) model for the binding specificity of the corresponding factor. These data files are regenerated at regular intervals by an automatic procedure that takes as input a reference genome, a comprehensive SNP catalogue and a collection of PWMs. SNP2TFBS is also accessible over a web interface, enabling users to view the information provided for an individual SNP, to extract SNPs based on various search criteria, to annotate uploaded sets of SNPs or to display statistics about the frequencies of binding sites affected by selected SNPs. Homepage: http://ccg.vital-it.ch/snp2tfbs/.\", 'To address this possible bias, phosphopeptides quantified in both enriched and non-enriched samples can be used to calculate factors that mitigate the bias.Results: We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics.Availability and implementation: The phosphonormalizer package is freely available under GPL (\\u2009>\\u2009=2) license from Bioconductor (https://bioconductor.org/packages/phosphonormalizer).Contact: sohrab.saraei@utu.fi or laura.elo@utu.fi.Supplementary information: Supplementary data are available at Bioinformatics online. Results\\nWe present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Results\\n\\nWe present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Results We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics. Global centering-based normalization is a commonly used normalization approach in mass spectrometry-based label-free proteomics. It scales the peptide abundances to have the same median intensities, based on an assumption that the majority of abundances remain the same across the samples. However, especially in phosphoproteomics, this assumption can introduce bias, as the samples are enriched during sample preparation which can mask the underlying biological changes. To address this possible bias, phosphopeptides quantified in both enriched and non-enriched samples can be used to calculate factors that mitigate the bias.Results: We present an R package phosphonormalizer for normalizing enriched samples in label-free mass spectrometry-based phosphoproteomics.Availability and implementation: The phosphonormalizer package is freely available under GPL (\\u2009>\\u2009=2) license from Bioconductor (https://bioconductor.org/packages/phosphonormalizer).', 'So far, results, especially from oral immunotherapy studies, have shown good efficacy in achieving desensitization to peanut with a good safety profile. Currently, two forms of peanut immunotherapy, oral immunotherapy (OIT) and epicutaneous immunotherapy (EPIT), are in Phase III clinical trials and have shown promise to induce desensitization in many subjects This article presents an overview of potential treatments of food allergy, with an emphasis on various forms of immunotherapy (including oral immunotherapy, sublingual immunotherapy, epicutaneous immunotherapy, immunotherapy with modified food antigens, and immunotherapy with a recombinant peanut vaccine). Efforts have been made to develop a vaccine for peanut allergy. Recent advances in immunotherapy and vaccine development for peanut allergy.', 'Feline infectious peritonitis (FIP) is a common and highly lethal coronavirus disease of domestic cats Feline coronaviruses (FCoV) exist as 2 biotypes: feline enteric coronavirus (FECV) and feline infectious peritonitis virus (FIPV). Feline coronavirus (FCoV) infection is very common in cats, usually causing only mild intestinal signs such as diarrhoea. Up to 10% of FCoV infections, however, result in the fatal disease feline infectious peritonitis (FIP).  Feline infectious peritonitis (FIP) is one of the most important infectious diseases in cats and is caused by feline coronavirus (FCoV). Feline infectious peritonitis (FIP) is a fatal disease caused by feline coronavirus (FCoV) infection.', 'Macrophage activation syndrome (MAS) is a life-threatening hyperinflammatory state mediated by uncontrolled cytokine storm and haemophagocytosis. Reactive oxygen species (ROS) are signaling molecules responsible for the production of cytokines and chemokines that can mediate hyperactivation of the immune response called cytokine storm.  During infectious processes, the production of inflammatory cytokines including tumor necrosis factor (TNF), interleukin-1β (IL-1β), gamma interferon (IFNγ) and chemokines orchestrates the anti-infectious innate immune response. However, an overzealous production, leading up to a cytokine storm, can be deleterious and contributes to mortality consecutive to sepsis or toxic shock syndrome.  cytokine storm leading to severe organ damage', 'Introduced monoclonal antibodies (infliximab, etanercept, adaluimumab, golimumab, rituximab), tested for efficacy in other pathologies associated with the formation of granulomas, have a limited application in patients with SA.  Although treatment was well tolerated, neither ustekinumab nor golimumab demonstrated efficacy in pulmonary sarcoidosis. Although treatment was well tolerated, neither ustekinumab nor golimumab demonstrated efficacy in pulmonary sarcoidosis. ', 'The viral spike glycoprotein (S) utilizes angiotensin-converting enzyme 2 (ACE2) as a host protein receptor and mediates fusion of the viral and host membranes, making S essential to viral entry into host cells and host species tropism. Cell entry studies demonstrated that three newly identified SARSr-CoVs with different S protein sequences are all able to use human ACE2 as the receptor Angiotensin-converting enzyme 2 (ACE2), a relatively new member of the RAS, has drawn extensive attention since 2003, because of the findings that ACE2 is the receptor for SARS Corona virus and that maintenance of normal ACE2 levels in the lung is beneficial for the host to combat inflammatory lung disease. The trimeric SARS coronavirus (SARS-CoV) surface spike (S) glycoprotein consisting of three S1-S2 heterodimers binds the cellular receptor angiotensin-converting enzyme 2 (ACE2) and mediates fusion of the viral and cellular membranes through a pre- to postfusion conformation transition The infection of target cells by the SARS CoV is mediated through the interaction of the viral Spike (S) protein (1255 amino acids) and its cellular receptor, angiotensin-converting enzyme 2 (ACE2).', \"On May 9 2011, the US FDA approved Sanofi Pasteur's Fluzone(®) Intradermal influenza vaccine, the first influenza vaccine licensed in the USA that uses a new microinjection system for intradermal delivery of vaccines (Soluvia™, Becton Dickinson). \", 'Whilst intact in all FECVs, the 3c gene was mutated in the majority (71.4 %) of FIPVs, but not in all, implying that mutation in 3c is not the (single) cause of FIP. Feline enteric coronavirus (FECV) causes inapparent or mild enteritis in cats, but a highly fatal disease, called feline infectious peritonitis (FIP), can arise through mutation of FECV to FIP virus (FIPV). Feline infectious peritonitis virus (FIPV) was presumed to arise from mutations in the 3c of a ubiquitous and largely nonpathogenic feline enteric coronavirus (FECV). Feline infectious peritonitis virus (FIPV) was presumed to arise from mutations in the 3c of a ubiquitous and largely nonpathogenic feline enteric coronavirus (FECV). Feline infectious peritonitis (FIP) is a lethal systemic disease caused by FIP virus (FIPV), a virulent mutant of apathogenic feline enteric coronavirus (FECV).', ' Its onset preceded the rearrangement of TCR genes, as Xlr expression was conserved in thymus cells from RAG1(0/0) mice.  Our results provide a novel framework for understanding ASE function and demonstrate a novel role for SATB1 as a regulator of Rag locus organization and gene expression in DP thymocytes. High level expression of the Xlr nuclear protein in immature thymocytes and colocalization with the matrix-associated region-binding SATB1 protein An anti-silencer- and SATB1-dependent chromatin hub regulates Rag1 and Rag2 gene expression during thymocyte development.  SATB1 binds to the ASE and Rag promoters, facilitating inclusion of Rag2 in the chromatin hub and the loading of RNA polymerase II to both the Rag1 and Rag2 promoters.', ' Like normal nascent chromatin, chromatin labeled for brief periods (0.5-1 min) in the presence of butyrate was more sensitive to digestion with DNase I and micrococcal nuclease than control bulk chromatin. Incubation of mature chromatin in butyrate for 1 h did not induce DNase I sensitivity: therefore, the presence of sodium butyrate was required during replication to preserve the increased digestibility of nascent chromatin DNA this class of nascent chromatin exhibits a shortened repeat length of approximately 165 bp, as opposed to the 288-bp repeat of bulk chromatin. Within 10 min of DNA synthesis, the spacing of mature chromatin is established; the spacing maturation can occur in the absence of protein synthesis. The second class of nascent DNA is distinguished from the nucleosomal component by its insolubility, lack of discernible nucleosomal organization, and dependence on protein synthesis to attain typical subunit structure In a previous study (Perry and Annunziato, Nucleic Acids Res. 17, 4275 [1989]) it was shown that when replication occurs in the presence of sodium butyrate (thereby inhibiting histone deacetylation), nascent chromatin fails to mature fully and instead remains preferentially sensitive to DNaseI, more soluble in magnesium, and depleted of histone H1 (relative to mature chromatin). Maturation of nucleosomal and nonnucleosomal components of nascent chromatin: differential requirements for concurrent protein synthesis.', \"Here, we show that promoter classes are significantly differentiated by nucleosome organization and chromatin structure. Dispersed promoters display higher associations with well-positioned nucleosomes downstream of the TSS and a more clearly defined nucleosome free region upstream, while focused promoters have a less organized nucleosome structure, yet higher presence of RNA polymerase II. Nucleosome depletion at 5'-HS4 was dependent on interaction of the insulator protein CCCTC-binding factor (CTCF) and was required for enhancer blocking.  hese differences extend to histone variants (H2A.Z) and marks (H3K4 methylation), as well as insulator binding (such as CTCF), independent of the expression levels of affected genes. This general architectural change correlates with enhanced binding of CTCF and cohesins and more pronounced insulation of contacts across chromatin boundaries in lineage-committed cells.  nucleosome occupancy at nucleosome-free regions (NFRs), many of which are located at sites occupied by the multivalent factors Ctcf and cohesin. \", 'The proteasome inhibitor bortezomib (also known as Velcade and PS-341) is a clinically effective antineoplastic drug that is FDA approved for treatment of hematologic malignancies such as multiple myeloma and mantle cell lymphoma. Bortezomib (Velcade™) is a reversible proteasome inhibitor that is approved for the treatment of multiple myeloma (MM). The proteasome inhibitor bortezomib is effective for a variety of tumors, but not for GBM.  The proteasome inhibitor bortezomib, registered for Multiple Myeloma treatment, is currently explored for activity in solid tumors including non-small cell lung cancer (NSCLC). The proteasome inhibitor Bortezomib is used to treat multiple myeloma (MM).', ' We found that although the post-SET zinc finger and the KRAB domains are not essential for the methyltransferase activity of PRDM9 in cell culture, the KRAB domain mutant mice show only residual PRDM9 methyltransferase activity and undergo meiotic arrest. In aggregate, our data indicate that domains typically involved in regulation of gene expression do not serve that role in PRDM9, but are likely involved in setting the proper chromatin environment for initiation and completion of homologous recombination. In many eukaryotes, sites of meiotic recombination, also called hotspots, are regions of accessible chromatin, but in many vertebrates, their location follows a distinct pattern and is specified by PR domain-containing protein 9 (PRDM9).  PRDM9 (PR domain-containing protein 9) is a meiosis-specific protein that trimethylates H3K4 and controls the activation of recombination hot spots. It is an essential enzyme in the progression of early meiotic prophase. Our findings do not identify the nature of the underlying DNA sequences, but argue against the proposed role of Prdm9 as an essential transcription factor in mouse meiosis PRDM9 Methyltransferase Activity Is Essential for Meiotic DNA Double-Strand Break Formation at Its Binding Sites.', \"The gene Fxy (also known as MID1 [7]) spans the pseudoautosomal boundary (PAB) in the laboratory mouse (Mus musculus domesticus, C57BL/6) such that the 5' three exons of the gene are located on the X chromosome but the seven exons encoding the carboxy-terminal two-thirds of the protein are located within the PAR and are therefore present on both the X and Y chromosomes The Fxy gene in mice is also located on the X chromosome but spans the pseudoautosomal boundary in this species. Here we describe a gene closely related to FXY/MID1, called FXY2, which also maps to the X chromosome within Xq22 We have previously described a gene, Fxy , that spans the pseudoautosomal boundary in mice such that the first three exons of the gene are located on the X chromosome, but the remainder of the gene is located on both X and Y chromosomes. \", 'RESULTS: At the end of the study, after 547 days of follow-up, 103 patients (51.7%) in the pentoxifylline group and 120 (59.7%) in the placebo group were alive (unadjusted risk 1.28, p = 0.107; adjusted risk 1.43, p = 0.02). .RESULTS: At the end of the study, after 547 days of follow-up, 103 patients (51.7%) in the pentoxifylline group and 120 (59.7%) in the placebo group were alive (unadjusted risk 1.28, p = 0.107; adjusted risk 1.43, p = 0.02). In contrast, analysis of secondary outcome functional variables did not show the same negative effect of the drug.  CONCLUSIONS: Pentoxifylline is not beneficial in ALS and should be avoided in patients treated with riluzole. ', 'Mapping brain metabolic connectivity in awake rats with μPET and optogenetic stimulation. PURPOSE\\nPositron emission tomography (PET) is a useful imaging modality that quantifies the physiological distributions of radiolabeled tracers in\\xa0vivo in humans and animals. PURPOSE\\n\\nPositron emission tomography (PET) is a useful imaging modality that quantifies the physiological distributions of radiolabeled tracers in\\xa0vivo in humans and animals. Positron emission tomography (PET) with [(18)F]2-fluoro-2-deoxy-D-glucose was used to measure changes in regional brain glucose metabolism Positron emission tomography (PET) allows the quantitative measurement of regional cerebral flow (rCBF) in humans in quantitative terms.', 'A significant association was observed in the discovery cohorts for the single-nucleotide polymorphism (SNP) rs17518584 (discovery P-value=3.12 × 10(-8)) and in the joint discovery and replication meta-analysis (P-value=3.28 × 10(-9) after adjustment for age, gender and education) in an intron of the gene cell adhesion molecule 2 (CADM2) for performance on the LDST/DSST. Rs17518584 is located about 170 kb upstream of the transcription start site of the major transcript for the CADM2 gene, but is within an intron of a variant transcript that includes an alternative first exon. The variant is associated with expression of CADM2 in the cingulate cortex (P-value=4 × 10(-4)). The protein encoded by CADM2 is involved in glutamate signaling (P-value=7.22 × 10(-15)), gamma-aminobutyric acid (GABA) transport (P-value=1.36 × 10(-11)) and neuron cell-cell adhesion (P-value=1.48 × 10(-13)). Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. Our findings suggest that genetic variation in the CADM2 gene is associated with individual differences in information processing speed. The CADM2 gene is associated with processing speed performance - evidence among elderly with type 2 diabetes. GWAS for executive function and processing speed suggests involvement of the CADM2 gene.', 'The exponential reduction in cost of genome sequencing has resulted in a rapid growth of genomic data. Most of the entropy of short read data lies not in the sequence of read bases themselves but in their Quality Scores-the confidence measurement that each base has been sequenced correctly. Lossless compression methods are now close to their theoretical limits and hence there is a need for lossy methods that further reduce the complexity of these data without impacting downstream analyses.RESULTS: We here propose GeneCodeq, a Bayesian method inspired by coding theory for adjusting quality scores to improve the compressibility of quality scores without adversely impacting genotyping accuracy. Our model leverages a corpus of k-mers to reduce the entropy of the quality scores and thereby the compressibility of these data (in FASTQ or SAM/BAM/CRAM files), resulting in compression ratios that significantly exceeds those of other methods. Our approach can also be combined with existing lossy compression schemes to further reduce entropy and allows the user to specify a reference panel of expected sequence variations to improve the model accuracy. In addition to extensive empirical evaluation, we also derive novel theoretical insights that explain the empirical performance and pitfalls of corpus-based quality score compression schemes in general. Finally, we show that as a positive side effect of compression, the model can lead to improved genotyping accuracy.AVAILABILITY AND IMPLEMENTATION: GeneCodeq is available at: github.com/genecodeq/eval GeneCodeq: quality score compression and improved genotyping using a Bayesian framework. GeneCodeq: quality score compression and improved genotyping using a Bayesian framework', 'programmed cell death protein 1 (PD-1)-blocking antibodies nivolumab or pembrolizumab  An improvement in the understanding of the role of the immune system in tumor immunosurveillance has led to the development of the programmed death-1 ( PD-1 ) immune checkpoint inhibitor nivolumab ( Opdivo) .  Nivolumab (Opdivo(®); Nivolumab BMS™) was the first programmed death (PD)-1 immune checkpoint inhibitor to be approved for use in advanced, squamous non-small cell lung cancer (NSCLC) following prior chemotherapy. Fatal Myocarditis Following Treatment with the PD-1 Inhibitor Nivolumab  PD-1 inhibitor nivolumab (Opdivo)', 'FINDINGS: ALSFRS-R score deterioration was faster in the minocycline group than in the placebo group (-1.30 vs -1.04 units/month, 95% CI for difference -0.44 to -0.08; p=0.005). Patients on minocycline also had non-significant tendencies towards faster decline in FVC (-3.48 vs -3.01, -1.03 to 0.11; p=0.11) and MMT score (-0.30 vs -0.26, -0.08 to 0.01; p=0.11), and greater mortality during the 9-month treatment phase (hazard ratio=1.32, 95% CI 0.83 to 2.10; p=0.23) than did patients on placebo. Quality-of-life scores did not differ between the treatment groups.  Patients on minocycline also had non-significant tendencies towards faster decline in FVC (-3.48 vs -3.01, -1.03 to 0.11; p=0.11) and MMT score (-0.30 vs -0.26, -0.08 to 0.01; p=0.11), and greater mortality during the 9-month treatment phase (hazard ratio=1.32, 95% CI 0.83 to 2.10; p=0.23) than did patients on placebo. This pilot study shows that minocycline and riluzole can be taken safely together. Further trials are needed to assess efficacy of such treatment. A recent publication of the results of a clinical trial of minocycline in 412 ALS patient has aroused considerable controversy in the ALS scientific community. As on previous occasions, the results obtained in the laboratory are not reproduced in clinical practice. A recent publication of the results of a clinical trial of minocycline in 412 ALS patient has aroused considerable controversy in the ALS scientific community.', 'Extracellular vesicles (EVs) are membrane-bound vesicles released into the extracellular space by almost all types of cells. EVs can cross the physiological barriers, and a variety of biological fluids are enriched in them. EVs are a heterogeneous population of vesicles, including exosomes, microvesicles, and apoptotic bodies.  Extracellular vesicles are cell-derived membrane particles ranging from 30 to 5,000\\xa0nm in size, including exosomes, microvesicles, and apoptotic bodies. They are released under physiological conditions, but also upon cellular activation, senescence, and apoptosis. They play an important role in intercellular communication.  Extracellular vesicles (EVs), microparticles, exosomes, and apoptotic bodies, originated by different cell types are emerging as a novel mean of cell-to-cell communication in physiology and pathology and represent a new way to convey fundamental information between cells.   In treated cells, the apoptotic hallmarks such as formation of apoptotic bodies,  Apoptotic bodies are generated on apoptotic cell shrinkage and death.', 'Despite discontinuation of the first inhalable insulin, Exubera®, due to suboptimal market acceptance, development of orally inhaled insulin delivery systems has been galvanized by the recent approval of Afrezza® and several others awaiting approval.', 'In vitro, the three known yeast nucleosome spacing enzymes (CHD1, ISW1 and ISW2) form arrays with different spacing. Isw1b acts in conjunction with Chd1 to regulate chromatin structure by preventing trans-histone exchange from taking place over coding regions. In this way, Isw1b and Chd1 are important in maintaining chromatin integrity during transcription elongation by RNAPII. We propose that CHD1 directs short spacing, resulting in eviction of H1 and chromatin unfolding, whereas ISW1 directs longer spacing, allowing H1 to bind and condense the chromatin CHD1 and ISW1 compete to set the spacing on most genes, such that CHD1 dominates genes with shorter spacing and ISW1 dominates genes with longer spacing The ISW1 and CHD1 ATP-dependent chromatin remodelers compete to set nucleosome spacing in vivo', 'Gamma-ray irradiation introduces single and/or double strand breaks into the DNA molecule of the cells. RESULTS BRCA2-defective cells were unable to repair the double-strand DNA breaks induced by ionizing radiation. Ionizing radiation and radiomimetic drugs such as bleomycin, calichieamycin, neocarzinostatin chromophore, and other synthetic agents can produce both single and double strand breaks in DNA. BACKGROUND Induction of DNA double strand breaks and alterations in the repair of these breaks is implicated in breast carcinogenesis. The induction of DNA interstrand cross-links by ionizing radiation has been largely ignored in favour of studies on double-strand break formation and repair.', ' This study provides strong evidence of positive associations between protracted low-dose radiation exposure and leukaemia.FU e highest risk of radiation- induced breast cancer is evidenced in the sub-population of female patients who have undergone radiotherapy for either malignant or non-malignant diseases Because latency period for different nosological forms of radiation-induced malignant tumors varies widely, profound attention in further studies should be drawn not only to thyroid, breast cancers and leukemia, but also to malignancies with longer latent period: lung, stomach, colon, ovary, urinary bladder, kidney cancer and multiple myeloma. Breast cancer ( BC ) is the most common type of malignancy in female patients and radio-treatment is the conventional therapy even if a great number of studies reported that enhanced sensitivity to ionizing radiation as measured as chromosome effects is present in a significant proportion of cancer patients , including breast cancer ones .  Ionizing radiation exposure is a risk factor for development of esophageal squamous cell carcinoma , a histological subtype of esophageal cancer that is highly aggressive and is associated with poor patient prognosis . ', 'KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies. Acute myeloid leukemia (AML) is an aggressive cancer with a poor prognosis, for which mainstream treatments have not changed for decades. To identify additional therapeutic targets in AML, we optimize a genome-wide clustered regularly interspaced short palindromic repeats (CRISPR) screening platform and use it to identify genetic vulnerabilities in AML cells. We identify 492 AML-specific cell-essential genes, including several established therapeutic targets such as DOT1L, BCL2, and MEN1, and many other genes including clinically actionable candidates. We validate selected genes using genetic and pharmacological inhibition, and chose KAT2A as a candidate for downstream study. KAT2A inhibition demonstrated anti-AML activity by inducing myeloid differentiation and apoptosis, and suppressed the growth of primary human AMLs of diverse genotypes while sparing normal hemopoietic stem-progenitor cells. Our results propose that KAT2A inhibition should be investigated as a therapeutic strategy in AML and provide a large number of genetic vulnerabilities of this leukemia that can be pursued in downstream studies.', 'BACKGROUND\\n\\nIdiopathic toe walking (ITW) is an exclusionary diagnosis given to healthy children who persist in walking on their toes after they should typically have achieved a heel-toe gait. BACKGROUND\\n\\nIdiopathic toe walking is a diagnosis of exclusion characterized by a persistent toe-toe gait pattern after three years of age. Idiopathic toe walking is a diagnosis of exclusion characterized by a persistent toe-toe gait pattern after three years of age.  Idiopathic toe walking (ITW) is a pathological gait pattern in which children walk on their tip toes with no orthopedic or neurological reason.  The diagnosis of idiopathic toe walking is a diagnosis of exclusion used for children with persistent toe walking and no associated medical condition . ', \"3'AmNic-rEPA recipients with the highest serum antinicotine Ab response (top 30% by area under the curve (AUC)) were significantly more likely than the placebo recipients (24.6% vs. 12.0%, P = 0.024, odds ratio (OR) = 2.69, 95% confidence interval (CI), 1.14-6.37) to attain 8 weeks of continuous abstinence from weeks 19 through 26.  FINDINGS: There was no difference in abstinence rates between NicVAX and placebo from weeks 9 to 52 [27.7 versus 30.0%, odds ratio (OR)\\u2009=\\u20090.89, 95% confidence interval (CI)\\u2009=\\u20090.62-1.29] or weeks 37 to 52 (33.8 versus 33.2%, OR\\u2009=\\u20091.03, 95% CI\\u2009=\\u20090.73-1.46). The top 30% antibody responders, compared to the placebo group, showed a non-significant tendency towards higher abstinence rates from weeks 37 to 52 (42.2 versus 33.2%, OR\\u2009=\\u20091.47, 95% CI\\u2009=\\u20090.89-2.42) FINDINGS\\nThere was no difference in abstinence rates between NicVAX and placebo from weeks 9 to 52 [27.7 versus 30.0%, odds ratio (OR)\\u2009=\\u20090.89, 95% confidence interval (CI)\\u2009=\\u20090.62-1.29] or weeks 37 to 52 (33.8 versus 33.2%, OR\\u2009=\\u20091.03, 95% CI\\u2009=\\u20090.73-1.46). AUTHORS' CONCLUSIONS: There is currently no evidence that nicotine vaccines enhance long-term smoking cessation. The RR for 12 month cessation in active and placebo groups was 1.35 (95% Confidence Interval (CI) 0.82 to 2.22) in\\xa0the trial of NIC002\\xa0and 1.74 (95% CI 0.73 to 4.18) in one NicVAX trial. Two Phase III NicVAX trials, for which full results were not available, reported similar quit rates of approximately 11% in both groups.\", 'Natalizumab did not achieve a statistically significant primary composite disability outcome in a trial of 887 patients with secondary progressive MS , but it did demonstrate a benefit on a prespecified component of the 9-Hole Peg Test .  In this review, we summarize the pathophysiological mechanisms involved in the development of SPMS and the rationale and clinical potential for natalizumab, which is currently approved for the treatment of relapsing forms of MS, to exert beneficial effects in reducing disease progression unrelated to relapses in SPMS. In this review, we summarize the pathophysiological mechanisms involved in the development of SPMS and the rationale and clinical potential for natalizumab, which is currently approved for the treatment of relapsing forms of MS, to exert beneficial effects in reducing disease progression unrelated to relapses in SPMS.  INTERPRETATION: Natalizumab treatment for secondary progressive multiple sclerosis did not reduce progression on the primary multicomponent disability endpoint in part 1, but it did reduce progression on its upper-limb component. Natalizumab treatment for secondary progressive multiple sclerosis did not reduce progression on the primary multicomponent disability endpoint in part 1, but it did reduce progression on its upper-limb component.', \"his hypothesis may add a new dimension to the understanding of MS etiology and help to design novel therapeutic agents against potential targets that might be discovered. If this hypothesis proves to be true, tau phosphorylation inhibitors can be potential candidates for MS drug development. The amyloid-forming proteins tau, αB crystallin, and amyloid P protein are all found in lesions of multiple sclerosis (MS). Alzheimer's disease (AD) is the most common cause of progressive dementia in the elderly. It is characterized by a progressive and irreversible loss of cognitive abilities and formation of senile plaques, composed mainly of amyloid β (Aβ), and neurofibrillary tangles (NFTs), composed of tau protein, in the hippocampus and cortex of afflicted human Abnormally hyperphosphorylated tau is the major protein constituent of neurofibrillary tangles (NFTs) in the brain of Alzheimer disease (AD) patients Alzheimer's disease (AD), progressive supranuclear palsy (PSP), frontotemporal dementia and parkinsonism linked to chromosome 17 (FTDP-17), and Pick's disease (PiD) are commonly known as tauopathies.\", 'Our goal was to compare the safety and immunogenicity of a combination vaccine (DTaP(5)-IPV-Hib; Pentacel) with that of its separately administered, US-licensed equivalent vaccines (diphtheria, tetanus, 5-component acellular pertussis vaccine [DTaP(5); Daptacel], inactivated poliovirus vaccine [IPV; IPOL], and Haemophilus influenzae type b [Hib] vaccine [ActHIB]), when administered to infants and toddlers concomitantly with other routinely recommended vaccines and to assess antibody persistence from the fourth dose in toddlers to the fifth (preschool) DTaP(5) dose.', 'Combining in vivo and in vitro findings, we identified nine AhR agonists, six of which are marketed therapeutics and have been approved by the U.S. Food and Drug Administration, including leflunomide, flutamide, and nimodipine. However, recent discoveries of new AhR ligands with potential therapeutic applications have been reported, inviting reconsideration of this policy. However, recent discoveries of new AhR ligands with potential therapeutic applications have been reported, inviting reconsideration of this policy. However , recent discoveries of new AhR ligands with potential therapeutic applications have been reported , inviting reconsideration of this policy  Based on our review of the data, there is little evidence to support the indiscriminate exclusion of AhR activators/Cyp1a1 inducers from early drug developmental pipelines. We also found no evidence that short-term treatment with RMAhRLs produce \"dioxin-like toxicity\"', 'Emerging lines of evidence indicate that histone variants (H2AX and H2A.Z), histone post-translational modifications (acetylation, phosphorylation, methylation and ubiquitination) and chromatin-remodeling complexes (INO80, SWR1, SWI/SNF, RSC and NuRD) are important and direct players in the DNA double-strand break (DSB) response as well. In contrast, H2A.Z deposition is dispensable for nucleosome positioning.  Accordingly, the absence of SWR-C or histone H2A.Z results in compromised chromatin remodeling and impaired gene expression in the absence of RSC and H3K4 methylation. In RSC-depleted cells, NFRs shrink such that the average positions of flanking nucleosomes move toward predicted sites. In contrast, the upstream nucleosome which covers the TATA box under repressed conditions is shifted approximately 50 bp further upstream by the ATP-dependent chromatin remodeler RSC upon activation. It is marked with the histone variant H2A.Z and H4K16 acetylation in active state', 'Amifostine is the first FDA approved cytoprotective and chemoprotective agent in the treatment of cancer.   Tea polyphenols (TPs) have been shown to reduce radiation-induced damage in multiple studies, CAPE was found to act both as radioprotector and radiosensitizer. Metformin as a radiation modifier; implications to normal tissue protection and tumor sensitization. Recent studies have shown that alpha-2-macroglobulin (α2M) possesses radioprotective effects. ', 'ur studies indicate that both nuclear matrix association and DNA binding are required for optimal SATB1-mediated repression of the integrated MMTV promoter and may allow insulation from cellular regulatory elements. knock-out of Satb1 significantly inhibited cell viability and migration, and promoted Schwann cells apoptosis. SATB1 is essential for maintaining TCR responsiveness during the induction and effector phases and may provide a novel therapeutic target for T cell-mediated autoimmune diseases.  T cells from SATB1 conditional knockout (SATB1cKO) mice, in which the Satb1 gene is deleted from hematopoietic cells, impair phosphorylation of signaling molecules in response to T cell receptor (TCR) crosslinking', 'This system was successfully tested in a Phase 2 clinical trial for the treatment of post-menopausal women with osteoporosis.', 'Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. The identification and characterization of signal pleckstrin homology (PH)-domain-binding motifs established critical sites for breast cancer growth in PAR1&2. The identification and characterization of signal pleckstrin homology (PH)-domain-binding motifs established critical sites for breast cancer growth in PAR1&2. Although emerging roles of protease-activated receptor1&2 (PAR1&2) in cancer are recognized, their underlying signalling events are poorly understood. Here we show signal-binding motifs in PAR1&2 that are critical for breast cancer growth. This occurs via the association of the pleckstrin homology (PH) domain with Akt/PKB as a key signalling event of PARs. Other PH-domain signal-proteins such as Etk/Bmx and Vav3 also associate with PAR1 and PAR2 through their PH domains. PAR1 and PAR2 bind with priority to Etk/Bmx. A point mutation in PAR2, H349A, but not in R352A, abrogates PH-protein association and is sufficient to markedly reduce PAR2-instigated breast tumour growth in vivo and placental extravillous trophoblast (EVT) invasion in vitro. Similarly, the PAR1 mutant hPar1-7A, which is unable to bind the PH domain, reduces mammary tumours and EVT invasion, endowing these motifs with physiological significance and underscoring the importance of these previously unknown PAR1 and PAR2 PH-domain-binding motifs in both pathological and physiological invasion processes.', 'Delays in vaccination can stymie the development of herd immunity, and a large proportion of children in the U.S. are known not to receive vaccines on time. Vaccines are very effective in providing individual and community (herd) immunity against a range of diseases.  We argue that individuals who have access to vaccines and for whom vaccination is not medically contraindicated have a moral obligation to contribute to the realisation of herd immunity by being vaccinated. is to drive population control of disease beyond those who are vaccinated (i.e. through herd immunity). ', 'Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. There was no difference between the multipotent adult progenitor cell group and placebo groups in global stroke recovery at day 90 (odds ratio 1·08 [95% CI 0·55-2·09], p=0·83).INTERPRETATION: Administration of multipotent adult progenitor cells was safe and well tolerated in patients with acute ischaemic stroke. Although no significant improvement was observed at 90 days in neurological outcomes with multipotent adult progenitor cells treatment, further clinical trials evaluating the efficacy of the intervention in an earlier time window after stroke (<36 h) are planned. INTERPRETATION\\n\\nAdministration of multipotent adult progenitor cells was safe and well tolerated in patients with acute ischaemic stroke.', ' MS and hospitalized were included and analyzed after which they were divided into the primary progressive MS A and B groups, the relapsing-remitting MS (RRMS) C and D groups  relapsing-remitting MS, primary and secondary progressive MS)  Multiple sclerosis presents with different phenotypes, most commonly a relapsing-remitting course and, less frequently, a progressive accumulation of disability from disease onset (primary progressive multiple sclerosis). The majority of people with relapsing-remitting multiple sclerosis, after a variable time, switch to a stage characterised by gradual neurological worsening known as secondary progressive multiple sclerosis.', ' The anti-LINGO-1 trial showed that the drug is safe and tolerable. A future phase II trial will provide more insights regarding the compound. Based on our knowledge of pathophysiology, three therapeutic strategies are proposed: anti-inflammatory (ocrelizumab, siponimod…); remyelinating (opicinumab); and neuroprotective (high-dose biotin, ibudilast, simvastatin…). Nevertheless, despite recent promising positive clinical trials, new methodological approaches for therapeutic protocols with adaptable outcomes to assess progression are still needed. Anti-LINGO-1 (opicinumab) is the first investigational product that achieved phase I trial with the aim of remyelination and axonal protection and/or repair in MS. Anti-LINGO-1 (opicinumab) is the first investigational product that achieved phase I trial with the aim of remyelination and axonal protection and/or repair in MS.  Participants (aged 18-58 years) with relapsing multiple sclerosis (relapsing-remitting multiple sclerosis and secondary progressive multiple sclerosis with relapses) were randomised in a 1:2:2:2:2 ratio by an interactive voice and web response system to opicinumab 3 mg/kg, 10 mg/kg, 30 mg/kg, or 100 mg/kg, or placebo.', 'In this respect , several B cell-targeted therapies emerged , including anti-CD20 antibodies ( rituximab , ocrelizumab , and ofatumumab) , anti-CD19 antibody ( inebilizumab) , and agents targeting the BAFF/APRIL signaling pathway ( atacicept , belimumab , and LY2127399) .  In this respect, several B cell-targeted therapies emerged, including anti-CD20 antibodies (rituximab, ocrelizumab, and ofatumumab), anti-CD19 antibody (inebilizumab), and agents targeting the BAFF/APRIL signaling pathway (atacicept, belimumab, and LY2127399). In this respect, several B cell-targeted therapies emerged, including anti-CD20 antibodies (rituximab, ocrelizumab, and ofatumumab), anti-CD19 antibody (inebilizumab), and agents targeting the BAFF/APRIL signaling pathway (atacicept, belimumab, and LY2127399).  In NMO , though there have yet to be any approved monoclonal antibodies , rituximab , anti-complement C5 ( eculizumab) , anti-IL-6 receptor ( tocilizumab) , anti-CD19 ( inebilizumab ) and non-pathogenic anti-aquaporin 4 ( aquaporumab ) have been suggested to be effective , and some of these are now under clinical trials .  In NMO, though there have yet to be any approved monoclonal antibodies, rituximab, anti-complement C5 (eculizumab), anti-IL-6 receptor (tocilizumab), anti-CD19 (inebilizumab) and non-pathogenic anti-aquaporin 4 (aquaporumab) have been suggested to be effective, and some of these are now under clinical trials.', 'At week 4, the proportion of the patients able to walk independently (functional grade ≤2) was 61% (90% CI 42-78; n=14) in the eculizumab group, and 45% (20-73; n=5) in the placebo group. Adverse events occurred in all 34 patients. Three patients had serious adverse events: two in the eculizumab group (anaphylaxis in one patient and intracranial haemorrhage and abscess in another patient) and one in the placebo group (depression). For the primary efficacy outcome at 4 weeks after recruitment, two of two placebo- and two of five eculizumab-treated subjects had improved by one or more grades on the GBS DS. At week 4, the proportion of the patients able to walk independently (functional grade ≤2) was 61% (90% CI 42-78; n=14) in the eculizumab group, and 45% (20-73; n=5) in the placebo group. A recent Japanese randomized controlled trial with eculizumab, a monoclonal antibody against the complement C5, indicated that eculizumab might improve the outcomes of GBS patients at six months from onset. However, because this is a small study without statistical comparison with the placebo group, the efficacy and safety of eculizumab could be investigated in larger, randomised controlled trials.', 'Mcl-1 inhibitor S63845 The recent description of S63845 as the first specific and potent MCL-1 inhibitor represents an important therapeutic advance  We tested the efficacy of the BH3 mimetic combination of A-1210477 (an MCL-1 inhibitor) A-1210477, a selective MCL-1 inhibitor a highly specific MCL-1 inhibitor, S63845,', 'Autophagy is a cellular survival pathway that is necessary for the degradation of cellular constituents such as long-lived proteins and damaged organelles. Autophagy is an evolutionarily conserved cellular process in which intracellular components are eliminated via lysosomal degradation to supply nutrients for organelle biogenesis and metabolic homeostasis. Autophagy is an evolutionarily conserved cellular process in which intracellular components are eliminated via lysosomal degradation to supply nutrients for organelle biogenesis and metabolic homeostasis. Autophagy is a homeostatic process involved in the turnover or elimination of cytoplasmic components, damaged organelles, and protein aggregates via a lysosomal degradation mechanism. In this study, we demonstrate that autophagy is a critical mediator of the viral degradation pathway and that this pathway is not HIV-1 specific.', 'MPE-seq provides a unique and straightforward means for the genome-wide analysis of chromatin structure with minimal DNA sequence bias.  methidiumpropyl-EDTA sequencing (MPE-seq), a method for the genome-wide characterization of chromatin that involves the digestion of nuclei withMPE-Fe(II) followed by massively parallel sequencing. ', 'Two rotavirus vaccines, RotaTeq and Rotarix, are licensed for global use; however, the protection they confer to unvaccinated individuals through indirect effects remains unknown. ', 'This revealed a 637-kb tandem duplication that in addition to DAX1 includes the four MAGEB genes, the hypothetical gene CXorf21, GK, and part of the MAP3K7IP3 gene Examination of X-linked DEGs, such as GTPBP6, TAF9L, and CXORF21, that show verbal cognition-gene expression correlations may establish a causal link between these genes, neurodevelopment, and language function.  For this, we selected five SNPs (rs1801274 in FCGR2A and rs2286672 in PLD2, rs887369 in CXorf21, rs9782955 in LYST, and rs3794060 in NADSYN1) Among statin users, diabetes cases had marginal but insignificantly different expression of ZNF532 (up-regulated 15%, Q-value=0.0584), CXORF21 (up-regulated 11%, Q-value=0.0584), and ZNHIT3 (up-regulated 19%, Q-value=0.0959), compared with controls.', 'In the context of genome-wide association studies (GWAS), there is a variety of statistical techniques in order to conduct the analysis, but, in most cases, the underlying genetic model is usually unknown. Under these circumstances, the classical Cochran-Armitage trend test (CATT) is suboptimal. Robust procedures that maximize the power and preserve the nominal type I error rate are preferable. Moreover, performing a meta-analysis using robust procedures is of great interest and has never been addressed in the past. The primary goal of this work is to implement several robust methods for analysis and meta-analysis in the statistical package Stata and subsequently to make the software available to the scientific community.Results: The CATT under a recessive, additive and dominant model of inheritance as well as robust methods based on the Maximum Efficiency Robust Test statistic, the MAX statistic and the MIN2 were implemented in Stata. Concerning MAX and MIN2, we calculated their asymptotic null distributions relying on numerical integration resulting in a great gain in computational time without losing accuracy. All the aforementioned approaches were employed in a fixed or a random effects meta-analysis setting using summary data with weights equal to the reciprocal of the combined cases and controls. Overall, this is the first complete effort to implement procedures for analysis and meta-analysis in GWAS using Stata.Availability and Implementation: A Stata program and a web-server are freely available for academic users at http://www.compgen.org/tools/GWAR. Availability and Implementation\\n\\nA Stata program and a web-server are freely available for academic users at http://www.compgen.org/tools/GWAR.', 'Single-cell RNA-sequencing (scRNA-seq) allows studying heterogeneity in gene expression in large cell populations. Such heterogeneity can arise due to technical or biological factors, making decomposing sources of variation difficult. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity. Our model jointly estimates the relevance of individual factors, refines gene set annotations, and infers factors without annotation. In applications to multiple scRNA-seq datasets, we find that f-scLVM robustly decomposes scRNA-seq datasets into interpretable components, thereby facilitating the identification of novel subpopulations. f-scLVM: scalable and versatile factor analysis for single-cell RNA-seq f-scLVM: scalable and versatile factor analysis for single-cell RNA-seq. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity. We here describe f-scLVM (factorial single-cell latent variable model), a method based on factor analysis that uses pathway annotations to guide the inference of interpretable factors underpinning the heterogeneity.', 'Animal promoters initiate transcription either at precise positions (narrow promoters) or dispersed regions (broad promoters), a distinction referred to as promoter shape. Although highly conserved, the functional properties of promoters with different shapes and the genetic basis of their evolution remain unclear. Here we used natural genetic variation across a panel of 81 Drosophila lines to measure changes in transcriptional start site (TSS) usage, identifying thousands of genetic variants affecting transcript levels (strength) or the distribution of TSSs within a promoter (shape). Our results identify promoter shape as a molecular trait that can evolve independently of promoter strength. Broad promoters typically harbor shape-associated variants, with signatures of adaptive selection. Single-cell measurements demonstrate that variants modulating promoter shape often increase expression noise, whereas heteroallelic interactions with other promoter variants alleviate these effects. These results uncover new functional properties of natural promoters and suggest the minimization of expression noise as an important factor in promoter evolution. Promoter shape varies across populations and affects promoter evolution and expression noise. Promoter shape varies across populations and affects promoter evolution and expression noise', ' Long noncoding RNAs (lncRNAs) are more than 200 nucleotides in length and lack transcriptional ability. lncRNAs are the nonprotein coding RNAs that have a size longer than 200 nucleotides.   lncRNAs are defined as RNA transcripts longer than 200 nucleotides that are not transcribed into proteins  long noncoding RNA (lncRNA, >200 nucleotides)', 'This phase 2, randomised, double-blind, placebo-controlled, single-centre, crossover trial assessed the effectiveness of an oral neurokinin 3 receptor antagonist (MLE4901) on menopausal hot flushes.  Treatment with a neurokinin 3 receptor antagonist (MLE4901) could be practice changing as it safely and effectively relieves hot flush symptoms without the need for oestrogen exposure. ', 'Chloroquine is a potent inhibitor of SARS coronavirus infection and spread. we investigated how chloroquine (CQ), a commonly employed disease-modifying antirheumatic drug, Historically chloroquine was used to treat the most deadly form of malaria, caused by the parasite Plasmodium falciparum.  Chloroquine (CQ) has been used for decades as the primary chemotherapeutic drug for the treatment of malaria.  Hydroxychloroquine (HCQ), a 4-aminoquinolone antimalarial, is regarded as the oral therapy of choice for cutaneous and systemic lupus erythematosus (SLE). It is also licensed for rheumatoid arthritis (RA).', 'The medications comprised vitamin E, baclofen, riluzole, L-threonine, xaliproden, indinavir, and memantine. Six studies assessed cramps as an adverse event. The medications comprised creatine, gabapentin, dextromethorphan, quinidine, and lithium. In all 20 studies no favourable effect for the treatment of cramps in ALS/MND could be demonstrated, but many studies were underpowered to draw a definite conclusion. An effect of xaliproden on functional parameters, especially VC, was noted. The six months intent-to-treat analysis showed no statistically significant effect but a trend in favour of 2 mg xaliproden compared to placebo for reduction in the rate of deterioration of FVC, limbs functional score, and manual muscle testing score (MMT). The six months intent-to-treat analysis showed no statistically significant effect but a trend in favour of 2 mg xaliproden compared to placebo for reduction in the rate of deterioration of FVC, limbs functional score, and manual muscle testing score (MMT). An effect of xaliproden on functional parameters, especially VC, was noted. Although this effect did not reach statistical significance, xaliproden had a small effect on clinically noteworthy aspects of disease progression in ALS.', 'Telangiectasia (macroscopically visible dilated skin vessels) occurring primarily on the hands and face, are a prominent feature in scleroderma and are present in the majority of patients. Telangiectasias are prominent small vessels (venules, capillaries or arterioles) that are visible as small red-purple focal lesions in the skin and mucous membranes. Hereditary hemorrhagic telangiectasia is a dominant disorder characterized by epistaxis, visceral arteriovenous malformations, and telangiectasias. Telangiectasias are small focal red macules and papules created by abnormally prominent capillaries, venules, and arterioles  Telangiectasia (macroscopically visible dilated skin vessels)', 'Immunofluorescence and immunoblotting investigations revealed the presence of cathepsin L in the nuclear compartment in addition to its expected endo-lysosomal localization in colorectal carcinoma cells. endosomal cathepsin L cleavage by the endosomal/lysosomal protease cathepsin L Cathepsin L in the Late Endosome/Lysosome', 'ACE2 has been identified in various tissues and is supposed to be a modulator of cardiovascular function. Decreases in ACE2 expression and activity have been reported in models of hypertension, heart failure, atherosclerosis, diabetic nephropathy and others.  Abundant ACE2 immunostaining was found in lung, kidney, heart, and islets of pancreas, but not in hepatocytes ACE2 is expressed at high level in the primary target cells of SARS-CoV, namely pneumocytes and surface enterocytes of the small intestine. in keeping with the variability of ACE2 expression we have observed previously endothelial cells, which express ACE2 to a high level,', 'We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). We selected one clone , LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange ( RMCE Transgenesis of human pluripotent stem cells (hPSCs) can enable and empower a variety of studies in stem cell research, including lineage tracing and functional genetics studies. While in recent years much progress has been made in the development of tools for gene targeting, little attention has been given to the identification of sites in the human genome where transgenes can be inserted and reliably expressed. In order to find human genomic sites capable of supporting long-term and high-level transgene expression in hPSCs, we performed a lentiviral screen in human induced pluripotent stem cells (iPSCs). We isolated 40 iPSC clones each harboring a single vector copy and characterized the level of transgene expression afforded by each unique integration site. We selected one clone, LiPS-A3 with an integration site in chromosome 15 maintaining robust expression without silencing and demonstrate that different transgenes can be inserted therein rapidly and efficiently through recombinase-mediated cassette exchange (RMCE). The LiPS-A3 line can greatly facilitate the insertion of reporter and other genes in hPSCs. Targeting transgenes in the LiPS-A3S genomic locus can find broad applications in stem cell research and possibly cell and gene therapy. LiPS-A3S, a human genomic site for robust expression of inserted transgenes.', 'A random effects meta-analytical model was used to compare adverse events and functional outcome.RESULTS: Thirty comparative studies comparing J, W, S and K pouch designs were included Formation of a colonic J-pouch with anastomosis to the rectal stump is an accepted form of reconstruction after low anterior resection (LAR) for rectal carcinoma Ileal pouch-anal anastomosis, or J pouch, surgery has become the procedure of choice for treatment of medically refractory ulcerative colitis and familial adenomatous polyposis. total proctocolectomy with outcomes after ileal J-pouch anal anastomosis (IPAA) at a single institution', 'SLC12A1 (NKCC2) Na-K-Cl cotransporter 2 (NKCC2) 2 chloride co-transporter (NKCC2) gene Na-K-2Cl cotransporter (NKCC2) function ', 'RESULTS\\n\\nCelecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. RESULTS\\nCelecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. In conclusion, the celecoxib-creatine combination was selected as preferable to the minocycline-creatine combination for further evaluation.  RESULTS: Celecoxib did not slow the decline in muscle strength, vital capacity, motor unit number estimates, ALS Functional Rating Scale-Revised, or affect survival. INTERPRETATION\\n\\nAt the dosage studied, celecoxib did not have a beneficial effect on research subjects with ALS, and it was safe.', 'increased expression of anti-apoptotic proteins (Bcl-xL, Mcl-1 and XIAP)  repression of anti-apoptotic proteins (Mcl-1, Bcl-xl and XIAP) anti-apoptotic BCL-2 family members, such as BCL-2, BCL-XL or MCL-1', 'Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib. Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib. Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib.  Cyclin-dependent kinase 4/6 inhibitors, which act by inhibiting progression from the G1 to S phases of the cell cycle, include palbociclib, ribociclib, abemaciclib, and trilaciclib We show that the coadministration of G1T28 (trilaciclib), which is a small-molecule inhibitor of cyclin-dependent kinases 4 and 6 (CDK4/6), contemporaneously with cytotoxic chemotherapy protects murine hematopoietic stem cells (HSCs) from chemotherapy-induced exhaustion in a serial 5-fluorouracil treatment model. ', 'To evaluate the clinical PK/PD of PTH(1-34) delivered by a novel transdermal drug-coated microneedle patch system (ZP-PTH) for the treatment of osteoporosis.', 'Parathyroid hormone (1-34)-coated microneedle patch system: clinical pharmacokinetics and pharmacodynamics for treatment of osteoporosis. To evaluate the clinical PK/PD of PTH(1-34) delivered by a novel transdermal drug-coated microneedle patch system (ZP-PTH) for the treatment of osteoporosis.', \"The highly variable phenotypes observed in patients with classic Bartter's syndrome (BS) remain unsatisfactorily explained. The wide spectrum of functional severity of CLCNKB mutations may contribute to the phenotypic variability, and the genotype-phenotype association has not been established.  Classic Bartter's syndrome has been demonstrated to result from defective chloride transport across the basolateral membrane in the distal nephron due to mutations in the chloride channel gene CLCNKB. Progressive renal failure in patients with classic Bartter's syndrome (cBS) due to inactivating mutations in CLCNKB gene is extraordinarily rare classic Bartter's syndrome by mutations of ClC-Kb\", 'We developed a search system named UC2 (Unique Connectivity of Uncharged Compounds), where compounds are tentatively neutralized into uncharged states and stored on the basis of their unique connectivity of atoms after removing their stereochemical information using the first block in the hash of the IUPAC International Chemical Identifier, by which false-positive hits are remarkably reduced, both charged and uncharged compounds are properly searched in a single query and records having a unique connectivity are compiled in a single search result.Availability and implementation: The UC2 search tool is available free of charge as a REST web service (http://webs2.kazusa.or.jp/mfsearcher) and a Java-based GUI tool.Contact: sakurai@kazusa.or.jp.Supplementary information: Supplementary data are available at Bioinformatics online. For metabolite annotation in metabolomics, variations in the registered states of compounds (charged molecules and multiple components, such as salts) and their redundancy among compound databases could be the cause of misannotations and hamper immediate recognition of the uniqueness of metabolites while searching by mass values measured using mass spectrometry. We developed a search system named UC2 (Unique Connectivity of Uncharged Compounds), where compounds are tentatively neutralized into uncharged states and stored on the basis of their unique connectivity of atoms after removing their stereochemical information using the first block in the hash of the IUPAC International Chemical Identifier, by which false-positive hits are remarkably reduced, both charged and uncharged compounds are properly searched in a single query and records having a unique connectivity are compiled in a single search result.Availability and implementation: The UC2 search tool is available free of charge as a REST web service (http://webs2.kazusa.or.jp/mfsearcher) and a Java-based GUI tool. UC2 search: using unique connectivity of uncharged compounds for metabolite annotation by database searching in mass spectrometry-based metabolomics. UC2 search: using unique connectivity of uncharged compounds for metabolite annotation by database searching in mass spectrometry-based metabolomics', 'Administration of senktide, highly potent and selective NK3R agonist, resulted in increase of serum LH concentration, induction of VMS, increase in heart rate, and skin temperature in postmenopausal women.', 'Either angiotensin converting enzyme inhibitor (ACEI) or angiotensin receptor 1 blocker (ARB) attenuates cardiac remodeling. However, the overall molecular modulation of the reversing remodeling process in response to the ACEI or ARB treatment is not yet well determined. In this study, we examined whether gene expressions are modulated by ACEI (temocapril), ARB (olmesartan)  ARB olmesartan (OLM) Olmesartan (OL) is the pharmacologically active metabolite of Olmesartan medoxomil (OM), an FDA-approved angiotensin II receptor antagonist for administrating cardiovascular diseases.', 'X-inactivation escape of the TLR7 gene was investigated in monoclonal B cell lines and, independently, in pDCs after cell sorting and single-cell picking, indicating regular silencing of one TLR7 allele in females.  TLR7 is encoded by an X chromosome locus, and we examined here whether the TLR7 gene evades silencing by X chromosome inactivation in immune cells from women and Klinefelter syndrome males Female predisposition to TLR7-driven autoimmunity: gene dosage and the escape from X chromosome inactivation', 'We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. Recent technological advances have enabled DNA methylation to be assayed at single-cell resolution. However, current protocols are limited by incomplete CpG coverage and hence methods to predict missing methylation states are critical to enable genome-wide analyses. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells. We evaluate DeepCpG on single-cell methylation data from five cell types generated using alternative sequencing protocols. DeepCpG yields substantially more accurate predictions than previous methods. Additionally, we show that the model parameters can be interpreted, thereby providing insights into how sequence composition affects methylation variability. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells. We report DeepCpG, a computational approach based on deep neural networks to predict methylation states in single cells.', 'After a decade in obscurity it (Vaxchora) has resurfaced again, now produced in the U.S. and equipped with a U.S. FDA license (June 10, 2016). ', 'We evaluated the efficacy and tolerability of 3- and 6-month leuprorelin acetate (LA) depot formulations (Eligard®, Astellas Pharma GmbH) in patients with advanced prostate cancer treated in routine clinical practice in Germany. Efficacy and Tolerability of Leuprorelin Acetate (Eligard®) in Daily Practice in Germany: Pooled Data from 2 Prospective, Non-Interventional Studies with 3- or 6-Month Depot Formulations in Patients with Advanced Prostate Cancer.', 'We evaluated the efficacy and tolerability of 3- and 6-month leuprorelin acetate (LA) depot formulations (Eligard®, Astellas Pharma GmbH) in patients with advanced prostate cancer treated in routine clinical practice in Germany.', 'XenofilteR: computational deconvolution of mouse and human reads in tumor xenograft sequence data. Mouse xenografts from (patient-derived) tumors (PDX) or tumor cell lines are widely used as models to study various biological and preclinical aspects of cancer. However, analyses of their RNA and DNA profiles are challenging, because they comprise reads not only from the grafted human cancer but also from the murine host. The reads of murine origin result in false positives in mutation analysis of DNA samples and obscure gene expression levels when sequencing RNA. However, currently available algorithms are limited and improvements in accuracy and ease of use are necessary.RESULTS: We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. To assess the accuracy of XenofilteR, we generated sequence data by in silico mixing of mouse and human DNA sequence data. These analyses revealed that XenofilteR removes >\\u200999.9% of sequence reads of mouse origin while retaining human sequences. This allowed for mutation analysis of xenograft samples with accurate variant allele frequencies, and retrieved all non-synonymous somatic tumor mutations.CONCLUSIONS: XenofilteR accurately dissects RNA and DNA sequences from mouse and human origin, thereby outperforming currently available tools. XenofilteR is open source and available at https://github.com/PeeperLab/XenofilteR . We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. RESULTS: We developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome. RESULTS\\n\\nWe developed the R-package XenofilteR, which separates mouse from human sequence reads based on the edit-distance between a sequence read and reference genome.', 'Here, we show that the elephant genome encodes 20 copies of the tumor suppressor gene TP53 and that the increase in TP53 copy number occurred coincident with the evolution of large body sizes, the evolution of extreme sensitivity to genotoxic stress, and a hyperactive TP53 signaling pathway in the elephant (Proboscidean) lineage.  While humans have 1 copy (2 alleles) of TP53, African elephants have at least 20 copies (40 alleles), including 19 retrogenes (38 alleles) with evidence of transcriptional activity measured by reverse transcription polymerase chain reaction. ', 'MannKind Corporation has developed a powder formulation of insulin that allows for a high percentage of the administered insulin to be absorbed via the lung. Their product, AFREZZA (Technosphere insulin), is currently under review by the FDA for use in patients with diabetes. ', 'Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. Applying DeepSynergy for classification of these novel drug combinations resulted in a high predictive performance of an AUC of 0.90. DeepSynergy significantly outperformed the other methods with an improvement of 7.2% over the second best method at the prediction of novel drug combinations within the space of explored drugs and cell lines.', 'The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12). The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12). The adjusted mean of the joint-rank score was -14·9 (SE 13·5) for the ozanezumab group and 15·0 (13·6) for the placebo group, with a least squares mean difference of -30·0 (95% CI -67·9 to 7·9; p=0·12).  INTERPRETATION Ozanezumab did not show efficacy compared with placebo in patients with ALS. INTERPRETATION\\n\\nOzanezumab did not show efficacy compared with placebo in patients with ALS.', 'At treatment end, no significant differences were seen for change in MADRS total score with TC-5214 versus placebo. At treatment end, no significant differences were seen for change in MADRS total score with TC-5214 versus placebo. Furthermore, there were no significant differences in any of the secondary endpoints.  No notable differences were observed between dexmecamylamine and placebo for any secondary end point. TC-5214 (dexmecamylamine) is a nicotinic channel modulator that has previously been evaluated for treatment of major depression disorder (MDD) and is currently being evaluated by Targacept as a treatment for overactive bladder. TC-5214 (dexmecamylamine) is a nicotinic channel modulator that has previously been evaluated for treatment of major depression disorder (MDD) and is currently being evaluated by Targacept as a treatment for overactive bladder. ', 'easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. The ever-growing availability of high-quality genotypes for a multitude of species has enabled researchers to explore the underlying genetic architecture of complex phenotypes at an unprecedented level of detail using genome-wide association studies (GWAS). The systematic comparison of results obtained from GWAS of different traits opens up new possibilities, including the analysis of pleiotropic effects. Other advantages that result from the integration of multiple GWAS are the ability to replicate GWAS signals and to increase statistical power to detect such signals through meta-analyses. In order to facilitate the simple comparison of GWAS results, we present easyGWAS, a powerful, species-independent online resource for computing, storing, sharing, annotating, and comparing GWAS. The easyGWAS tool supports multiple species, the uploading of private genotype data and summary statistics of existing GWAS, as well as advanced methods for comparing GWAS results across different experiments and data sets in an interactive and user-friendly interface. easyGWAS is also a public data repository for GWAS data and summary statistics and already includes published data and results from several major GWAS. We demonstrate the potential of easyGWAS with a case study of the model organism Arabidopsis thaliana, using flowering and growth-related traits.', 'Data suggesting this vaccine may improve OS for SS and MRCL patients is exciting but early, and on-going work is testing the impact of CMB305 on patient outcomes. CMB305 induces NY-ESO-1 specific T cell responses in both SS and MRC patients and these patients had excellent overall survival (OS) outcomes in the initial phase I study. The potential of the CMB305 vaccine regimen to target NY-ESO-1 and improve outcomes for synovial sarcoma and myxoid/round cell liposarcoma patients.', ' In a randomized, controlled, phase 3, pivotal trial, Quadracel proved to be as efficacious and safe as Daptacel (diphtheria and tetanus toxoids and acellular pertussis vaccine adsorbed, Sanofi Pasteur Inc.) and IPOL (poliovirus vaccine inactivated, Sanofi Pasteur Inc.), given separately, to children between the ages of 4 and 6 years Quadracel (diphtheria and tetanus toxoids and acellular pertussis adsorbed and inactivated poliovirus vaccine, Sanofi Pasteur Inc.) is a new vaccination developed to condense the last dose of both DTaP and IPV so they do not have to be given separately, thus reducing the total number of vaccinations required.', \"On May 9 2011, the US FDA approved Sanofi Pasteur's Fluzone(®) Intradermal influenza vaccine, the first influenza vaccine licensed in the USA that uses a new microinjection system for intradermal delivery of vaccines (Soluvia™, Becton Dickinson). \", 'Despite discontinuation of the first inhalable insulin, Exubera®, due to suboptimal market acceptance, development of orally inhaled insulin delivery systems has been galvanized by the recent approval of Afrezza® and several others awaiting approval.', 'These results strongly suggest that recombination is the primary determinant of the isochore organization of mammalian genomes. Thus, chromosomal position can directly affect the rate of evolution of a gene. Such an accelerated rate of evolution might be due to factors other than natural selection, in particular GC-biased gene conversion. This is true of neutral sequences, but also of constrained sequences, which can be illustrated using the mouse Fxy gen We have previously described a gene, Fxy , that spans the pseudoautosomal boundary in mice such that the first three exons of the gene are located on the X chromosome, but the remainder of the gene is located on both X and Y chromosomes. Therefore, this gene might be in a state of transition between pseudoautosomal and X-unique locations.  The mouse Fxy gene was translocated into the highly recombining pseudoautosomal region comparatively recently in evolutionary terms. This event resulted in a rapid increase of GC content.', 'The insertion is generated by non-allelic homologous recombination between a 548 bp LTR6B repeat within the Y chromosome PAR1 and a second LTR6B repeat located 105 kb from the PAR boundary on the X chromosome.   The four genes within PAR2 have been proposed to constitute two zones with different base ratios and transcription, one of which was added recently to the X chromosome. In contrast, so far only 4 genes have been discovered in the PAR2 region. The pseudoautosomal regions (PAR1 and PAR2) of the human X and Y chromosomes pair and recombine during meiosis. Thus genes in this region are not inherited in a strictly sex-linked fashion. PAR1 is located at the terminal region of the short arms and PAR2 at the tips of the long arms of these chromosomes. To date, 24 genes have been assigned to the PAR1 region. The human sex chromosomes differ in sequence, except for the pseudoautosomal regions (PAR) at the terminus of the short and the long arms, denoted as PAR1 and PAR2. The boundary between PAR1 and the unique X and Y sequences was established during the divergence of the great apes. ']\n"
          ],
          "name": "stdout"
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "vRwNTTlBBvXW"
      },
      "source": [
        "def create_answer_df(test_df,summaries):\n",
        "  qid_test_df = test_df['id']\n",
        "  type_test_df = test_df['type']\n",
        "  summaries_test_df = summaries\n",
        "  return qid_test_df, type_test_df, summaries_test_df"
      ],
      "execution_count": 44,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "RU5LfaAfCIF-"
      },
      "source": [
        "qid_test_df, type_test_df, summaries_test_df = create_answer_df(test_df, topsentences_summary)"
      ],
      "execution_count": 45,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "hcQ70mPYC1qh"
      },
      "source": [
        "question_details = []\n",
        "for i in range(len(qid_test_df)):\n",
        "  dicti = {\"id\" : qid_test_df[i], \"ideal_answer\" : summaries_test_df[i], \"exact_answer\" : \"yes\"}\n",
        "  question_details.append(dicti)"
      ],
      "execution_count": 46,
      "outputs": []
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "v0mF069HDAFb"
      },
      "source": [
        "import json\n",
        "\n",
        "x = {\"questions\" : question_details}\n",
        "\n",
        "# Serializing json \n",
        "json_object = json.dumps(x, indent = 2)\n",
        "\n",
        "with open('/content/drive/MyDrive/Thesis_B/BIOASQ_8b_batch5_bertbase_typeids_results', 'w') as outfile:\n",
        "    outfile.write(json_object)"
      ],
      "execution_count": 47,
      "outputs": []
    }
  ]
}